---
title: "Cleaning d2m Cytoscape workflow"
output: html_notebook
---
### Clean, library and data loading
```{r echo=TRUE, message=FALSE, warning=FALSE}



#Clean
#update.packages(checkBuilt=TRUE, ask=FALSE) after copying from old to new version of R library packages.
rm(list=ls())
.libPaths(new = "/home/marius/R/x86_64-pc-linux-gnu-library/4.1/")
gc()
set.seed(123456)
seed_it <- c(123456)
library("ggplot2")
library("dplyr")
library("gridGraphics")
library("gridExtra")
library("ggrepel")
#library("splashr")
library("DBI")
library("svglite")
library("htmltools")
#BiocManager::install(c("org.Mm.eg.db","org.Dr.eg.db"))
#BiocManager::install("AnnotationDbi")
library("org.Mm.eg.db")
MmOrgDb <- org.Mm.eg.db
library("igraph")
library("psych")
require("org.Dr.eg.db")
require("clusterProfiler")
#Load danio rerio, data base
DrOrgDb <- org.Dr.eg.db
#library("pathview")
#library("sva")
library("ggplot2")
library("gridExtra")
library("edgeR")
library("UpSetR")
library("plyr")
#library("DESeq2")
require("readxl")
library("plotly")
library("data.table")
library("installr")
library("knitr")
library("gtools")
library("vsn")
#library("SummarizedExperiment")
library("gridExtra")
#library("sva")
#library("BiocParallel")
#library("pcaExplorer")
library("biomaRt")
#library("RCurl")
library("curl")
library("stats")
library("rvcheck")
library("clusterProfiler")
require("tidyverse")
library("grid")
library("VennDiagram")
library("RColorBrewer")
#library("DEGreport")
library("ComplexHeatmap")
library("svglite")
library("colorspace")
library("RColorBrewer")
library("circlize")
library("tidyr")
require("dplyr")

#Set the working directory.
sysname <- Sys.info()["sysname"]
if ( sysname == "Linux"){
  setwd("/home/marius/Documents/Projects/prsa/")
  register(MulticoreParam())
}else if (sysname == "Windows"){
  setwd("C:/Users/marius/Documents/Phd/Projects/prsa/")
  register(SnowParam())
}


##Load the data
files <- list.files(all.files = TRUE,
                    #path ="/home/marius/Documents/Projects/prsa/02_Data/STAR/mapping/release102",
                    #path = "D:/PhD/Projects/prsa/02_Data/STAR/mapping/release102/",
                    path = "/media/marius/Samsung_T5/PhD/Projects/prsa/02_Data/STAR/mapping/release102",
                    pattern = "*ReadsPerGene.out.tab",
                    recursive = TRUE, 
                    full.names = TRUE)
#Sort the files by the number of output
files <- files[order(nchar(files))]
#files
#name_files <- gsub("/home/marius/Documents/Projects/prsa/02_Data/STAR/mapping/release102/","",files)
# name_files <- gsub("D:/PhD/Projects/prsa/02_Data/STAR/mapping/release102/","",files)
name_files <- gsub("/media/marius/Samsung_T5/PhD/Projects/prsa/02_Data/STAR/mapping/release102/","",files)
#Subsittute all the string for a shorter name.
name_files <- gsub("_ReadsPerGene.out.tab","",name_files)
#Set the names
names(files) <- name_files
#head(files)
#Read it inside lists.
#lapply(names(files),function(x) head(x))
list_counts <- lapply(files,function(fj) read.table(fj,header = FALSE))
paste0("List of ",length(list_counts)," samples.\n")

#Remove the first 4 rows as they are now useful for our analysis.
matrix_counts <- do.call(cbind,lapply(files,function(n) read.table(n,header = FALSE)[-c(0:4),3]))
dim(matrix_counts)
```

### Design table and counts matrix 

```{r echo=TRUE, warning=FALSE}
####################
#Set the Gene Names#
#  in the Matrix   #
####################
#setwd("../")
read.table(files[1],header = TRUE)[,c(1,3)]#[-c(0:4),3]
gene_names_1 <- read.table(files[1],header = FALSE)[-c(0:4),c(1,3)]
gene_names_3 <- read.table(file = files[3],header = FALSE)[-c(0:4),c(1,3)]
gene_names_11 <- read.table(files[11],header = FALSE)[-c(0:4),c(1,3)]

if (all.equal(gene_names_1$V1,gene_names_11$V1,gene_names_3$V1)) {
  cat("All good, GENENAMES are equal to the different samples!")
  rownames(matrix_counts) <- gene_names_1$V1
  cat("\nMatrix Counts rownames, established to the rownmaes of the first sample")
}else{
  cat("STOP\nSTOP\nSTOP\nDATA MUST BE REARRANGED\nFOR A CORRECT READING OF THE SAMPLE TABLE!")
}
###########################################
#                                         #
#       SAMPLE TABLE                      #
#                                         #
###########################################
#Create the sample table for cleaning the batch effects.
mc_s2c <- data.frame(colnames(matrix_counts))
#rownames(mc_s2c) <- mc_s2c$colnames.matrix_counts.
#mc_s2c$colnames.matrix_counts. <- NULL
colnames(mc_s2c) <- "ID"
#mc_s2c


#s2c <- data.frame(read.table("/home/marius/Documents/Projects/prsa/02_Data/STAR/mapping/release102/table_mapps.tsv", sep = "\t", header = TRUE))
# s2c <- data.frame(read.table("D:/PhD/Projects/prsa/02_Data/STAR/mapping/release102/table_mapps.tsv",
s2c <- data.frame(read.table("/media/marius/Samsung_T5/PhD/Projects/prsa/02_Data/STAR/mapping/release102/table_mapps.tsv",
                             sep = "\t",
                             header = TRUE))
#s2c_raw <- data.frame(read.table("/home/marius/Documents/Projects/prsa/02_Data/STAR/mapping/release102/s2c_raw.tsv",sep="\t",header = TRUE))
# s2c_raw <- data.frame(read.table("D:/PhD/Projects/prsa/02_Data/STAR/mapping/release102/s2c_raw.tsv",
s2c_raw <- data.frame(read.table("/media/marius/Samsung_T5/PhD/Projects/prsa/02_Data/STAR/mapping/release102/s2c_raw.tsv",
                                 sep="\t",
                                 header = TRUE))

#s2c
#s2c_raw
#s2c <- s2c[sort.list(s2c$V3,decreasing = TRUE),]
#s2c_raw <- s2c_raw[sort.list(s2c_raw$V1,decreasing = TRUE),]
#all.equal(s2c$V3,s2c_raw$V1)

#s3c <- merge(x = s2c_raw,y = s2c,by.x="V1",by.y="V3",sort=TRUE,all.x=TRUE)
s3c <- plyr::join(x = s2c_raw,y = s2c)

#?merge
#rownames(mc_s2c)
#s4c <- merge(mc_s2c,s3c,by.x="row.names",by.y="V1",all=TRUE)
s4c <- plyr::join(mc_s2c,s3c)
#rownames(mc_s2c)==(s4c$V1)


rownames(s4c) <- s4c[,1]
s4c[1] <- NULL
#s4c$Condition[c(22:24)] <- "Amputation"
#s4c$Length[c(22:24)] <- "mapp150"
#rownames(s4c)==mc_s2c$ID

#Control the sample table and the data are logically related
#if (all(colnames(matrix_counts)!=(rownames(s3c)))){
if (all(colnames(matrix_counts)!=(rownames(s4c)))){  
  cat("STOP\nSTOP\nSTOP\nDATA MUST BE REARRANGED\nFOR A CORRECT READING OF THE SAMPLE TABLE!")
  #print("STOP\nSTOP\nSTOP\nDATA MUST BE REARRANGED\nFOR A CORRECT READING OF THE SAMPLE TABLE!")
}else{
  cat("All good, sampletable is correlated to the samples!")
  #print("All good, sampletable is correlated to the samples!")
}
#Table is READY if results in TRUE.
#s5c_raw <- data.frame(read_xlsx("../02_Data/STAR/mapping/release102/SelectedCorrectingTable (3).xlsx",col_names = TRUE,sheet = "Guide"))
#s5c_raw
#s4c
(s4c)
s4c$Length <- gsub(x = s4c$Length,pattern = "mapp120",replacement = "mapp150")
s4c
```


```{r echo=TRUE, warning=FALSE}
s4c$Lab <- c(rep(x = "A",4),
             rep(x = "B",8),
             rep(x = "C",6),
             rep(x = "D",3),
             rep(x = "E",3),
             rep(x ="F",6),
             rep(x = "G",6))
s4c$Platform <- c(rep("GenomeAnalyzerII",4),rep("NextSeq500",8),rep("BGISEQ500",6),rep("HiSeqXTen",3),rep("HiSeq2000",3),rep("BGISEQ500",6),rep("NovaSeq6000",6))
#s4c$Platform <- c(rep("HiSeq2000",3),rep("BGISEQ500",6),rep("NovaSeq6000",6),rep("GenomeAnalyzerII",4),rep("NextSeq500",8),rep("BGISEQ500",6),rep("HiSeqXTen",3))
#s4c$Platform <- c(rep("HiSeq2000",3),rep("BGISEQ500",6),rep("NovaSeq6000",6),rep("GenomeAnalyzerII",4),rep("NextSeq500",8),rep("BGISEQ500",6),rep("HiSeqXTen",3))
s4c$Tissue <- c(rep("Ventricle",12),rep("Heart",6),rep("Ventricle",18))
#s4c$Tissue <- c(rep("Ventricle",27),rep("Heart",6),rep("Ventricle",3))
#s4c$Length <- gsub(x = s4c$Length,pattern = "mapp",replacement = "")
#s4c$Length <- c(rep("150",3),rep("50",6),rep("150",6),rep("50",4),rep("76",8),rep("50",6),rep("150",3))
s4c$Length <- c(rep("50",4),rep("75",8),rep("50",6),rep("150",6),rep("50",6),rep("150",6))
s4c$Library <- ifelse(as.integer(s4c$Length) > 100, "Paired", "Single")
#s4c$Extract <- c(rep("polyA",3),rep("totalRNA",12),rep("polyA",18),rep("totalRNA",3))
s4c$Extract <- c(rep("polyA",18),rep("totalRNA",3),rep("polyA",3),rep("totalRNA",12))

#head(s4c)
#write.table(file = "/home/marius/Documents/Projects/prsa/BatchCorrected/colData_s4c.txt",x = s4c,row.names = TRUE,col.names = TRUE,sep= "\t")

# write.table(file = "D:/PhD/Projects/prsa/BatchCorrected/colData_s4c.txt",x = s4c,row.names = TRUE,col.names = TRUE,sep = "\t")
write.table(file = "/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/colData_s4c.txt",x = s4c,row.names = TRUE,col.names = TRUE,sep = "\t")

s4c
#head(matrix_counts)
#dim(matrix_counts)
```
```{r}
cbf_1 <- c("#E69F00","#CC79A7","#0072B2","#999999","#009E73")
cbf_1_1 <- c("#E69F00","#CC79A7","#0072B2","#999999","#009E73","#D55E00")
myCol <- c("#CC79A7","#0072B2","#E69F00","#009E73")
myCol_2 <- c("#CC79A7","#0072B2","#E69F00","#009E73","#D55E00")
shape_values_lab <- c(1,2,4,13,11,8,0)
```

### Counts raw PCA

Raw data after mapping is stored in a Matrix format and now running to see for distributions of the samples with PCA based on svd.

```{r}
# PCAs
pca_uncorrected_obj = prcomp(matrix_counts[,name_files])

#Extract percentVar data. Calculat the PC "%"'s 

percentVar_uncorrected_obj <- pca_uncorrected_obj$sdev^2/sum(pca_uncorrected_obj$sdev^2)
#head(percentVar_uncorrected_obj)

# loading_scores <- pca_uncorrected_obj$rotation[,1]
# samples_scores <- abs(loading_scores)
# samples_score_ranked <- sort(samples_scores, decreasing = TRUE)
# top_10_samples <- names(samples_score_ranked[1:10])
# top_10_samples
# pca_uncorrected_obj$rotation[top_10_samples,]T

#pull PCA values out of the PCA object  #Save as a df
pca_uncorrected_df = as.data.frame(pca_uncorrected_obj[2]$rotation)
pca_uncorrected_df[,"Lab"] <- s4c$Lab
pca_uncorrected_df[,"Platform"] <- s4c$Platform
pca_uncorrected_df[,"Library"] <- s4c$Library
pca_uncorrected_df[,"Condition"] <- s4c$Condition
pca_uncorrected_df[,"Tissue"] <- s4c$Tissue
pca_uncorrected_df[,"Extract"] <- s4c$Extract
pca_uncorrected_df[,"Length"] <- s4c$Length
pca_uncorrected_df[,"Sample"] <- rownames(s4c)
pca_raw_uncorrected = ggplot(data=pca_uncorrected_df, aes(x=PC1, y=PC2, color=Condition, shape=Lab,label=Sample))

pca_raw_uncorrected = pca_raw_uncorrected + labs(title="PCA RNA-seq raw data",
                       shape="Lab",
                       color="Condition",
                       subtitle = "raw PCA") +
  xlab(paste0("PC",substr("PC1",start = 3,stop = 3),": ",
              round(percentVar_uncorrected_obj[as.numeric(substr("PC1",start = 3,stop = 3))] * 100), "% variance")) +
  ylab(paste0("PC",substr("PC2",start = 3,stop = 3),": ",
              round(percentVar_uncorrected_obj[as.numeric(substr("PC2",start = 3,stop = 3))] * 100), "% variance"))
pca_uncorrected_df |> pull(Condition) |> unique() |> sort()
shape_values_lab <- c(1,2,4,13,11,8,0)
cat(paste0("Plotting different variables on PCA.\n"))
```


```{r}
# PCA function to add the labels from s4c
pca_var_plot_func <- function(df,var,subtitle,percentVar_obj,pca,pcb){
  n <- ggplot(data = df,aes(x = df[[pca]],
                            y = df[[pcb]],
                            color = s4c[["Condition"]],
                            shape = s4c[[var]]))
  
  #n <- n + geom_point(size=3)
  n <- n + theme(panel.grid = element_blank(),
                 panel.border = element_rect(fill = "transparent"),
                 panel.background = element_rect(fill = "transparent"),
                 legend.title = element_text(size = 15),
                 legend.text = element_text(size = 12))
  
  n <- n + labs(title="PCA of RNA-seq",
                color = "Condition",
                shape = var,
                subtitle = paste0(subtitle)) +
    xlab(paste0("PC",substr(paste0("PC",pca),start = 3,stop = 3),": ",
                round(percentVar_obj[as.numeric(substr(paste0("PC",pca),start = 3,stop = 3))] * 100), "% variance")) +
    ylab(paste0("PC",substr(paste0("PC",pcb),start = 3,stop = 3),": ",
                round(percentVar_obj[as.numeric(substr(paste0("PC",pcb),start = 3,stop = 3))] * 100), "% variance"))
  
  #n <- n + scale_color_brewer(palette = "Paired") #Accent #Dark2
  n <- n + scale_color_manual(values=cbf_1)
  n <- n + scale_shape_manual(values=shape_values_lab)
  #n <- n + geom_jitter(width = 0.01,height = 0.01,size=3)
  n <- n + geom_jitter(position = position_jitter(width = 0.000000001,height = 0.000000001,seed = 123456),size=8)# + xlim(0,0.5) + ylim(0,0.5)
  n}
```
###PCA of raw data gathering.

```{r,fig.width=22,fig.height=32}
p_raw_Lab <- pca_var_plot_func(df = pca_uncorrected_df,var = "Lab",subtitle = "Raw-Data Lab",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 2)
p_raw_Plt <- pca_var_plot_func(df = pca_uncorrected_df,var = "Platform",subtitle = "Raw-Data Platform",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 2)
p_raw_Lib <- pca_var_plot_func(df = pca_uncorrected_df,var = "Library",subtitle = "Raw-Data Library",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 2)
p_raw_Ext <- pca_var_plot_func(df = pca_uncorrected_df,var = "Extract",subtitle = "Raw-Data Extract",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 2)
p_raw_LabLib <- pca_var_plot_func(df = pca_uncorrected_df,var = "LabLib",subtitle = "Raw-Data LabLib",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 2)
p_raw_Len <- pca_var_plot_func(df = pca_uncorrected_df,var = "Length",subtitle = "Raw-Data Length",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 2)
p_raw_Tissue <- pca_var_plot_func(df = pca_uncorrected_df,var = "Tissue",subtitle = "Raw-Data Tissue",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 2)
  
  
#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa//edgeR/",modell_name,"/Graphs/PCA_edgeR_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),plot = grid.arrange(p_raw_Lab,p_raw_Plt,p_raw_Lib,p_raw_Ext,p_raw_LabLib,p_raw_Len),width = 15,  height = 17)
#ggsave(filename = paste0("C:/Users/marius/Documents/Phd/Projects/Graphs/PCA/RAW/PCA_Raw_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/RAW/PCA_Raw_Counts.svg"),
ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/RAW/PCA_Raw_Counts.svg"),
       device = "svg",
       plot = grid.arrange(p_raw_Lab,
                           p_raw_Plt,
                           p_raw_Lib,
                           p_raw_Ext,
                           p_raw_LabLib,
                           p_raw_Len,
                           p_raw_Tissue,ncol=2),
       dpi = 320,
       width = 22,
       height = 32)

grid.arrange(p_raw_Lab,
             p_raw_Plt,
             p_raw_Lib,
             p_raw_Ext,
             p_raw_LabLib,
             p_raw_Len,
             p_raw_Tissue,ncol=2)
```

```{r}
pca_raw_pc1_pc2_list <- list("Lab" = p_raw_Lab,
                             "Platform" = p_raw_Plt,
                             "Library" = p_raw_Lib,
                             "Extraction" = p_raw_Ext,
                             "LabLib" = p_raw_LabLib,
                             "Length" = p_raw_Len,
                             "Tissue" = p_raw_Tissue)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_raw_pc1_pc2_list),function(plotIt){
  #ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/RAW/PCA_RAW_",plotIt,"_Counts.svg"),plot = pca_raw_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/RAW/PCA_RAW_",plotIt,"_Counts.svg"),plot = pca_raw_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```




### PC1,PC3
```{r,fig.width=22,fig.height=32}
p_raw_Lab_pc1_pc3 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Lab",subtitle = "Raw-Data Lab",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 3)
p_raw_Plt_pc1_pc3 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Platform",subtitle = "Raw-Data Platform",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 3)
p_raw_Lib_pc1_pc3 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Library",subtitle = "Raw-Data Library",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 3)
p_raw_Ext_pc1_pc3 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Extract",subtitle = "Raw-Data Extract",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 3)
p_raw_LabLib_pc1_pc3 <- pca_var_plot_func(df = pca_uncorrected_df,var = "LabLib",subtitle = "Raw-Data LabLib",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 3)
p_raw_Len_pc1_pc3 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Length",subtitle = "Raw-Data Length",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 3)
p_raw_Tissue_pc1_pc3 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Tissue",subtitle = "Raw-Data Tissue",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 3)
  
  
#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa//edgeR/",modell_name,"/Graphs/PCA_edgeR_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),plot = grid.arrange(p_raw_Lab,p_raw_Plt,p_raw_Lib,p_raw_Ext,p_raw_LabLib,p_raw_Len),width = 15,  height = 17)
#ggsave(filename = paste0("C:/Users/marius/Documents/Phd/Projects/Graphs/PCA/RAW/PCA_Raw_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/RAW/PCA_Raw_Counts_PC1_PC3.svg"),
ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/RAW/PC1PC3/PCA_Raw_Counts_PC1_PC3.svg"),
       device = "svg",
       plot = grid.arrange(p_raw_Lab_pc1_pc3,
                           p_raw_Plt_pc1_pc3,
                           p_raw_Lib_pc1_pc3,
                           p_raw_Ext_pc1_pc3,
                           p_raw_LabLib_pc1_pc3,
                           p_raw_Len_pc1_pc3,
                           p_raw_Tissue_pc1_pc3,
                           ncol=2),
       dpi = 320,
       width = 22,
       height = 32)

grid.arrange(p_raw_Lab_pc1_pc3,
             p_raw_Plt_pc1_pc3,
             p_raw_Lib_pc1_pc3,
             p_raw_Ext_pc1_pc3,
             p_raw_LabLib_pc1_pc3,
             p_raw_Len_pc1_pc3,
             p_raw_Tissue_pc1_pc3,
             ncol=2)
```

```{r}
#dir.create("D:/PhD/Writing/Graphs_f/PCA/RAW/PC1PC3")
pca_raw_pc1_pc3_list <- list("Lab" = p_raw_Lab_pc1_pc3,
                             "Platform" = p_raw_Plt_pc1_pc3,
                             "Library" = p_raw_Lib_pc1_pc3,
                             "Extraction" = p_raw_Ext_pc1_pc3,
                             "LabLib" = p_raw_LabLib_pc1_pc3,
                             "Length" = p_raw_Len_pc1_pc3,
                             "Tissue" = p_raw_Tissue_pc1_pc3)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_raw_pc1_pc3_list),function(plotIt){
  #ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/RAW/PC1PC3/PCA_RAW_",plotIt,"pc1_pc3_Counts.svg"),plot = pca_raw_pc1_pc3_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/RAW/PC1PC3/PCA_RAW_",plotIt,"pc1_pc3_Counts.svg"),plot = pca_raw_pc1_pc3_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
})
```

### PC1,PC4
```{r,fig.width=22,fig.height=32}
p_raw_Lab_pc1_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Lab",subtitle = "Raw-Data Lab",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 4)
p_raw_Plt_pc1_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Platform",subtitle = "Raw-Data Platform",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 4)
p_raw_Lib_pc1_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Library",subtitle = "Raw-Data Library",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 4)
p_raw_Ext_pc1_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Extract",subtitle = "Raw-Data Extract",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 4)
p_raw_LabLib_pc1_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "LabLib",subtitle = "Raw-Data LabLib",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 4)
p_raw_Len_pc1_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Length",subtitle = "Raw-Data Length",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 4)
p_raw_Tissue_pc1_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Tissue",subtitle = "Raw-Data Tissue",percentVar_obj = percentVar_uncorrected_obj,pca = 1,pcb = 4)
  
  
#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa//edgeR/",modell_name,"/Graphs/PCA_edgeR_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),plot = grid.arrange(p_raw_Lab,p_raw_Plt,p_raw_Lib,p_raw_Ext,p_raw_LabLib,p_raw_Len),width = 15,  height = 17)
#ggsave(filename = paste0("C:/Users/marius/Documents/Phd/Projects/Graphs/PCA/RAW/PCA_Raw_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/RAW/PCA_Raw_Counts_PC1_PC4.svg"),
ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/RAW/PC1PC4/PCA_Raw_Counts_PC1_PC4.svg"),
       device = "svg",
       plot = grid.arrange(p_raw_Lab_pc1_pc4,
                           p_raw_Plt_pc1_pc4,
                           p_raw_Lib_pc1_pc4,
                           p_raw_Ext_pc1_pc4,
                           p_raw_LabLib_pc1_pc4,
                           p_raw_Len_pc1_pc4,
                           p_raw_Tissue_pc1_pc4,
                           ncol=2),
       dpi = 320,
       width = 22,
       height = 32)

grid.arrange(p_raw_Lab_pc1_pc4,
             p_raw_Plt_pc1_pc4,
             p_raw_Lib_pc1_pc4,
             p_raw_Ext_pc1_pc4,
             p_raw_LabLib_pc1_pc4,
             p_raw_Len_pc1_pc4,
             p_raw_Tissue_pc1_pc4,
             ncol=2)
```

```{r}
#dir.create("D:/PhD/Writing/Graphs_f/PCA/RAW/PC1PC4")
pca_raw_pc1_pc4_list <- list("Lab" = p_raw_Lab_pc1_pc4,
                             "Platform" = p_raw_Plt_pc1_pc4,
                             "Library" = p_raw_Lib_pc1_pc4,
                             "Extraction" = p_raw_Ext_pc1_pc4,
                             "LabLib" = p_raw_LabLib_pc1_pc4,
                             "Length" = p_raw_Len_pc1_pc4,
                             "Tissue" = p_raw_Tissue_pc1_pc4)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_raw_pc1_pc4_list),function(plotIt){
  #ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/RAW/PC1PC4/PCA_RAW_",plotIt,"pc1_pc4_Counts.svg"),plot = pca_raw_pc1_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/RAW/PC1PC4/PCA_RAW_",plotIt,"pc1_pc4_Counts.svg"),plot = pca_raw_pc1_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```


### PC3,PC4
```{r,fig.width=22,fig.height=32}

p_raw_Lab_pc3_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Lab",subtitle = "Raw-Data Lab",percentVar_obj = percentVar_uncorrected_obj,pca = 3,pcb = 4)
p_raw_Plt_pc3_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Platform",subtitle = "Raw-Data Platform",percentVar_obj = percentVar_uncorrected_obj,pca = 3,pcb = 4)
p_raw_Lib_pc3_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Library",subtitle = "Raw-Data Library",percentVar_obj = percentVar_uncorrected_obj,pca = 3,pcb = 4)
p_raw_Ext_pc3_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Extract",subtitle = "Raw-Data Extract",percentVar_obj = percentVar_uncorrected_obj,pca = 3,pcb = 4)
p_raw_LabLib_pc3_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "LabLib",subtitle = "Raw-Data LabLib",percentVar_obj = percentVar_uncorrected_obj,pca = 3,pcb = 4)
p_raw_Len_pc3_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Length",subtitle = "Raw-Data Length",percentVar_obj = percentVar_uncorrected_obj,pca = 3,pcb = 4)
p_raw_Tissue_pc3_pc4 <- pca_var_plot_func(df = pca_uncorrected_df,var = "Tissue",subtitle = "Raw-Data Tissue",percentVar_obj = percentVar_uncorrected_obj,pca = 3,pcb = 4)
  
  
#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa//edgeR/",modell_name,"/Graphs/PCA_edgeR_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),plot = grid.arrange(p_raw_Lab,p_raw_Plt,p_raw_Lib,p_raw_Ext,p_raw_LabLib,p_raw_Len),width = 15,  height = 17)
#ggsave(filename = paste0("C:/Users/marius/Documents/Phd/Projects/Graphs/PCA/RAW/PCA_Raw_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/RAW/PCA_Raw_Counts_PC3_PC4.svg"),
ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/RAW/PC3PC4/PCA_Raw_Counts_PC3_PC4.svg"),
       device = "svg",
       plot = grid.arrange(p_raw_Lab_pc3_pc4,
                           p_raw_Plt_pc3_pc4,
                           p_raw_Lib_pc3_pc4,
                           p_raw_Ext_pc3_pc4,
                           p_raw_LabLib_pc3_pc4,
                           p_raw_Len_pc3_pc4,
                           p_raw_Tissue_pc3_pc4,
                           ncol=2),
       dpi = 320,
       width = 22,
       height = 32)

grid.arrange(p_raw_Lab_pc3_pc4,
             p_raw_Plt_pc3_pc4,
             p_raw_Lib_pc3_pc4,
             p_raw_Ext_pc3_pc4,
             p_raw_LabLib_pc3_pc4,
             p_raw_Len_pc3_pc4,
             p_raw_Tissue_pc3_pc4,
             ncol=2)
```

```{r}
#dir.create("D:/PhD/Writing/Graphs_f/PCA/RAW/PC3PC4")
pca_raw_pc3_pc4_list <- list("Lab" = p_raw_Lab_pc3_pc4,
                             "Platform" = p_raw_Plt_pc3_pc4,
                             "Library" = p_raw_Lib_pc3_pc4,
                             "Extraction" = p_raw_Ext_pc3_pc4,
                             "LabLib" = p_raw_LabLib_pc3_pc4,
                             "Length" = p_raw_Len_pc3_pc4,
                             "Tissue" = p_raw_Tissue_pc3_pc4)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_raw_pc3_pc4_list),function(plotIt){
  #ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/RAW/PC3PC4/PCA_RAW_",plotIt,"pc3_pc4_Counts.svg"),plot = pca_raw_pc3_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/RAW/PC3PC4/PCA_RAW_",plotIt,"pc3_pc4_Counts.svg"),plot = pca_raw_pc3_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```

### UMAP function
```{r,fig.width=9,fig.height=11}
umap_func_by_filtered_expression_normalized <- function(df,n,subtitle){
  cat(paste0("Raw data dim: ",dim(df)))
  
  df_f <- df[rowSums(df) >= n,]
  
  cat(paste0("Filtered by ",n," data dim: ",dim(df_f)))
  hp_filtered <- hist(log(df_f),xlim = c(0,20),breaks = 100,col=scales::alpha('skyblue',.8),border=F)
  
  #transpose and calculate umap
  df_f <- df_f |> t()
  umap_df_f <- umap::umap(d = df_f,
                          preserve.seed = seed_it)
  
  #merge and extract
  umap_df_f <- as.data.frame(umap_df_f$layout)
  #rownames(umap_batch_norm_platform_layout_df)
  #rownames(s4c)
  umap_df_f_m <- merge(x = umap_df_f,
                       y = s4c,
                       by.x = "row.names",
                       by.y= "row.names")
  #Plot
  n <- umap_df_f_m |> ggplot(aes(x = V1,y = V2)) +
    geom_jitter(aes(color=Condition,
                    shape= Lab),
                position = position_jitter(width = 0.01,height = 0.01,seed = seed_it),
                size=8)+ # + xlim(0,0.5) + ylim(0,0.5)
    scale_color_manual(values=cbf_1) + 
    scale_shape_manual(values=shape_values_lab) +
    theme(panel.grid = element_blank(),
          panel.border = element_rect(fill = "transparent"),
          panel.background = element_rect(fill = "transparent"),
          legend.title = element_text(size = 15),
          legend.text = element_text(size = 12)) +
    
    labs(title="UMAP of RNA-seq",
         color = "Condition",
         subtitle = paste0(subtitle,""))
  
  #n <- n + geom_text(aes(label=umap_df_f_m$Row.names))
  
  
  return(list(umap_plot = n,
              hist_plot = hp_filtered))
  }
```

### UMAP Raw Counts
```{r,fig.height=8,fig.width=8}
#dir.create("D:/PhD/Writing/Graphs_f/UMAP/RAW/",recursive = TRUE)
umap_raw_counts <- umap_func_by_filtered_expression_normalized(df = matrix_counts,n = 0,subtitle = "Raw Counts")
umap_raw_counts$umap_plot

#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/RAW/UMAP_Raw_Counts.svg"),
#ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/RAW/UMAP_Raw_Counts.svg"),
#ggsave(filename = "D:/PhD/Writing/Graphs_f/UMAP/RAW/UMAP_Raw_Counts.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/RAW/UMAP_Raw_Counts.svg",
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_raw_counts$umap_plot),
       width = 12,
       height = 10)
```
##### Raw Counts
```{r}
head(matrix_counts)
dim(matrix_counts)
```



##### Batch Groups
```{r,echo=TRUE}
s4c[,"LenPlt"] <- paste0(s4c$Length,"_",s4c$Platform)
s4c[,"LenLib"] <- paste0(s4c$Length,"_",s4c$Library)
s4c[,"LibPlt"] <- paste0(s4c$Library,"_",s4c$Platform)
s4c[,"LibTiss"] <- paste0(s4c$Library,"_",s4c$Tissue)
s4c[,"PltTiss"] <- paste0(s4c$Platform,"_",s4c$Tissue)
s4c[,"PltLibTiss"] <- paste0(s4c$Platform,"_",s4c$Library,"_",s4c$Tissue)
s4c[,"PltTissLab"] <- paste0(s4c$Platform,"_",s4c$Tissue,"_",s4c$Lab)
s4c[,"PltLab"] <- paste0(s4c$Platform,"_",s4c$Lab)
s4c[,"PltLabLib"] <- paste0(s4c$Platform,"_",s4c$Lab,"_",s4c$Library)
s4c[,"PltExt"] <- paste0(s4c$Platform,"_",s4c$Extract)
s4c[,"PltExtLib"] <- paste0(s4c$Platform,"_",s4c$Extract,"_",s4c$Library)
s4c[,"PltExtTiss"] <- paste0(s4c$Platform,"_",s4c$Extract,"_",s4c$Tissue)
s4c[,"LabLib"] <- paste0(s4c$Lab,"_",s4c$Library)
s4c[,"LabLen"] <- paste0(s4c$Lab,"_",s4c$Length)
s4c[,"LabTiss"] <- paste0(s4c$Lab,"_",s4c$Tissue)
s4c[,"LabExtTissLibLen"] <- paste0(s4c$Lab,"_",s4c$Extract,"_",s4c$Tissue,"_",s4c$Library,"_",s4c$Length)
s4c[,"LenExt"] <- paste0(s4c$Length,"_",s4c$Extract)
s4c[,"PltExt"] <- paste0(s4c$Platform,"_",s4c$Extract)
s4c[,"TissExt"] <- paste0(s4c$Tissue,"_",s4c$Extract)
s4c[,"LibExt"] <- paste0(s4c$Library,"_",s4c$Extract)
s4c[,"LabExt"] <- paste0(s4c$Lab,"_",s4c$Extract)


lapply(s4c,unique)
```

### Batch correction Platform

```{r}
groups = sapply(as.character(s4c$Condition), switch,
                "Ablated" = 1,
                "Amputation" = 2,
                "Cryoinjury" = 3,
                "Sham" = 4,
                "Uninjured" = 5,
                USE.NAMES = F)

#batches_Library = sapply(as.character(s4c$Library),switch,"Single" = 1,"Paired" = 2,USE.NAMES = FALSE)
#batches_Length = sapply(as.character(s4c$Length), switch, "50" = 1, "75" = 2, "120" = 3, "150" = 4, USE.NAMES = FALSE)
batches_Platform = sapply(as.character(s4c$Platform), switch, "GenomeAnalyzerII" = 1, "NextSeq500" = 2, "BGISEQ500" = 3,
                          "HiSeq2000" = 4, "NovaSeq6000" = 5, "HiSeqXTen" = 6, USE.NAMES = FALSE)

corrected_data_Platform = sva::ComBat_seq(counts = as.matrix(matrix_counts[,name_files]), batch = batches_Platform, group = groups)
```


### Counts Platform Batch corrected PCA
**Check out the percent corresponds to the correction.**

```{r}
#Calculate the PCA for corrected data
pca_corrected_obj_Platform = prcomp(corrected_data_Platform[,name_files])

percentVar_corrected_obj_Platform <- pca_corrected_obj_Platform$sdev^2/sum(pca_corrected_obj_Platform$sdev^2)

pca_corrected_Platform_df = as.data.frame(pca_corrected_obj_Platform[2]$rotation)
#Add labels
pca_corrected_Platform_df[,"Lab"] <- s4c$Lab
pca_corrected_Platform_df[,"Platform"] <- s4c$Platform
pca_corrected_Platform_df[,"Library"] <- s4c$Library
pca_corrected_Platform_df[,"Condition"] <- s4c$Condition
pca_corrected_Platform_df[,"Tissue"] <- s4c$Tissue
pca_corrected_Platform_df["LenExt"] <- s4c$LenExt
pca_corrected_Platform_df["PltExt"] <- s4c$PltExt
pca_corrected_Platform_df["TissExt"] <- s4c$TissExt
pca_corrected_Platform_df["LabLib"] <- s4c$LabLib
pca_corrected_Platform_df["Extract"] <- s4c$Extract
```


```{r}
pca_var_plot_func <- function(df,var,subtitle,percentVar_obj){
  n <- ggplot(data = df,aes(x = PC1,
                            y = PC2,
                            color=s4c[["Condition"]],
                            shape=s4c[[var]]))
  n <- n + theme(panel.grid = element_blank(),
                 panel.border = element_rect(fill = "transparent"),
                 panel.background = element_rect(fill = "transparent"),
                 legend.title = element_text(size=15),
                 legend.text = element_text(size=12))
  n <- n + labs(title="PCA of RNA-seq",
                color = "Condition",
                shape = var,
                subtitle = paste0(subtitle,"")) +
    xlab(paste0("PC",substr("PC1",start = 3,stop = 3),": ",
                round(percentVar_obj[as.numeric(substr("PC1",start = 3,stop = 3))] * 100), "% variance")) +
    ylab(paste0("PC",substr("PC2",start = 3,stop = 3),": ",
                round(percentVar_obj[as.numeric(substr("PC2",start = 3,stop = 3))] * 100), "% variance"))
  #n <- n + scale_color_brewer(palette = "Paired") #Accent #Dark2
  n <- n + scale_color_manual(values=cbf_1)
  n <- n + scale_shape_manual(values=shape_values_lab)
  n <- n + geom_jitter(position = position_jitter(width = 0.01,height = 0.01,seed = 123456),size=8)
  n}
```




```{r,fig.width=22,fig.height=32,echo=TRUE}
#dir.create(recursive = TRUE,path = "../BatchCorrected/CombatSeq/Graphs/Platform")

p_Platform_Platform <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Platform",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform)
p_Platform_Len <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Length",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform)
p_Platform_Lib <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Library",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform)
p_Platform_LabLib <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "LabLib",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform)
p_Platform_Ext <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Extract",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform)
p_Platform_Lab <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Lab",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform)
p_Platform_Tissue <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Tissue",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform)


#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/CombatSeq/Graphs/Platform/PCA_combatseq_Platform_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/Platform/PCA_combatseq_Platform_batch_corrected_Counts.svg"),
       device = "svg",
       dpi = 320,
       plot = grid.arrange(p_Platform_Platform,
                           p_Platform_Len,
                           p_Platform_Lib,
                           p_Platform_LabLib,
                           p_Platform_Ext,
                           p_Platform_Lab,
                           p_Platform_Tissue,ncol=2),
       width = 22,
       height = 32)

grid.arrange(p_Platform_Platform,
             p_Platform_Len,
             p_Platform_Lib,
             p_Platform_LabLib,
             p_Platform_Ext,
             p_Platform_Lab,
             p_Platform_Tissue,ncol=2)
```

```{r}
#dir.create(recursive = TRUE,path = "D:/PhD/Writing/Graphs_f/PCA/BATCH/")
pca_batch_pc1_pc2_list <- list("Platform" = p_Platform_Platform,
                               "Lab" = p_Platform_Lab,
                               "Library" = p_Platform_Lib,
                               "Extraction" = p_Platform_Ext,
                               "LabLib" = p_Platform_LabLib,
                               "Length" = p_Platform_Len,
                               "Tissue" = p_Platform_Tissue)
#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_batch_pc1_pc2_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/BATCH/PCA_BATCH_",plotIt,"pc1_pc2_Counts.svg"),plot = pca_batch_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PCA_BATCH_",plotIt,"pc1_pc2_Counts.svg"),plot = pca_batch_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```

### UMAP Platform Batch Corrected
```{r,fig.height=9,fig.width=9}
umap_platform_batch_corrected <- umap_func_by_filtered_expression_normalized(df = corrected_data_Platform,n = 0,subtitle = "Batch Platform")
umap_platform_batch_corrected$umap_plot

#dir.create("D:/PhD/Writing/Graphs_f/UMAP/BATCH")
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/UMAP_combatseq_Platform_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/UMAP_combatseq_Platform_batch_corrected_Counts.svg"),
#ggsave(filename = "D:/PhD/Writing/Graphs_f/UMAP/BATCH/UMAP_combatseq_Platform_batch_corrected_Counts.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/BATCH/UMAP_combatseq_Platform_batch_corrected_Counts.svg",
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_platform_batch_corrected$umap_plot),
       width = 12,
       height = 10)

```


### Check PC 1 and 3

```{r,fig.width=22,fig.height=32}
#dir.create(recursive = TRUE,path = "../BatchCorrected/CombatSeq/Graphs/Platform")

p_Platform_Len_pc1_3 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Length",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 3)
p_Platform_Platform_pc1_3 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Platform",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 3)
p_Platform_Lib_pc1_3 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Library",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 3)
p_Platform_LabLib_pc1_3 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "LabLib",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 3)
p_Platform_Ext_pc1_3 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Extract",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 3)
p_Platform_Lab_pc1_3 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Lab",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 3)
p_Platform_Tissue_pc1_3 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Tissue",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 3)


#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/CombatSeq/Graphs/Platform/PCA_combatseq_Platform_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts_PC1_PC3.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts_PC1_PC3.svg"),
#dir.create("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/")
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PC1PC3/PCA_combatseq_Platform_batch_corrected_Counts_PC1_PC3.svg"),
       device = "svg",
       dpi = 320,
       plot = grid.arrange(p_Platform_Platform_pc1_3,
                           p_Platform_Len_pc1_3,
                           p_Platform_Lib_pc1_3,
                           p_Platform_LabLib_pc1_3,
                           p_Platform_Ext_pc1_3,
                           p_Platform_Lab_pc1_3,
                           p_Platform_Tissue_pc1_3,ncol=2),
       width = 22,
       height = 32)

grid.arrange(p_Platform_Platform_pc1_3,
             p_Platform_Len_pc1_3,
             p_Platform_Lib_pc1_3,
             p_Platform_LabLib_pc1_3,
             p_Platform_Ext_pc1_3,   
             p_Platform_Lab_pc1_3,
             p_Platform_Tissue_pc1_3,ncol=2)
```


```{r}
#dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PC1PC3")
pca_batch_pc1_pc3_list <- list("Platform" = p_Platform_Platform_pc1_3,
                               "Lab" = p_Platform_Lab_pc1_3,
                               "Library" = p_Platform_Lib_pc1_3,
                               "Extraction" = p_Platform_Ext_pc1_3,
                               "LabLib" = p_Platform_LabLib_pc1_3,
                               "Length" = p_Platform_Len_pc1_3,
                               "Tissue" = p_Platform_Tissue_pc1_3)
#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_batch_pc1_pc3_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/BATCH/PC1PC3/PCA_BATCH_",plotIt,"pc1_pc3_Counts.svg"),plot = pca_batch_pc1_pc3_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PC1PC3/PCA_BATCH_",plotIt,"pc1_pc3_Counts.svg"),plot = pca_batch_pc1_pc3_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
})
```

###PC1,4

```{r,fig.width=22,fig.height=32}

p_Platform_Platform_pc1_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Platform",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 4)
p_Platform_Len_pc1_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Length",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 4)
p_Platform_Lib_pc1_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Library",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 4)
p_Platform_LabLib_pc1_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "LabLib",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 4)
p_Platform_Ext_pc1_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Extract",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 4)
p_Platform_Lab_pc1_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Lab",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 4)
p_Platform_Tissue_pc1_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Tissue",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 1,pcb = 4)


#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/CombatSeq/Graphs/Platform/PCA_combatseq_Platform_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts_PC1_PC4.svg"),
# ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts_PC1_PC4.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PC1PC4/PCA_combatseq_Platform_batch_corrected_Counts_PC1_PC4.svg"),device = "svg",
       dpi = 320,
       plot = grid.arrange(p_Platform_Platform_pc1_4,
                           p_Platform_Len_pc1_4,
                           p_Platform_Lib_pc1_4,
                           p_Platform_LabLib_pc1_4,
                           p_Platform_Ext_pc1_4,
                           p_Platform_Lab_pc1_4,
                           p_Platform_Tissue_pc1_4,ncol=2),
       width = 22,
       height = 32)

grid.arrange(p_Platform_Platform_pc1_4,
             p_Platform_Len_pc1_4,
             p_Platform_Lib_pc1_4,
             p_Platform_LabLib_pc1_4,
             p_Platform_Ext_pc1_4,   
             p_Platform_Lab_pc1_4,
             p_Platform_Tissue_pc1_4,ncol=2)
```
```{r}
#dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PC1PC4")
pca_batch_pc1_pc4_list <- list("Platform" = p_Platform_Platform_pc1_4,
                               "Lab" = p_Platform_Lab_pc1_4,
                               "Library" = p_Platform_Lib_pc1_4,
                               "Extraction" = p_Platform_Ext_pc1_4,
                               "LabLib" = p_Platform_LabLib_pc1_4,
                               "Length" = p_Platform_Len_pc1_4,
                               "Tissue" = p_Platform_Tissue_pc1_4)
#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_batch_pc1_pc4_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/BATCH/PC1PC4/PCA_BATCH_",plotIt,"pc1_pc4_Counts.svg"),plot = pca_batch_pc1_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PC1PC4/PCA_BATCH_",plotIt,"pc1_pc4_Counts.svg"),plot = pca_batch_pc1_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```

###PC3,4

```{r,fig.width=22,fig.height=32}
p_Platform_Platform_pc3_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Platform",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 3,pcb = 4)
p_Platform_Len_pc3_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Length",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 3,pcb = 4)
p_Platform_Lib_pc3_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Library",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 3,pcb = 4)
p_Platform_LabLib_pc3_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "LabLib",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 3,pcb = 4)
p_Platform_Ext_pc3_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Extract",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 3,pcb = 4)
p_Platform_Lab_pc3_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Lab",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 3,pcb = 4)
p_Platform_Tissue_pc3_4 <- pca_var_plot_func(df = pca_corrected_Platform_df,var = "Tissue",subtitle = "Batch Corrected for Platform",percentVar_obj = percentVar_corrected_obj_Platform,pca = 3,pcb = 4)


#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/CombatSeq/Graphs/Platform/PCA_combatseq_Platform_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts_PC3_PC4.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts_PC3_PC4.svg"),
# ggsave(filename = "D:/PhD/Writing/Graphs_f/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts_PC3_PC4.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PCA_combatseq_Platform_batch_corrected_Counts_PC3_PC4.svg",
       device = "svg",
       dpi = 320,
       plot = grid.arrange(p_Platform_Platform_pc3_4,
                           p_Platform_Len_pc3_4,
                           p_Platform_Lib_pc3_4,
                           p_Platform_LabLib_pc3_4,
                           p_Platform_Ext_pc3_4,
                           p_Platform_Lab_pc3_4,
                           p_Platform_Tissue_pc3_4,ncol=2),
       width = 22,
       height = 32)

grid.arrange(p_Platform_Platform_pc3_4,
             p_Platform_Len_pc3_4,
             p_Platform_Lib_pc3_4,
             p_Platform_LabLib_pc3_4,
             p_Platform_Ext_pc3_4,   
             p_Platform_Lab_pc3_4,
             p_Platform_Tissue_pc3_4,ncol=2)
```


```{r}
#dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PC3PC4")
pca_batch_pc3_pc4_list <- list("Platform" = p_Platform_Platform_pc3_4,
                               "Lab" = p_Platform_Lab_pc3_4,
                               "Library" = p_Platform_Lib_pc3_4,
                               "Extraction" = p_Platform_Ext_pc3_4,
                               "LabLib" = p_Platform_LabLib_pc3_4,
                               "Length" = p_Platform_Len_pc3_4,
                               "Tissue" = p_Platform_Tissue_pc3_4)
#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_batch_pc3_pc4_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/BATCH/PC3PC4/PCA_BATCH_",plotIt,"pc3_pc4_Counts.svg"),plot = pca_batch_pc3_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PC3PC4/PCA_BATCH_",plotIt,"pc3_pc4_Counts.svg"),plot = pca_batch_pc3_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
})
```


### Counts Library Batch corrected PCA

```{r}
groups = sapply(as.character(s4c$Condition), switch,
                "Ablated" = 1,
                "Amputation" = 2,
                "Cryoinjury" = 3,
                "Sham" = 4,
                "Uninjured" = 5,
                USE.NAMES = F)
batches_Library = sapply(as.character(s4c$Library),switch,"Single" = 1,"Paired" = 2,USE.NAMES = FALSE)
#batches_Library

corrected_data_Library = ComBat_seq(counts = as.matrix(matrix_counts[,name_files]), batch = batches_Library, group = groups)
#corrected_data = cbind(uncorrected_data[,c("Gene","Chr")], corrected_data)
```


```{r}
#Calculate the PCA for corrected data
pca_corrected_obj_Library = prcomp(corrected_data_Library[,name_files])
#pca_corrected_obj it seems that prcomp rotates the samples.....
#pull PCA values out of the PCA object
percentVar_corrected_obj_Library <- pca_corrected_obj_Library$sdev^2/sum(pca_corrected_obj_Library$sdev^2)
#percentVar_corrected_obj
pca_corrected_Library_df = as.data.frame(pca_corrected_obj_Library[2]$rotation)
#head(pca_corrected)

#Add labels
pca_corrected_Library_df[,"Lab"] <- s4c$Lab
pca_corrected_Library_df[,"Platform"] <- s4c$Platform
pca_corrected_Library_df[,"Library"] <- s4c$Library
pca_corrected_Library_df[,"Condition"] <- s4c$Condition
pca_corrected_Library_df[,"Tissue"] <- s4c$Tissue
pca_corrected_Library_df[,"Extract"] <- s4c$Extract
pca_corrected_Library_df[,"Length"] <- s4c$Length
pca_corrected_Library_df[,"Sample"] <- rownames(s4c)
pca_corrected_Library_df["LenExt"] <- s4c$LenExt
pca_corrected_Library_df["PltExt"] <- s4c$PltExt
pca_corrected_Library_df["TissExt"] <- s4c$TissExt
pca_corrected_Library_df["LibExt"] <- s4c$LibExt
pca_corrected_Library_df["LabLen"] <- s4c$LabLen
```



```{r,fig.width=22,fig.height=32,echo=TRUE}
#dir.create(recursive = TRUE,path = "../BatchCorrected/CombatSeq/Graphs/Library")
dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/Library")
p_Library_Lib <- pca_var_plot_func(df = pca_corrected_Library_df,var = "Library",subtitle = "Batch Corrected for Library",percentVar_obj = percentVar_corrected_obj_Library,pca = 1,pcb = 2)
p_Library_Plt <- pca_var_plot_func(df = pca_corrected_Library_df,var = "Platform",subtitle = "Batch Corrected for Library",percentVar_obj = percentVar_corrected_obj_Library,pca = 1,pcb = 2)
p_Library_Len <- pca_var_plot_func(df = pca_corrected_Library_df,var = "Length",subtitle = "Batch Corrected for Library",percentVar_obj = percentVar_corrected_obj_Library,pca = 1,pcb = 2)
p_Library_LabLib <- pca_var_plot_func(df = pca_corrected_Library_df,var = "LabLib",subtitle = "Batch Corrected for Library",percentVar_obj = percentVar_corrected_obj_Library,pca = 1,pcb = 2)
p_Library_Ext <- pca_var_plot_func(df = pca_corrected_Library_df,var = "Extract",subtitle = "Batch Corrected for Library",percentVar_obj = percentVar_corrected_obj_Library,pca = 1,pcb = 2)
p_Library_Lab <- pca_var_plot_func(df = pca_corrected_Library_df,var = "Lab",subtitle = "Batch Corrected for Library",percentVar_obj = percentVar_corrected_obj_Library,pca = 1,pcb = 2)
p_Library_Tissue <- pca_var_plot_func(df = pca_corrected_Library_df,var = "Tissue",subtitle = "Batch Corrected for Library",percentVar_obj = percentVar_corrected_obj_Library,pca = 1,pcb = 2)

#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/CombatSeq/Graphs/Library/PCA_combatseq_Library_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Library_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Library_batch_corrected_Counts.svg"),
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/Library/PCA_combatseq_Library_batch_corrected_Counts.svg",
       device = "svg",
       dpi = 320,
       plot = grid.arrange(p_Library_Lib,
                           p_Library_Lab,
                           p_Library_Ext,
                           p_Library_Len,
                           p_Library_LabLib,
                           p_Library_Plt,
                           p_Library_Tissue,
                           ncol=2),
       width = 22,
       height = 32)
grid.arrange(p_Library_Lib,p_Library_Lab,p_Library_Ext,p_Library_Len,p_Library_LabLib,p_Library_Plt,p_Library_Tissue,ncol=2)
```

```{r}
#dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/Library/")
pca_batch_lib_pc1_pc2_list <- list("Platform" = p_Library_Plt,
                                   "Lab" = p_Library_Lab,
                                   "Library" = p_Library_Lib,
                                   "Extraction" = p_Library_Ext,
                                   "LabLib" = p_Library_LabLib,
                                   "Length" = p_Library_Len,
                                   "Tissue" = p_Library_Tissue)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_batch_lib_pc1_pc2_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/BATCH/Library/PCA_BATCH_libray",plotIt,"pc1_pc2_Counts.svg"),plot = pca_batch_lib_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/Library/PCA_BATCH_libray",plotIt,"pc1_pc2_Counts.svg"),plot = pca_batch_lib_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
})
```

### UMAP Platform Batch Corrected
```{r,fig.height=9,fig.width=9}
umap_library_batch_corrected <- umap_func_by_filtered_expression_normalized(df = corrected_data_Library,n = 0,subtitle = "Batch Library")
umap_library_batch_corrected$umap_plot

#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/UMAP_combatseq_Library_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/UMAP_combatseq_Library_batch_corrected_Counts.svg"),
# ggsave(filename = "D:/PhD/Writing/Graphs_f/UMAP/BATCH/UMAP_combatseq_Library_batch_corrected_Counts.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/BATCH/UMAP_combatseq_Library_batch_corrected_Counts.svg",
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_library_batch_corrected$umap_plot),
       width = 12,
       height = 10)
```

### Counts Length Batch corrected PCA
```{r}
groups = sapply(as.character(s4c$Condition), switch,
                "Ablated" = 1,
                "Amputation" = 2,
                "Cryoinjury" = 3,
                "Sham" = 4,
                "Uninjured" = 5,
                USE.NAMES = F)

#batches_Library = sapply(as.character(s4c$Library),switch,"Single" = 1,"Paired" = 2,USE.NAMES = FALSE)
batches_Length = sapply(as.character(s4c$Length), switch, "50" = 1, "75" = 2, "150" = 3, USE.NAMES = FALSE)
corrected_data_Length = sva::ComBat_seq(counts = as.matrix(matrix_counts[,name_files]), batch = batches_Length, group = groups)
```


```{r}
#Calculate the PCA for corrected data
pca_corrected_obj_Length = prcomp(corrected_data_Length[,name_files])

percentVar_corrected_obj_Length <- pca_corrected_obj_Length$sdev^2/sum(pca_corrected_obj_Length$sdev^2)

pca_corrected_Length_df = as.data.frame(pca_corrected_obj_Length[2]$rotation)
#Add labels
pca_corrected_Length_df[,"Lab"] <- s4c$Lab
pca_corrected_Length_df[,"Platform"] <- s4c$Platform
pca_corrected_Length_df[,"Library"] <- s4c$Library
pca_corrected_Length_df[,"Condition"] <- s4c$Condition
pca_corrected_Length_df[,"Sample"] <- rownames(s4c)
pca_corrected_Length_df["LenExt"] <- s4c$LenExt
pca_corrected_Length_df["PltExt"] <- s4c$PltExt
pca_corrected_Length_df["TissExt"] <- s4c$TissExt
pca_corrected_Length_df["LibExt"] <- s4c$LibExt
pca_corrected_Length_df["LabExt"] <- s4c$LabExt
pca_corrected_Length_df["LabLib"] <- s4c$LabLib
pca_corrected_Length_df["Extract"] <- s4c$Extract

```

##### Arrange the plots of the different variables.
```{r,fig.width=22,fig.height=32,echo=TRUE}

#dir.create(recursive = TRUE,path = "../BatchCorrected/CombatSeq/Graphs/Length")
#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/BATCH/Length/")
p_Length_Length <- pca_var_plot_func(df = pca_corrected_Length_df,var = "Length",subtitle = "Batch Corrected for Length",percentVar_obj = percentVar_corrected_obj_Length,pca = 1,pcb = 2)
p_Length_Plt <- pca_var_plot_func(df = pca_corrected_Length_df,var = "Platform",subtitle = "Batch Corrected for Length",percentVar_obj = percentVar_corrected_obj_Length,pca = 1,pcb = 2)
p_Length_Lib <- pca_var_plot_func(df = pca_corrected_Length_df,var = "Library",subtitle = "Batch Corrected for Length",percentVar_obj = percentVar_corrected_obj_Length,pca = 1,pcb = 2)
p_Length_LabLib <- pca_var_plot_func(df = pca_corrected_Length_df,var = "LabLib",subtitle = "Batch Corrected for Length",percentVar_obj = percentVar_corrected_obj_Length,pca = 1,pcb = 2)
p_Length_Ext <- pca_var_plot_func(df = pca_corrected_Length_df,var = "Extract",subtitle = "Batch Corrected for Length",percentVar_obj = percentVar_corrected_obj_Length,pca = 1,pcb = 2)
p_Length_Lab <- pca_var_plot_func(df = pca_corrected_Length_df,var = "Lab",subtitle = "Batch Corrected for Length",percentVar_obj = percentVar_corrected_obj_Length,pca = 1,pcb = 2)
p_Length_Tissue <- pca_var_plot_func(df = pca_corrected_Length_df,var = "Tissue",subtitle = "Batch Corrected for Length",percentVar_obj = percentVar_corrected_obj_Length,pca = 1,pcb = 2)

#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/CombatSeq/Graphs/Length/PCA_combatseq_Length_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Length_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Length_batch_corrected_Counts.svg"),
# ggsave(filename = "D:/PhD/Writing/Graphs_f/PCA/BATCH/Length/PCA_combatseq_Length_batch_corrected_Counts.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/Length/PCA_combatseq_Length_batch_corrected_Counts.svg",
       device = "svg",
       dpi = 320,
       plot = grid.arrange(p_Length_Length,
                           p_Length_Plt,
                           p_Length_Lib,
                           p_Length_LabLib,
                           p_Length_Ext,
                           p_Length_Lab,
                           p_Length_Tissue,ncol=2),
       width = 22,
       height = 32)
grid.arrange(p_Length_Length,p_Length_Plt,p_Length_Lib,p_Length_LabLib,p_Length_Ext,p_Length_Lab,p_Length_Tissue,ncol=2)
```

```{r}
#dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/Length/")
pca_batch_len_pc1_pc2_list <- list("Platform" = p_Length_Plt,
                                   "Lab" = p_Length_Lab,
                                   "Library" = p_Length_Lib,
                                   "Extraction" = p_Length_Ext,
                                   "LabLib" = p_Length_LabLib,
                                   "Length" = p_Length_Length,
                                   "Tissue" = p_Length_Tissue)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_batch_len_pc1_pc2_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/BATCH/Length/PCA_BATCH_length",plotIt,"pc1_pc2_Counts.svg"),plot = pca_batch_len_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/Length/PCA_BATCH_length",plotIt,"pc1_pc2_Counts.svg"),plot = pca_batch_len_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```


### Batch Condition Lab 3 Batches
```{r,fig.width=30,fig.height=12}
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Length_Platform_Library_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/PCA_combatseq_Length_Platform_Library_batch_corrected_Counts.svg"),
# ggsave(filename = "D:/PhD/Writing/Graphs_f/PCA/BATCH/PCA_combatseq_Length_Platform_Library_batch_corrected_Counts.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/BATCH/PCA_combatseq_Length_Platform_Library_batch_corrected_Counts.svg",
       device = "svg",
       dpi = 320,
       plot = grid.arrange(p_Length_Lab,
                           p_Platform_Lab,
                           p_Library_Lab,nrow = 1),
       width = 30,
       height = 12)

grid.arrange(p_Length_Lab,
             p_Platform_Lab,
             p_Library_Lab,nrow = 1)

```

### UMAP Batch Length
```{r,fig.height=8,fig.width=8}
umap_length_batch_corrected <- umap_func_by_filtered_expression_normalized(df = corrected_data_Length,n = 0,subtitle = "Batch Length")
umap_length_batch_corrected$umap_plot

#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/BATCH/UMAP_combatseq_Length_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/BATCH/UMAP_combatseq_Length_batch_corrected_Counts.svg"),
# ggsave(filename = "D:/PhD/Writing/Graphs_f/UMAP/BATCH/UMAP_combatseq_Length_batch_corrected_Counts.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/BATCH/UMAP_combatseq_Length_batch_corrected_Counts.svg",
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_length_batch_corrected$umap_plot),
       width = 12,
       height = 10)
```

### EdgeR Normalized Function

Normalize the data for the following steps working with edgeR TMM method, all of this is encapsulated in a function for running it with less mistakes.


```{r echo=TRUE}

runModellEdgeR <- function(modell_in,modell_name){

  library(data.table)
  #Run edgeR on the COMBATseq batch corrected data.

  cat("Builduing the DEG data list for edgeR.")
  #.1- Build DGE list and normalize
  y = DGEList(counts = modell_in[,name_files], group = s4c$Condition, genes = rownames(matrix_counts))
  #perform TMM normalization
  cat("Normalizing using TMM.\n")
  y_norm <- calcNormFactors(object = y,method = "TMM")

  y_normalized_corrected <- cpm(y = y_norm,normalized.lib.sizes = TRUE)


  cat(paste0("Plotting the man ",modell_name," batch corrected and normalized PCA of th model.\n"))
  # PCAs
  y_normalized_corrected_pcas <- prcomp(y_normalized_corrected)

  #calculat the PC "%"'s
  percentVar_y_normalized_corrected_pcas <- y_normalized_corrected_pcas$sdev^2/sum(y_normalized_corrected_pcas$sdev^2)

  #Save as a df
  y_normalized_corrected_pcas_df <- as.data.frame(y_normalized_corrected_pcas[2]$rotation)

  # PCA of the normalized TMM counts done, now add subsequent labels on it, to see how the variables affect it.
  # PCA function to add the labels from s4c
  cat(paste0("Plotting different variables on the ",modell_name," batch corrected and normalized model PCA.\n"))
  pca_var_plot_func <- function(df,var,subtitle){
    n <- ggplot(data = df,aes(x = PC1,
                              y = PC2,
                              color=s4c[["Condition"]],
                              shape=s4c[[var]]))
    #n <- n + geom_point(size=5)
    n <- n + labs(title="PCA of RNA-seq (batch corrected) \nNormalized edgeR",
                  color = var ,
                  shape = "Condition",
                  subtitle = paste0(subtitle," Batch Corrected")) +
      xlab(paste0("PC",substr("PC1",start = 3,stop = 3),": ",
                  round(percentVar_y_normalized_corrected_pcas[as.numeric(substr("PC1",start = 3,stop = 3))] * 100), "% variance")) +
      ylab(paste0("PC",substr("PC2",start = 3,stop = 3),": ",
                  round(percentVar_y_normalized_corrected_pcas[as.numeric(substr("PC2",start = 3,stop = 3))] * 100), "% variance"))
    n <- n + theme(panel.grid = element_blank(),
                   panel.border = element_rect(fill = "transparent"),
                   panel.background = element_rect(fill = "transparent"),
                   legend.title = element_text(size = 15),
                   legend.text = element_text(size = 12))
    #n <- n + scale_color_brewer(palette = "Paired") #Accent #Dark2
    n <- n + scale_color_manual(values=cbf_1)
    n <- n + scale_shape_manual(values=shape_values_lab)
    #n <- n + geom_jitter(width = 0.01,height = 0.01,size=3)
    n <- n + geom_jitter(position = position_jitter(width = 0.01,height = 0.01,seed = 123456),size=8)
    n}

  p_norm_corrected_Lab <- pca_var_plot_func(df = y_normalized_corrected_pcas_df,var = "Lab",subtitle = modell_name)
  p_norm_corrected_Plt <- pca_var_plot_func(df = y_normalized_corrected_pcas_df,var = "Platform",subtitle = modell_name)
  p_norm_corrected_Lib <- pca_var_plot_func(df = y_normalized_corrected_pcas_df,var = "Library",subtitle = modell_name)
  p_norm_corrected_Ext <- pca_var_plot_func(df = y_normalized_corrected_pcas_df,var = "Extract",subtitle = modell_name)
  p_norm_corrected_LabLib <- pca_var_plot_func(df = y_normalized_corrected_pcas_df,var = "LabLib",subtitle = modell_name)
  p_norm_corrected_Len <- pca_var_plot_func(df = y_normalized_corrected_pcas_df,var = "Length",subtitle = modell_name)
  p_norm_corrected_Tissue <- pca_var_plot_func(df = y_normalized_corrected_pcas_df,var = "Tissue",subtitle = modell_name)

  
  
  #ggsave(filename =paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Graphs/PCA_edgeR_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
  #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Graphs/PCA_edgeR_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
  ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Graphs/PCA_edgeR_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
         dpi = 320,
         device = "svg",
         plot = grid.arrange(p_norm_corrected_Lab,
                             p_norm_corrected_Plt,
                             p_norm_corrected_Lib,
                             p_norm_corrected_Ext,
                             p_norm_corrected_LabLib,
                             p_norm_corrected_Len,
                             p_norm_corrected_Tissue,
                             ncol=2),
         width = 22,
         height = 32)
  
  grid.arrange(p_norm_corrected_Lab,
               p_norm_corrected_Plt,
               p_norm_corrected_Lib,
               p_norm_corrected_Ext,
               p_norm_corrected_LabLib,
               p_norm_corrected_Len,
               p_norm_corrected_Tissue,
               ncol=2)

  #################
  #   DEG edgeR   #
  #################

  cat(paste0("Preparing for the DEG analysis in the ",modell_name,"batch corrected and TMM normalized model.\n"))
  #y_Library$samples$lib.size

  #colSums(y_Library$counts)

  all(rownames(y_norm$counts) == rownames(y$counts))
  #plotMDS(log(y_Library$counts), col = as.numeric(y_Library$samples$group))
  #legend("bottomright",as.character(unique(y_Library$samples$group)),col = 1:5, pch = 20)


  #unique(s4c$Condition)
  #unique(s4c$Tissue)
  cat("Generating design matrix.\n")
  design <- model.matrix(~0 + s4c$Condition)#* s4c$Tissue) #+ s4c$Length)

  #design
  rownames(design) <- colnames(y_norm)
  #design

  #Normalization factors are taken in to account even by using the not normalized matrix.
  y_norm <- estimateDisp(y_norm,design = design, robust = TRUE)

  #plotBCV(y_Library)

  cat("Fitting first part of the model.\n")
  fit <- glmQLFit(y = y_norm,design = design)#,contrast=c(1,-1,0,0,0))


  #Comparrissons among different conditions.
  #my_list4 <- lapply(seq(1:10),function(i) i)
  my_list4 <- list()
  #colnames(fit_Library)[2]

  cat("Generating Contrasts.\n")

  for (i in 1:(ncol(fit)-1)){
    for (j in (i+1):ncol(fit)){
      contrast=rep(0,ncol(fit))
      contrast[i]=1
      #print(contrast)
      contrast[j]=-1
      my_list4[[length(my_list4)+1]] <- contrast}
  }


  my_list4 <- setNames(object = my_list4,nm = letters[1:10])
  dataMetrics  <- (x = data.frame(ID = rownames(y$counts),y$counts))

  cat("Running contrasts on the model generated\n")
  #Run the contrasts in the modell.
  mylist_res <- lapply(my_list4,
                       function(x){

                         #Perform contrasts modelling
                         lrt <- glmQLFTest(fit, contrast = x)
                         #colnames(lrt)
                         #colnames(fit_Library)[1]
                         #gsub(x = gsub(x = strsplit(x = lrt$comparison,split = "-1")[[1]][1],pattern = "1\\*",replacement = ""),pattern = "\\$",replacement = "")
                         #gsub(x = strsplit(x = lrt$comparison,split = "-1")[[1]][2],pattern = "\\*",replacement = "")
                         contrastName <- paste0(gsub(x = gsub(x = strsplit(x = lrt$comparison,split = "-1")[[1]][1],pattern = "1\\*",replacement = ""),pattern = "\\ $",replacement = ""),
                                                "_vs_",
                                                gsub(x = strsplit(x = lrt$comparison,split = "-1")[[1]][2],pattern = "\\*",replacement = ""))
                         lrt <- topTags(lrt, n = nrow(y_norm[[1]]))[[1]]
                         #lrt
                         setDT(lrt, keep.rownames = TRUE)[]
                         lrt <- as.data.frame(lrt)

                         #colnames(lrt) <- paste0(colnames(fit_Library)[i], "_",colnames(fit_Library)[j], ".",colnames(lrt))
                         colnames(lrt)[1] = "ID"
                         lrt$contrastName <- contrastName
                         reorderID <- full_join(x = data.frame(ID = rownames(y$counts),y$counts),
                                                y =  lrt,
                                                by = "ID")

                         #reorderID
                         dataMetrics <- cbind(dataMetrics,
                                              reorderID[,-c(1:ncol(y$counts))])
                         dataMetrics})

  #Number of DEG for each comparisson.
  lapply(mylist_res,function(x) {
    paste0("Number of Differentially expressed genes: ",
           dim(subset(x,FDR < 0.05 & abs(logFC) > 1)),
           " in, ",
           x[["contrastName"]][1])[1]})

  #Save data.
  cat("Saving data.\n")
  lapply(mylist_res,function(x) {
    write.table(x = x,
                # file = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",
                #file = paste0("D:/PhD/Projects/prsa/BatchCorrected/edgeR/",
                file = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/edgeR/",
                              modell_name,
                              "/Contrasts_Results/",
                              x[["contrastName"]][1],
                              "_analysis_rna_",
                              modell_name,
                              ".txt"),
                sep = "\t",
                row.names = FALSE,
                col.names = TRUE)})
  return(list(
    mylist_res=mylist_res,
    normalized_tmm = y_normalized_corrected))
}
```

##### EdgeR Length

```{r, echo=TRUE,fig.height=32, fig.width=15}
require("dplyr")
modell_name = "Length"
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name))
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Contrasts_Results"))
#dir.create(recursive = TRUE,path =paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Graphs"))

#write.table(x = corrected_data_Platform,file = "C:/Users/marius/Desktop/corrected_batch_data_Platform.txt",col.names = TRUE,row.names = TRUE,sep = "\t")
#write.table(x = s4c,file = "C:/Users/marius/Desktop/s4c.txt",col.names = TRUE,row.names = TRUE,sep = "\t")

lists_platform_batch_corrected_edger_condition <- runModellEdgeR(modell_name = "Length",modell_in = corrected_data_Length)
```

##### EdgeR Library

```{r, echo=TRUE,fig.height=32, fig.width=15}
require("dplyr")
modell_name = "Platform"
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name))
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Contrasts_Results"))
#dir.create(recursive = TRUE,path =paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Graphs"))

#write.table(x = corrected_data_Platform,file = "C:/Users/marius/Desktop/corrected_batch_data_Platform.txt",col.names = TRUE,row.names = TRUE,sep = "\t")
#write.table(x = s4c,file = "C:/Users/marius/Desktop/s4c.txt",col.names = TRUE,row.names = TRUE,sep = "\t")

lists_platform_batch_corrected_edger_condition <- runModellEdgeR(modell_name = "Library",modell_in = corrected_data_Library)
```

##### EdgeR Platform

###### Check as the list returns now corrected counts by TMM and the results of the contrasts.
**Platform used for sequencing corrected modell edgeR**
```{r, echo=TRUE,fig.height=32, fig.width=15}
require("dplyr")
modell_name = "Platform"
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name))
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Contrasts_Results"))
#dir.create(recursive = TRUE,path =paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/",modell_name,"/Graphs"))

#write.table(x = corrected_data_Platform,file = "C:/Users/marius/Desktop/corrected_batch_data_Platform.txt",col.names = TRUE,row.names = TRUE,sep = "\t")
#write.table(x = s4c,file = "C:/Users/marius/Desktop/s4c.txt",col.names = TRUE,row.names = TRUE,sep = "\t")

lists_platform_batch_corrected_edger_condition <- runModellEdgeR(modell_name = "Platform",modell_in = corrected_data_Platform)
```
### UMAP Edger TMM
```{r,fig.height=8,fig.width=8}
umap_edger_platform <- umap_func_by_filtered_expression_normalized(df = lists_platform_batch_corrected_edger_condition$normalized_tmm ,n = 0,subtitle = "EdgeR TMM")
umap_edger_platform$umap_plot

#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/UMAP_EDGER_TMM_Platform_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/UMAP_EDGER_TMM_Platform_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP_EDGER_TMM_Platform_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP_EDGER_TMM_Platform_Counts.svg"),
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_edger_platform$umap_plot),
       width = 12,
       height = 10)
```


### DESeq2 function

Perform DESeq2 normalization after batch correction and see how data behaves to select a best workflow at the moment.

```{r echo=TRUE, fig.height=12, fig.width=15}
runModell <- function(modell_name ,modell_in){
  #Based on the model fitting run the analysis of the Contrasts based on the Condition.
  
  # This will run the DESeq2 modell specified above,
  # just specify the model name and the modell in for the Folder output and the DESeq2 model to perform analysis

  # --- Plot the original Counts
  cat(paste0("Plotting the boxplot of the raw counts (",modell_name," batch corrected)\n"))
  #pdf(file = paste0("../BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_Batch_Corrected_Data_Boxplot.pdf"),height = 12,width = 8)
  nbcd <- ggplot(stack(as.data.frame(modell_in@assays@data$counts)),aes(x = ind,y=values)) +
    geom_boxplot() +
    labs(x = "ID",
         y = "Values",
         title = paste0("Matrix of (",modell_name," batch corrected)\n")) +
    theme(axis.text.x = element_text(angle = 90))
  #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_Batch_Corrected_Data_Boxplot.svg"),
  #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_Batch_Corrected_Data_Boxplot.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_Batch_Corrected_Data_Boxplot.svg"),
         height = 12,
         width = 8,
         plot = nbcd)

  # Log10 original counts
  cat(paste0("Plotting the boxplot of the log10 raw counts (",modell_name," batch corrected)\n"))
  #pdf(file = paste0("../BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_log10_Batch_Corrected_Data_Boxplot.pdf"),height = 12,width = 8)
  log10nbcd <- ggplot(stack(as.data.frame(log10(modell_in@assays@data$counts))),aes(x = ind,y=values)) +
    geom_boxplot() +
    labs(x = "ID",
         y = "Log10 Values",
         title = paste0("Matrix of log 10 (",modell_name," batch corrected)\n")) +
    theme(axis.text.x = element_text(angle = 90))
  #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_log10_Batch_Corrected_Data_Boxplot.svg"),
  #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_log10_Batch_Corrected_Data_Boxplot.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_log10_Batch_Corrected_Data_Boxplot.svg"),
         height = 12,
         width = 8,
         plot = log10nbcd)


  # Log2 original counts
  cat(paste0("Plotting the boxplot of the log2 raw counts(",modell_name," batch corrected)\n"))
  #pdf(file = paste0("../BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_log2_Batch_Corrected_Data_Boxplot.pdf"),height = 12,width = 8)
  log2nbcd <- ggplot(stack(as.data.frame(log2(modell_in@assays@data$counts))),aes(x = ind,y=values)) +
    geom_boxplot() +
    labs(x = "ID",
         y = "Log2 Values",
         title = paste0("Matrix of log2 (",modell_name," batch corrected)\n")) +
    theme(axis.text.x = element_text(angle = 90))
  #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_log2_Batch_Corrected_Data_Boxplot.svg"),
  #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_log2_Batch_Corrected_Data_Boxplot.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/",modell_name,"_log2_Batch_Corrected_Data_Boxplot.svg"),
         height = 12,
         width = 8,
         plot = log2nbcd)


  # --- DESeq2 Procedure & Normalized Counts
  #DESeq2 Model main function.

  cat(paste0("Generating the DESeq2 model for ",modell_name," batch corrected data\n"))
  dds1 = DESeq2::DESeq(modell_in,betaPrior = FALSE,parallel = TRUE) #change back if in ubuntu
  #head(dds1@colData)
  #head(coefficients(dds1))
  
  #Get the normalized Counts Plots
  #head(assay(dds1))
  # head(counts(dds1))
  # head(dds1@assays@data$counts)
  # head(corrected_data_Platform)
  # head(DESeq2::counts(dds1,normalized=TRUE))
  # Plot Normalized Counts
  #boxplot
  cat(paste0("Plotting the boxplot of the ",modell_name," batch corrected and DESeq2 normalized counts\n"))
  #pdf(file = paste0("../BatchCorrected/",modell_name,"/Graphs/Normalize_Data_Boxplot_normalized_counts",modell_name,".pdf"),height = 12,width = 8)
  #pdf(file = "./Graphs/Normalize_Data_Boxplot_pirnas_rld.pdf",height = 12,width = 8)
  #pdf(file = paste0("../BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_normalized_",modell_name,"_batch_corrected_Counts.pdf"),height = 12,width = 8)

  normbcd <- ggplot(stack(as.data.frame(DESeq2::counts(dds1,normalized=TRUE))),
                    aes(x = ind,
                        y = values)) +
    geom_boxplot() +
    labs(x = "ID",
         y = "Values",
         title = paste0("Matrix of DESeq2 normalized ",modell_name," batch corrected Counts\n")) +
    theme(axis.text.x = element_text(angle = 90))
  #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_normalized_",modell_name,"_batch_corrected_Counts.svg"),
  # ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_normalized_",modell_name,"_batch_corrected_Counts.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_normalized_",modell_name,"_batch_corrected_Counts.svg"),
         plot = normbcd,
         width = 8,
         height = 12)

  # Log 10 Normalized Counts
  cat(paste0("Plotting the boxplot of the log10 ",modell_name," batch corrected DESeq2 normalized counts\n"))
  # #pdf(file = paste0("../BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_normalized_",modell_name,"_batch_corrected_log10_Counts.pdf"),height = 12,width = 8)
  # #pdf(file = "./Graphs/Normalize_Data_Boxplot_pirnas_rld.pdf",height = 12,width = 8)
  #pdf(file = "../03_Graphs/Original_Data_Boxplot_log10.pdf",height = 12,width = 8)
  log10normbcd <- ggplot(stack(as.data.frame(log10(DESeq2::counts(dds1,normalized=TRUE)))),
                         aes(x = ind,
                             y=values)) +
    geom_boxplot() +
    labs(x = "ID",
         y = "Log10 Values",
         title = paste0("Matrix of log10 ",modell_name," batch corrected DESeq2 normalized Counts\n")) +
    theme(axis.text.x = element_text(angle = 90))
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_normalized_",modell_name,"_batch_corrected_log10_Counts.svg"),
  #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_normalized_",modell_name,"_batch_corrected_log10_Counts.svg"),
  #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_normalized_",modell_name,"_batch_corrected_log10_Counts.svg"),
         plot = log10normbcd,
         width = 8,
         height = 12)


  # RLOG transformed counts
  cat(paste0("Plotting the boxplot of the rlog transformed ",modell_name," batch corrected DESeq2 normalized counts\n"))
  rld <- DESeq2::rlog(dds1,blind = FALSE)
  #head(assay(rld))
  #pdf(file = paste0("../BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_RLD_",modell_name,"_batch_corrected_Counts.pdf"),height = 12,width = 8)
  #pdf(file = "./Graphs/Normalize_Data_Boxplot_pirnas_rld.pdf",height = 12,width = 8)
  #pdf(file = "../03_Graphs/Original_Data_Boxplot_log10.pdf",height = 12,width = 8)
  rlognormbcd <- ggplot(stack(as.data.frame(assay(rld))),
                        aes(x = ind,
                            y=values)) +
    geom_boxplot() +
    labs(x = "ID",
         y = "RLOG Values",
         title = paste0("Matrix of rld Transformed ",modell_name," batch corrected DESeq2 normalized Counts\n")) +
    theme(axis.text.x = element_text(angle = 90))
  #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_RLD_",modell_name,"_batch_corrected_Counts.svg"),
  # ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_RLD_",modell_name,"_batch_corrected_Counts.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_RLD_",modell_name,"_batch_corrected_Counts.svg"),

         plot = rlognormbcd,
         height = 12,
         width = 8)



  #VSD transformed counts
  cat(paste0("Plotting the boxplot of the vsd transformed ",modell_name," batch corrected DESeq2 normalized Counts\n"))
  vsd <- DESeq2::varianceStabilizingTransformation(dds1,blind = FALSE)
  #head(assay(vsd))


  #svg(file = paste0("../BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_VSD_",modell_name,"_batch_corrected_Counts.svg"),height = 12,width = 8)
  #pdf(file = "./Graphs/Normalize_Data_Boxplot_pirnas_rld.pdf",height = 12,width = 8)
  #pdf(file = "../03_Graphs/Original_Data_Boxplot_log10.pdf",height = 12,width = 8)
  vsdnormbcd <- ggplot(stack(as.data.frame(assay(vsd))),
                       aes(x = ind,
                           y=values)) +
    geom_boxplot() +
    labs(x = "ID",
         y = "VSD Values",
         title = paste0("Matrix of vsd transformed ",modell_name," batch corrected DESeq2 normalized Counts\n")) +
    theme(axis.text.x = element_text(angle = 90))
  
  #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_VSD_",modell_name,"_batch_corrected_Counts.svg"),
  #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_VSD_",modell_name,"_batch_corrected_Counts.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_VSD_",modell_name,"_batch_corrected_Counts.svg"),
         plot = vsdnormbcd,
         height = 12,
         width = 8)


  #Gather them together
    #ggsave(filename = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_Normalizations_",modell_name,"_Counts.pdf"),
    #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_Normalizations_",modell_name,"_Counts.pdf"),
    ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/DESeq2_Normalizations_",modell_name,"_Counts.pdf"),
           plot = grid.arrange(nbcd,log2nbcd,log10nbcd,normbcd,log10normbcd,rlognormbcd,vsdnormbcd),
         width = 18,
         height = 20)

  grid.arrange(nbcd,
               log2nbcd,
               log10nbcd,
               normbcd,
               log10normbcd,
               rlognormbcd,
               vsdnormbcd)

  # --- Add PCA of the DESeq2 Normalization processed, for comparing with EdgeR.
  
  
  normalized_corrected_pcas <- prcomp(DESeq2::counts(dds1,normalized=TRUE))
  percentVar_normalized_corrected <- normalized_corrected_pcas$sdev^2/sum(normalized_corrected_pcas$sdev^2)
  normalized_corrected_pcas_df <- as.data.frame(normalized_corrected_pcas[2]$rotation)

  #Add labels
  normalized_corrected_pcas_df[,"Lab"] <- s4c$Lab
  normalized_corrected_pcas_df[,"Platform"] <- s4c$Platform
  normalized_corrected_pcas_df[,"Library"] <- s4c$Library
  normalized_corrected_pcas_df[,"Condition"] <- s4c$Condition
  normalized_corrected_pcas_df[,"Sample"] <- rownames(s4c)
  normalized_corrected_pcas_df["Length"] <- s4c$Length
  normalized_corrected_pcas_df["LabLib"] <- s4c$LabLib
  normalized_corrected_pcas_df["Extract"] <- s4c$Extract

  
  cat("Adding the PCA labels plot.\n")
  # PCA function to add the labels from s4c
  pca_var_plot_func <- function(df,var,subtitle){
    n <- ggplot(data = df,aes(x = PC1,
                              y = PC2,
                              color = s4c[["Condition"]],
                              shape = s4c[[var]]))
    
    #n <- n + geom_point(size=3)
    n <- n + theme(panel.grid = element_blank(),
                   panel.border = element_rect(fill = "transparent"),
                   panel.background = element_rect(fill = "transparent"),
                   legend.title = element_text(size = 15),
                   legend.text = element_text(size = 12))
    n <- n + labs(title="PCA of RNA-seq",
                  color = "Condition",
                  shape = var,
                  subtitle = paste0(subtitle," DESeq2 Normalized")) +
      # xlim(0,0.5) +
      # ylim(0,0.5) +
      xlab(paste0("PC",substr("PC1",start = 3,stop = 3),": ",
                  round(percentVar_normalized_corrected[as.numeric(substr("PC1",start = 3,stop = 3))] * 100), "% variance")) +
      ylab(paste0("PC",substr("PC2",start = 3,stop = 3),": ",
                  round(percentVar_normalized_corrected[as.numeric(substr("PC2",start = 3,stop = 3))] * 100), "% variance"))
    #n <- n + scale_color_brewer(palette = "Paired") #Accent #Dark2
    n <- n + scale_color_manual(values=cbf_1)
    n <- n + scale_shape_manual(values=shape_values_lab)
    #n <- n + geom_jitter(width = 0.01,height = 0.01,size=3)
    n <- n + geom_jitter(position = position_jitter(width = 0.01,height = 0.01,seed = 123456),size=8)
    n}  
  
  #s4c[["LabLib"]]
  p_norm_corrected_Lab <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Lab",subtitle = modell_name)
  p_norm_corrected_Plt <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Platform",subtitle = modell_name)
  p_norm_corrected_Lib <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Library",subtitle = modell_name)
  p_norm_corrected_Ext <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Extract",subtitle = modell_name)
  p_norm_corrected_LabLib <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "LabLib",subtitle = modell_name)
  p_norm_corrected_Len <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Length",subtitle = modell_name)
  p_norm_corrected_Tissue <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Tissue",subtitle = modell_name)
  #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
  #ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),dpi = 320,device = "svg",
  #ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),dpi = 320,device = "svg",
  #ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
         dpi = 320,
         device = "svg",
         plot = grid.arrange(p_norm_corrected_Lab,
                             p_norm_corrected_Plt,
                             p_norm_corrected_Lib,
                             p_norm_corrected_Ext,
                             p_norm_corrected_LabLib,
                             p_norm_corrected_Len,
                             p_norm_corrected_Tissue,
                             ncol=2),
         width = 22,
         height = 32)
  
  grid.arrange(p_norm_corrected_Lab,
               p_norm_corrected_Plt,
               p_norm_corrected_Lib,
               p_norm_corrected_Ext,
               p_norm_corrected_LabLib,
               p_norm_corrected_Len,
               p_norm_corrected_Tissue,
               ncol=2)
  
  
  
  #RLD PCA
  
  
  normalized_corrected_pcas <- prcomp(assay(rld))
  percentVar_normalized_corrected <- normalized_corrected_pcas$sdev^2/sum(normalized_corrected_pcas$sdev^2)
  normalized_corrected_pcas_df <- as.data.frame(normalized_corrected_pcas[2]$rotation)

  
  cat("Adding the PCA RLD plot.\n")
  # PCA function to add the labels from s4c
  pca_var_plot_func <- function(df,var,subtitle){
    n <- ggplot(data = df,aes(x = PC1,
                              y = PC2,
                              color = s4c[["Condition"]],
                              shape = s4c[[var]]))
    
    #n <- n + geom_point(size=3)
    n <- n + theme(panel.grid = element_blank(),
                   panel.border = element_rect(fill = "transparent"),
                   panel.background = element_rect(fill = "transparent"),
                   legend.title = element_text(size = 15),
                   legend.text = element_text(size = 12))
    n <- n + labs(title="PCA of RNA-seq",
                  color = "Condition",
                  shape = var,
                  subtitle = paste0(subtitle," DESeq2 RLD")) +
      # xlim(0,0.3) +
      # ylim(0,0.3) +
      xlab(paste0("PC",substr("PC1",start = 3,stop = 3),": ",
                  round(percentVar_normalized_corrected[as.numeric(substr("PC1",start = 3,stop = 3))] * 100), "% variance")) +
      ylab(paste0("PC",substr("PC2",start = 3,stop = 3),": ",
                  round(percentVar_normalized_corrected[as.numeric(substr("PC2",start = 3,stop = 3))] * 100), "% variance"))
    #n <- n + scale_color_brewer(palette = "Paired") #Accent #Dark2
    n <- n + scale_color_manual(values=cbf_1)
    n <- n + scale_shape_manual(values=shape_values_lab)
    #n <- n + geom_jitter(width = 0.01,height = 0.01,size=3)
    n <- n + geom_jitter(position = position_jitter(width = 0.01,height = 0.01,seed = 123456),size=8)
    n}  
  
  #s4c[["LabLib"]]
  p_norm_corrected_Lab <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Lab",subtitle = modell_name)
  p_norm_corrected_Plt <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Platform",subtitle = modell_name)
  p_norm_corrected_Lib <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Library",subtitle = modell_name)
  p_norm_corrected_Ext <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Extract",subtitle = modell_name)
  p_norm_corrected_LabLib <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "LabLib",subtitle = modell_name)
  p_norm_corrected_Len <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Length",subtitle = modell_name)
  p_norm_corrected_Tissue <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Tissue",subtitle = modell_name)
  
  #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
  #ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),dpi = 320,device = "svg",
  #ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),dpi = 320,device = "svg",
  #ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_RLD_",modell_name,"_batch_corrected_Counts.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_RLD_",modell_name,"_batch_corrected_Counts.svg"),
         dpi = 320,
         device = "svg",
         plot = grid.arrange(p_norm_corrected_Lab,
                             p_norm_corrected_Plt,
                             p_norm_corrected_Lib,
                             p_norm_corrected_Ext,
                             p_norm_corrected_LabLib,
                             p_norm_corrected_Len,
                             p_norm_corrected_Tissue,
                             ncol=2),
         width = 22,
         height = 32)
  
  
  
  
  #VSD PCA
  
  
  normalized_corrected_pcas <- prcomp(assay(vsd))
  percentVar_normalized_corrected <- normalized_corrected_pcas$sdev^2/sum(normalized_corrected_pcas$sdev^2)
  normalized_corrected_pcas_df <- as.data.frame(normalized_corrected_pcas[2]$rotation)
  
  cat("Adding the PCA VSD plot.\n")
  # PCA function to add the labels from s4c
  pca_var_plot_func <- function(df,var,subtitle){
    n <- ggplot(data = df,aes(x = PC1,
                              y = PC2,
                              color = s4c[["Condition"]],
                              shape = s4c[[var]]))
    
    #n <- n + geom_point(size=3)
    n <- n + theme(panel.grid = element_blank(),
                   panel.border = element_rect(fill = "transparent"),
                   panel.background = element_rect(fill = "transparent"),
                   legend.title = element_text(size = 15),
                   legend.text = element_text(size = 12))
    n <- n + labs(title="PCA of RNA-seq",
                  color = "Condition",
                  shape = var,
                  subtitle = paste0(subtitle," DESeq2 VSD")) +
      #xlim(0,0.3) +
      #ylim(0,0.3) +
      xlab(paste0("PC",substr("PC1",start = 3,stop = 3),": ",
                  round(percentVar_normalized_corrected[as.numeric(substr("PC1",start = 3,stop = 3))] * 100), "% variance")) +
      ylab(paste0("PC",substr("PC2",start = 3,stop = 3),": ",
                  round(percentVar_normalized_corrected[as.numeric(substr("PC2",start = 3,stop = 3))] * 100), "% variance"))
    #n <- n + scale_color_brewer(palette = "Paired") #Accent #Dark2
    n <- n + scale_color_manual(values=cbf_1)
    n <- n + scale_shape_manual(values=shape_values_lab)
    #n <- n + geom_jitter(width = 0.01,height = 0.01,size=3)
    n <- n + geom_jitter(position = position_jitter(width = 0.01,height = 0.01,seed = 123456),size=8)
    n}  
  
  #s4c[["LabLib"]]
  p_norm_corrected_Lab <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Lab",subtitle = modell_name)
  p_norm_corrected_Plt <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Platform",subtitle = modell_name)
  p_norm_corrected_Lib <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Library",subtitle = modell_name)
  p_norm_corrected_Ext <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Extract",subtitle = modell_name)
  p_norm_corrected_LabLib <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "LabLib",subtitle = modell_name)
  p_norm_corrected_Len <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Length",subtitle = modell_name)
  p_norm_corrected_Tissue <- pca_var_plot_func(df = normalized_corrected_pcas_df,var = "Tissue",subtitle = modell_name)

  
  #ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
  #ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),dpi = 320,device = "svg",
  #ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),dpi = 320,device = "svg",
  #ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_VSD_",modell_name,"_batch_corrected_Counts.svg"),dpi = 320,device = "svg",
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_VSD_",modell_name,"_batch_corrected_Counts.svg"),
  #ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_VSD_",modell_name,"_batch_corrected_Counts.svg"),
         dpi = 320,
         device = "svg",
         plot = grid.arrange(p_norm_corrected_Lab,
                             p_norm_corrected_Plt,
                             p_norm_corrected_Lib,
                             p_norm_corrected_Ext,
                             p_norm_corrected_LabLib,
                             p_norm_corrected_Len,
                             p_norm_corrected_Tissue,
                             ncol=2),
         width = 22,
         height = 32)
  # --- Build Contrasts and Run them.
  #Run contrast builder
  cat("Building the contrasts table\n")
  my_list <- as.list.data.frame(gtools::permutations(v = s4c$Condition,n = 5,r = 2))
  #class(my_list)
  my_list <- cbind(my_list,rep("Condition",length(my_list[,1])))
  #my_list
  #Reorder
  my_list2 <-my_list[,c(3,1,2)]
  names(my_list2) <- NULL
  #my_list2
  #my_list2[,1]

  ##########
  #Long way#
  ##########
  #my_list4 <- list()
  #Create the list of the  length desired.
  my_list4 <- lapply(seq_along(1:length(my_list2[,1])),function(i) i)
  #Run the list filler.
  for (i in 1:nrow(my_list2)){
    for (j in 1:ncol(my_list2)){
      #print(my_list2[i,j])
      my_list4[[i]][[j]] = my_list2[i,j]
      #my_list4[[i]][[1]] <- my_list2[i,1]
      #my_list4[[i]][[2]] <- my_list2[i,2]
      #my_list4[[i]][[3]] <- my_list2[i,3]
    }
  }

  my_list4 <- setNames(object = my_list4,nm = letters[1:length(my_list2[,1])])
  #my_list4
  cat("Running contrasts on the model generated\n")
  #Run the contrasts in the modell.
  # mylist_res <- lapply(my_list4,
  #                      function(x){
  #                        dds_res_compare = DESeq2::results(object = dds1,
  #                                                          contrast = c(x),
  #                                                          #parallel=TRUE,
  #                                                          pAdjustMethod = "BH",
  #                                                          tidy=F)
  #                        #,
  #                        #BPPARAM = MulticoreParam(2))
  #                        dds_res_compare
  #                      })
  #
  mylist_res <- lapply(my_list4,
                       function(x){
                         dds_res_compare = DESeq2::lfcShrink(dds = dds1,
                                                             #coef = 2,
                                                             contrast = c(x),
                                                             type = "ashr")#,parallel=TRUE) not with ashr
                         # pAdjustMethod = "BH",
                         # tidy=F)
                         #,
                         #BPPARAM = MulticoreParam(2))
                         dds_res_compare})
  #lapply(mylist_res,head)
  #getwd()
  cat("Saving the contrasts results\n")
  comparissons_realised <- lapply(mylist_res, function(x) mcols(x)$description)
  getwd()
  write.table(x = comparissons_realised,
              # file = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",
              #file = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",
              file = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",
                            modell_name,
                            "/Contrasts_Results/comparrissons_realised_",
                            modell_name,
                            ".txt"),
              sep = "\t")
  
  #Write and report the results.
  #lapply(names(mylist_res),function(x) mylist_res[[x]][c(2,3)])
  my_list_comp <- lapply(names(my_list4),function(x) paste(my_list4[[x]][c(2,3)],collapse = "_vs_"))
  #lapply(seq_along(mylist_res),function(x)x)
  #my_list_comp[[2]]
  lapply(seq_along(mylist_res),function(x)
    write.table(mylist_res[[x]],
                #paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/",
                paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/",
                       modell_name,
                       "/Contrasts_Results/",
                       my_list_comp[[x]],
                       "_analysis_rna_lfcS_",
                       modell_name,
                       ".txt"),
                sep = "\t",
                row.names = TRUE,
                col.names = TRUE))
  #write.table(mylist_res[[x]],paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Contrasts_Results/",my_list_comp[[x]],"_analysis_rna_lfcS_",modell_name,".txt"),sep = "\t",row.names = TRUE,col.names = TRUE))

  return(mylist_res)

  #Log2FoldChange Distributions
  #cat("Generating the log2FoldChanges distributions for all the samples, excepting the Uninjured vs...\n")
}
```






```{r}
#BiocManager::install("BiocParallel")
library("BiocParallel")
library("DESeq2")
library("stats")
library("matrixStats")
#SnowParam(workers = multicoreWorkers())
#snow <- SnowParam(workers = 6, type = "SOCK")
#param <- SnowParam(workers = 6, type = "MPI")
register(MulticoreParam())#ubuntu
#register(SnowParam(workers = 8)) #windows
```

### PCA DESeq2 Length
```{r,echo=TRUE,fig.height=12, fig.width=15}
dds_Length <- DESeqDataSetFromMatrix(countData = corrected_data_Length,
                                      colData = s4c,
                                      design = ~ Condition)
head(dds_Length)

modell_name = "Length"
modell_in <- dds_Length

cat(paste0("Generating the DESeq2 model for ",modell_name," batch corrected data\n"))
dds1_Length = DESeq2::DESeq(modell_in,betaPrior = FALSE, parallel = TRUE)#,BPPARAM = param)

normalized_corrected_pcas_Length <- prcomp(DESeq2::counts(dds1_Length,normalized=TRUE))
percentVar_normalized_corrected_Length <- normalized_corrected_pcas_Length$sdev^2/sum(normalized_corrected_pcas_Length$sdev^2)
normalized_corrected_pcas_Length_df <- as.data.frame(normalized_corrected_pcas_Length[2]$rotation)
```


```{r,echo=TRUE,fig.height=36, fig.width=22}
#s4c[["LabLib"]]
p_norm_Len_corrected_Lab <- pca_var_plot_func(df = normalized_corrected_pcas_Length_df , var = "Lab",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Length,pca = 1,pcb = 2)
p_norm_Len_corrected_Plt <- pca_var_plot_func(df = normalized_corrected_pcas_Length_df , var = "Platform",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Length,pca = 1,pcb = 2)
p_norm_Len_corrected_Lib <- pca_var_plot_func(df = normalized_corrected_pcas_Length_df , var = "Library",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Length,pca = 1,pcb = 2)
p_norm_Len_corrected_Ext <- pca_var_plot_func(df = normalized_corrected_pcas_Length_df, var = "Extract",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Length,pca = 1,pcb = 2)
p_norm_Len_corrected_LabLib <- pca_var_plot_func(df = normalized_corrected_pcas_Length_df, var = "LabLib",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Length,pca = 1,pcb = 2)
p_norm_Len_corrected_Len <- pca_var_plot_func(df =normalized_corrected_pcas_Length_df, var = "Length",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Length,pca = 1,pcb = 2)
p_norm_Len_corrected_Tissue <- pca_var_plot_func(df =normalized_corrected_pcas_Length_df, var = "Tissue",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Length,pca = 1,pcb = 2)

#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Length/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Length/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
       dpi = 320,
       device = "svg",
       plot = grid.arrange(p_norm_Len_corrected_Lab,
                           p_norm_Len_corrected_Plt,
                           p_norm_Len_corrected_Lib,
                           p_norm_Len_corrected_Ext,
                           p_norm_Len_corrected_LabLib,
                           p_norm_Len_corrected_Len,
                           p_norm_Len_corrected_Tissue,
                           ncol=2),
       width = 22,
       height = 32)
grid.arrange(p_norm_Len_corrected_Lab,
             p_norm_Len_corrected_Plt,
             p_norm_Len_corrected_Lib,
             p_norm_Len_corrected_Ext,
             p_norm_Len_corrected_LabLib,
             p_norm_Len_corrected_Len,
             p_norm_Len_corrected_Tissue,
             ncol=2)
```


```{r}
#dir.create(recursive = TRUE,path = "D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Length/")
pca_nrm_len_pc1_pc2_list <- list("Platform" = p_norm_Len_corrected_Plt,
                                   "Lab" = p_norm_Len_corrected_Lab,
                                   "Library" = p_norm_Len_corrected_Lib,
                                   "Extraction" = p_norm_Len_corrected_Ext,
                                   "LabLib" = p_norm_Len_corrected_LabLib,
                                   "Length" = p_norm_Len_corrected_Len,
                                   "Tissue" = p_norm_Len_corrected_Tissue)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_nrm_len_pc1_pc2_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Length/PCA_NORMALIZED_Length",plotIt,"pc1_pc2_Counts.svg"),plot = pca_nrm_len_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Length/PCA_NORMALIZED_Length",plotIt,"pc1_pc2_Counts.svg"),plot = pca_nrm_len_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```

### UMAP DESeq2 Length
```{r,fig.height=8,fig.width=8}
umap_deseq_length <- umap_func_by_filtered_expression_normalized(df = DESeq2::counts(dds1_Length,normalized=TRUE) ,n = 0,subtitle = "DESeq2 Length")
umap_deseq_length$umap_plot

#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_deseq_length$umap_plot),
       width = 24,
       height = 20)
```

### Run model for data
```{r,echo=TRUE,fig.height=32, fig.width=22}
modell_name = "Length"
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name))
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Contrasts_Results"))
#dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs"))
lists_length_batch_corrected_deseq_condition <- runModell(modell_name = "Length",modell_in = dds_Length)
```

### PCA DESeq2 Library

```{r,echo=TRUE,fig.height=12, fig.width=15}
require(DESeq2)
dds_Library <- DESeqDataSetFromMatrix(countData = corrected_data_Library,
                                      colData = s4c,
                                      design = ~ Condition)
head(dds_Library)
modell_name = "Library"
modell_in <- dds_Library

cat(paste0("Generating the DESeq2 model for ",modell_name," batch corrected data\n"))
dds1_Library = DESeq2::DESeq(modell_in,betaPrior = FALSE, parallel = TRUE)

normalized_corrected_pcas_Library <- prcomp(DESeq2::counts(dds1_Library,normalized=TRUE))
percentVar_normalized_corrected_Library <- normalized_corrected_pcas_Library$sdev^2/sum(normalized_corrected_pcas_Library$sdev^2)
normalized_corrected_pcas_Library_df <- as.data.frame(normalized_corrected_pcas_Library[2]$rotation)
```

```{r,echo=TRUE,fig.height=32, fig.width=22}
#s4c[["LabLib"]]
p_norm_Lib_corrected_Lab <- pca_var_plot_func(df = normalized_corrected_pcas_Library_df,var = "Lab",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Library,pca = 1,pcb = 2)
p_norm_Lib_corrected_Plt <- pca_var_plot_func(df = normalized_corrected_pcas_Library_df,var = "Platform",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Library,pca = 1,pcb = 2)
p_norm_Lib_corrected_Lib <- pca_var_plot_func(df = normalized_corrected_pcas_Library_df,var = "Library",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Library,pca = 1,pcb = 2)
p_norm_Lib_corrected_Ext <- pca_var_plot_func(df = normalized_corrected_pcas_Library_df,var = "Extract",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Library,pca = 1,pcb = 2)
p_norm_Lib_corrected_LabLib <- pca_var_plot_func(df = normalized_corrected_pcas_Library_df,var = "LabLib",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Library,pca = 1,pcb = 2)
p_norm_Lib_corrected_Len <- pca_var_plot_func(df = normalized_corrected_pcas_Library_df,var = "Length",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Library,pca = 1,pcb = 2)
p_norm_Lib_corrected_Tissue <- pca_var_plot_func(df = normalized_corrected_pcas_Library_df,var = "Tissue",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Library,pca = 1,pcb = 2)


#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Library/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Library/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
       dpi = 320,
       device = "svg",
       plot = grid.arrange(p_norm_Lib_corrected_Lab,
                           p_norm_Lib_corrected_Plt,
                           p_norm_Lib_corrected_Lib,
                           p_norm_Lib_corrected_Ext,
                           p_norm_Lib_corrected_LabLib,
                           p_norm_Lib_corrected_Len,
                           p_norm_Lib_corrected_Tissue,
                           ncol=2),
       width = 22,
       height = 32)
grid.arrange(p_norm_Lib_corrected_Lab,
             p_norm_Lib_corrected_Plt,
             p_norm_Lib_corrected_Lib,
             p_norm_Lib_corrected_Ext,
             p_norm_Lib_corrected_LabLib,
             p_norm_Lib_corrected_Len,
             p_norm_Lib_corrected_Tissue,
             ncol=2)
```

```{r}
#dir.create(recursive = TRUE,path = "D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Library/")
pca_nrm_lib_pc1_pc2_list <- list("Platform" = p_norm_Lib_corrected_Plt,
                                 "Lab" = p_norm_Lib_corrected_Lab,
                                 "Library" = p_norm_Lib_corrected_Lib,
                                 "Extraction" = p_norm_Lib_corrected_Ext,
                                 "LabLib" = p_norm_Lib_corrected_LabLib,
                                 "Length" = p_norm_Lib_corrected_Len,
                                 "Tissue" = p_norm_Lib_corrected_Tissue)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_nrm_lib_pc1_pc2_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Library/PCA_NORMALIZED_Library",plotIt,"pc1_pc2_Counts.svg"),plot = pca_nrm_lib_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Library/PCA_NORMALIZED_Library",plotIt,"pc1_pc2_Counts.svg"),plot = pca_nrm_lib_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```

### UMAP DESeq2 Library
```{r,fig.height=8,fig.width=8}
umap_deseq_library <- umap_func_by_filtered_expression_normalized(df = DESeq2::counts(dds1_Library,normalized=TRUE) ,n = 0,subtitle = "DESeq2 Library")
umap_deseq_library$umap_plot

#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_deseq_library$umap_plot),
       width = 12,
       height = 10)
```

### Run model for data
```{r,echo=TRUE,fig.height=32, fig.width=22}
modell_name = "Library"
# dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name))
# dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Contrasts_Results"))
# dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs"))
lists_library_batch_corrected_deseq_condition <- runModell(modell_name = "Library",modell_in = dds_Library)
```
### PCA DESeq2 Platform

##### DESeq2 Platform 
```{r,echo=TRUE,fig.height=12, fig.width=15}
require(DESeq2)
dds_Platform <- DESeqDataSetFromMatrix(countData = corrected_data_Platform,
                                       colData = s4c,
                                       design = ~ Condition)
head(dds_Platform)
modell_name = "Platform"
modell_in <- dds_Platform
cat(paste0("Generating the DESeq2 model for ",modell_name," batch corrected data\n"))
dds1_Platform = DESeq2::DESeq(modell_in,betaPrior = FALSE, parallel = TRUE)

normalized_corrected_pcas_Platform <- prcomp(DESeq2::counts(dds1_Platform,normalized=TRUE))
percentVar_normalized_corrected_Platform <- normalized_corrected_pcas_Platform$sdev^2/sum(normalized_corrected_pcas_Platform$sdev^2)
normalized_corrected_pcas_Platform_df <- as.data.frame(normalized_corrected_pcas_Platform[2]$rotation)
```



```{r,echo=TRUE,fig.height=32, fig.width=22}
#s4c[["LabLib"]]
p_norm_Plt_corrected_Lab <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Lab",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 2)
p_norm_Plt_corrected_Plt <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Platform",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 2)
p_norm_Plt_corrected_Lib <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Library",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 2)
p_norm_Plt_corrected_Ext <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Extract",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 2)
p_norm_Plt_corrected_LabLib <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "LabLib",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 2)
p_norm_Plt_corrected_Len <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Length",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 2)
p_norm_Plt_corrected_Tissue <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Tissue",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 2)


#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
       dpi = 320,
       device = "svg",
       plot = grid.arrange(p_norm_Plt_corrected_Lab,
                           p_norm_Plt_corrected_Plt,
                           p_norm_Plt_corrected_Lib,
                           p_norm_Plt_corrected_Ext,
                           p_norm_Plt_corrected_LabLib,
                           p_norm_Plt_corrected_Len,
                           p_norm_Plt_corrected_Tissue,
                           ncol=2),
       width = 22,
       height = 32)
grid.arrange(p_norm_Plt_corrected_Lab,
             p_norm_Plt_corrected_Plt,
             p_norm_Plt_corrected_Lib,
             p_norm_Plt_corrected_Ext,
             p_norm_Plt_corrected_LabLib,
             p_norm_Plt_corrected_Len,
             p_norm_Plt_corrected_Tissue,ncol=2)
```

```{r}
#dir.create(recursive = TRUE,path = "D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/")
pca_nrm_plt_pc1_pc2_list <- list("Platform" = p_norm_Plt_corrected_Plt,
                                   "Lab" = p_norm_Plt_corrected_Lab,
                                   "Library" = p_norm_Plt_corrected_Lib,
                                   "Extraction" = p_norm_Plt_corrected_Ext,
                                   "LabLib" = p_norm_Plt_corrected_LabLib,
                                   "Length" = p_norm_Plt_corrected_Len,
                                   "Tissue" = p_norm_Plt_corrected_Tissue)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_nrm_plt_pc1_pc2_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PCA_NORMALIZED_Platform",plotIt,"pc1_pc2_Counts.svg"),plot = pca_nrm_plt_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PCA_NORMALIZED_Platform",plotIt,"pc1_pc2_Counts.svg"),plot = pca_nrm_plt_pc1_pc2_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```

### UMAP DESeq2 Platform
```{r,fig.height=8,fig.width=8}
umap_deseq_platform <- umap_func_by_filtered_expression_normalized(df = DESeq2::counts(dds1_Platform,normalized=TRUE) ,n = 0,subtitle = "DESeq2 Platform")
umap_deseq_platform$umap_plot

#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_deseq_platform$umap_plot),
       width = 12,
       height = 10)
```

```{r,fig.width=22,fig.height=32}
# dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name))
# dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Contrasts_Results"))
# dir.create(recursive = TRUE,path = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs"))
lists_platform_batch_corrected_deseq_condition <- runModell(modell_name = "Platform",modell_in = dds_Platform)
# lapply(lists_platform_batch_corrected_deseq_condition,head)
# head(lists_platform_batch_corrected_deseq_condition$t)
# head(lists_platform_batch_corrected_deseq_condition$p)
# 
# dim(lists_platform_batch_corrected_deseq_condition$g)
#lists_platform_batch_corrected_deseq_condition_ann$Sham_vs_Uninjured |> group_by(log2FoldChange) |> arrange(log2FoldChange)
#results(lists_platform_batch_corrected_deseq_condition$t) |> group_by(log2FoldChange) |> arrange(log2FoldChange)
```


```{r,fig.width=30,fig.height=10}
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/UMAP__Length_Platform_Library_normalized_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/UMAP__Length_Platform_Library_normalized_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP__Length_Platform_Library_normalized_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/UMAP/NORMALIZED/UMAP__Length_Platform_Library_normalized_Counts.svg"),
       device = "svg",
       dpi = 320,
       plot = grid.arrange(umap_deseq_length$umap_plot,
                           umap_deseq_platform$umap_plot,
                           umap_deseq_library$umap_plot,
                           nrow = 1),
       width = 30,
       height = 12)

grid.arrange(umap_deseq_length$umap_plot,
                           umap_deseq_platform$umap_plot,
                           umap_deseq_library$umap_plot,
                           nrow = 1)

```


x### Three batch corrected data.
```{r, fig.width=30,fig.height=10}
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA__Length_Platform_Library_normalized_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA__Length_Platform_Library_normalized_Counts.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/PCA__Length_Platform_Library_normalized_Counts.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/PCA__Length_Platform_Library_normalized_Counts.svg"),
       device = "svg",
       dpi = 320,
       plot = grid.arrange(p_norm_Len_corrected_Lab,
                           p_norm_Plt_corrected_Lab,
                           p_norm_Lib_corrected_Lab,
                           nrow = 1),
       width = 28,
       height = 10)

grid.arrange(p_norm_Len_corrected_Lab,
             p_norm_Plt_corrected_Lab,
             p_norm_Lib_corrected_Lab,
             nrow = 1)
```


### Check other PCn
```{r,fig.width=22,fig.height=32}
# PCA function to add the labels from s4c
pca_var_plot_func <- function(df,var,subtitle,percentVar_obj,pca,pcb){
  n <- ggplot(data = df,aes(x = df[[pca]],
                            y = df[[pcb]],
                            color = s4c[["Condition"]],
                            shape = s4c[[var]]))
  
  #n <- n + geom_point(size=3)
  n <- n + theme(panel.grid = element_blank(),
                 panel.border = element_rect(fill = "transparent"),
                 panel.background = element_rect(fill = "transparent"),
                 legend.title = element_text(size = 15),
                 legend.text = element_text(size = 12))
  
  n <- n + labs(title="PCA of RNA-seq",
                color = "Condition",
                shape = var,
                subtitle = paste0(subtitle)) +
    xlab(paste0("PC",substr(paste0("PC",pca),start = 3,stop = 3),": ",
                round(percentVar_obj[as.numeric(substr(paste0("PC",pca),start = 3,stop = 3))] * 100), "% variance")) +
    ylab(paste0("PC",substr(paste0("PC",pcb),start = 3,stop = 3),": ",
                round(percentVar_obj[as.numeric(substr(paste0("PC",pcb),start = 3,stop = 3))] * 100), "% variance"))
  
  #n <- n + scale_color_brewer(palette = "Paired") #Accent #Dark2
  n <- n + scale_color_manual(values=cbf_1)
  n <- n + scale_shape_manual(values=shape_values_lab)
  #n <- n + geom_jitter(width = 0.01,height = 0.01,size=3)
  n <- n + geom_jitter(position = position_jitter(width = 0.000000001,height = 0.000000001,seed = 123456),size=8)# + xlim(0,0.5) + ylim(0,0.5)
  n}
```



```{r,fig.width=11,fig.height=7}
s4c$T
p_norm_Plt_corrected_Tissue_pc1_2 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Tissue",subtitle = modell_name, percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 2)

p_norm_Plt_corrected_Lab_pc1_2
```

### PC1 & PC3
```{r,fig.width=22,fig.height=32}
p_norm_Plt_corrected_Lab_pc1_3 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Lab",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 3)
p_norm_Plt_corrected_Plt_pc1_3 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Platform",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 3)
p_norm_Plt_corrected_Lib_pc1_3 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Library",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 3)
p_norm_Plt_corrected_Ext_pc1_3 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Extract",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 3)
p_norm_Plt_corrected_LabLib_pc1_3 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "LabLib",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 3)
p_norm_Plt_corrected_Len_pc1_3 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Length",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 3)
p_norm_Plt_corrected_Tissue_pc1_3 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Tissue",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 3)

#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC1_PC3.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC1_PC3.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC1_PC3.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC1_PC3.svg"),
       dpi = 320,
       device = "svg",
       plot = grid.arrange(p_norm_Plt_corrected_Lab_pc1_3,
                           p_norm_Plt_corrected_Plt_pc1_3,
                           p_norm_Plt_corrected_Lib_pc1_3,
                           p_norm_Plt_corrected_Ext_pc1_3,
                           p_norm_Plt_corrected_LabLib_pc1_3,
                           p_norm_Plt_corrected_Len_pc1_3,
                           p_norm_Plt_corrected_Tissue_pc1_3,ncol = 2),
       width = 22,
       height = 32)

```

```{r}
#dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC1PC3")
pca_nrm_plt_pc1_pc3_list <- list("Platform" = p_norm_Plt_corrected_Plt_pc1_3,
                                   "Lab" = p_norm_Plt_corrected_Lab_pc1_3,
                                   "Library" = p_norm_Plt_corrected_Lib_pc1_3,
                                   "Extraction" = p_norm_Plt_corrected_Ext_pc1_3,
                                   "LabLib" = p_norm_Plt_corrected_LabLib_pc1_3,
                                   "Length" = p_norm_Plt_corrected_Len_pc1_3,
                                   "Tissue" = p_norm_Plt_corrected_Tissue_pc1_3)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_nrm_plt_pc1_pc3_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC1PC3/PCA_NORMALIZED_Platform",plotIt,"pc1_pc3_Counts.svg"),plot = pca_nrm_plt_pc1_pc3_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC1PC3/PCA_NORMALIZED_Platform",plotIt,"pc1_pc3_Counts.svg"),plot = pca_nrm_plt_pc1_pc3_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```

### PC1 and PC4
```{r,fig.width=22,fig.height=32}
p_norm_Plt_corrected_Lab_pc1_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Lab",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 4)
p_norm_Plt_corrected_Plt_pc1_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Platform",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 4)
p_norm_Plt_corrected_Lib_pc1_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Library",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 4)
p_norm_Plt_corrected_Ext_pc1_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Extract",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 4)
p_norm_Plt_corrected_LabLib_pc1_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "LabLib",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 4)
p_norm_Plt_corrected_Len_pc1_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Length",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 4)
p_norm_Plt_corrected_Tissue_pc1_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Tissue",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 1,pcb = 4)

#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC1_PC4.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC1_PC4.svg"),
#ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC1_PC4.svg"),
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC1_PC4.svg"),

       dpi = 320,
       device = "svg",
       plot = grid.arrange(p_norm_Plt_corrected_Lab_pc1_4,
                           p_norm_Plt_corrected_Plt_pc1_4,
                           p_norm_Plt_corrected_Lib_pc1_4,
                           p_norm_Plt_corrected_Ext_pc1_4,
                           p_norm_Plt_corrected_LabLib_pc1_4,
                           p_norm_Plt_corrected_Len_pc1_4,
                           p_norm_Plt_corrected_Tissue_pc1_4,ncol = 2),
       width = 22,
       height = 32)
```

```{r}
#dir.create(recursive = TRUE,path = "D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC1PC4")
pca_nrm_plt_pc1_pc4_list <- list("Platform" = p_norm_Plt_corrected_Plt_pc1_4,
                                   "Lab" = p_norm_Plt_corrected_Lab_pc1_4,
                                   "Library" = p_norm_Plt_corrected_Lib_pc1_4,
                                   "Extraction" = p_norm_Plt_corrected_Ext_pc1_4,
                                   "LabLib" = p_norm_Plt_corrected_LabLib_pc1_4,
                                   "Length" = p_norm_Plt_corrected_Len_pc1_4,
                                   "Tissue" = p_norm_Plt_corrected_Tissue_pc1_4)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_nrm_plt_pc1_pc4_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC1PC4/PCA_NORMALIZED_Platform",plotIt,"pc1_pc4_Counts.svg"),plot = pca_nrm_plt_pc1_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC1PC4/PCA_NORMALIZED_Platform",plotIt,"pc1_pc4_Counts.svg"),plot = pca_nrm_plt_pc1_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)

})
```
### PC3 and PC4

```{r,fig.width=22,fig.height=32}
# PCA function to add the labels from s4c

p_norm_Plt_corrected_Lab_pc3_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Lab",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 3,pcb = 4)
p_norm_Plt_corrected_Plt_pc3_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Platform",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 3,pcb = 4)
p_norm_Plt_corrected_Lib_pc3_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Library",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 3,pcb = 4)
p_norm_Plt_corrected_Ext_pc3_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Extract",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 3,pcb = 4)
p_norm_Plt_corrected_LabLib_pc3_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "LabLib",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 3,pcb = 4)
p_norm_Plt_corrected_Len_pc3_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Length",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 3,pcb = 4)
p_norm_Plt_corrected_Tissue_pc3_4 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Tissue",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 3,pcb = 4)

#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC3_PC4.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC3_PC4.svg"),
# ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC3PC4PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC3_PC4.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC3PC4PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC3_PC4.svg"),
       dpi = 320,
       device = "svg",
       plot = grid.arrange(p_norm_Plt_corrected_Lab_pc3_4,
                           p_norm_Plt_corrected_Plt_pc3_4,
                           p_norm_Plt_corrected_Lib_pc3_4,
                           p_norm_Plt_corrected_Ext_pc3_4,
                           p_norm_Plt_corrected_LabLib_pc3_4,
                           p_norm_Plt_corrected_Len_pc3_4,
                           p_norm_Plt_corrected_Tissue_pc3_4,ncol = 2),
       width = 22,
       height = 32)
```

```{r}
#dir.create(recursive = TRUE,path = "D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC3PC4")
pca_nrm_plt_pc3_pc4_list <- list("Platform" = p_norm_Plt_corrected_Plt_pc3_4,
                                 "Lab" = p_norm_Plt_corrected_Lab_pc3_4,
                                 "Library" = p_norm_Plt_corrected_Lib_pc3_4,
                                 "Extraction" = p_norm_Plt_corrected_Ext_pc3_4,
                                 "LabLib" = p_norm_Plt_corrected_LabLib_pc3_4,
                                 "Length" = p_norm_Plt_corrected_Len_pc3_4,
                                 "Tissue" = p_norm_Plt_corrected_Tissue_pc3_4)
                                

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_nrm_plt_pc3_pc4_list),function(plotIt){
  # ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC3PC4/PCA_NORMALIZED_Platform",plotIt,"pc3_pc4_Counts.svg"),plot = pca_nrm_plt_pc3_pc4_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC3PC4/PCA_NORMALIZED_Platform",plotIt,"pc3_pc4_Counts.svg"),
         plot = pca_nrm_plt_pc3_pc4_list[[plotIt]],
         device = "svg",
         dpi = 320,
         width = 12,
         height = 10)
  })
```


### PC4 and PC5

```{r,fig.width=22,fig.height=32}
# PCA function to add the labels from s4c

p_norm_Plt_corrected_Lab_pc4_5 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Lab",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 4,pcb = 5)
p_norm_Plt_corrected_Plt_pc4_5 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Platform",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 4,pcb = 5)
p_norm_Plt_corrected_Lib_pc4_5 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Library",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 4,pcb = 5)
p_norm_Plt_corrected_Ext_pc4_5 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Extract",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 4,pcb = 5)
p_norm_Plt_corrected_LabLib_pc4_5 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "LabLib",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 4,pcb = 5)
p_norm_Plt_corrected_Len_pc4_5 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Length",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 4,pcb = 5)
p_norm_Plt_corrected_Tissue_pc4_5 <- pca_var_plot_func(df = normalized_corrected_pcas_Platform_df,var = "Tissue",subtitle = paste0("Normalized ",modell_name), percentVar_obj = percentVar_normalized_corrected_Platform,pca = 4,pcb = 5)

#ggsave(filename = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/",modell_name,"/Graphs/PCA_DESeq2_Normalizations_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("C:/Users/marius/Documents/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC4_PC5.svg"),
#ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/PCA/NORMALIZED/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC4_PC5.svg"),
#ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC4PC5/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC4_PC5.svg"),
ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC4PC5/PCA_DESeq2_Normalized_",modell_name,"_batch_corrected_Counts_PC4_PC5.svg"),
       dpi = 320,
       device = "svg",
       plot = grid.arrange(p_norm_Plt_corrected_Lab_pc4_5,
                           p_norm_Plt_corrected_Plt_pc4_5,
                           p_norm_Plt_corrected_Lib_pc4_5,
                           p_norm_Plt_corrected_Ext_pc4_5,
                           p_norm_Plt_corrected_LabLib_pc4_5,
                           p_norm_Plt_corrected_Len_pc4_5,
                           p_norm_Plt_corrected_Tissue_pc4_5,ncol = 2),
       width = 22,
       height = 32)
```



```{r}
#dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC4PC5")
pca_nrm_plt_pc4_pc5_list <- list("Platform" = p_norm_Plt_corrected_Plt_pc4_5,
                                 "Lab" = p_norm_Plt_corrected_Lab_pc4_5,
                                 "Library" = p_norm_Plt_corrected_Lib_pc4_5,
                                 "Extraction" = p_norm_Plt_corrected_Ext_pc4_5,
                                 "LabLib" = p_norm_Plt_corrected_LabLib_pc4_5,
                                 "Length" = p_norm_Plt_corrected_Len_pc4_5,
                                 "Tissue" = p_norm_Plt_corrected_Tissue_pc4_5)

#dir.create(recursive = TRUE,"D:/PhD/Writing/Graphs_f/PCA/RAW")
lapply(names(pca_nrm_plt_pc4_pc5_list),function(plotIt){
#  ggsave(filename = paste0("D:/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC4PC5/PCA_NORMALIZED_Platform",plotIt,"pc4_pc5_Counts.svg"),plot = pca_nrm_plt_pc4_pc5_list[[plotIt]],device = "svg",dpi = 320,width = 12,height = 10)
  ggsave(filename = paste0("/media/marius/Samsung_T5/PhD/Writing/Graphs_f/PCA/NORMALIZED/Platform/PC4PC5/PCA_NORMALIZED_Platform",plotIt,"pc4_pc5_Counts.svg"),
         plot = pca_nrm_plt_pc4_pc5_list[[plotIt]],
         device = "svg",
         dpi = 320,
         width = 12,
         height = 10)

})
```


### Annotate DESeq2
Create ensembl annotate with the species and the data wanted.

```{r}
#BiocManager::install("plotly",force = TRUE)
#BiocManager::install("httr",force = TRUE)
#require("httr")
library("openssl")
library("curl")

#Check the ensembl gene ids of all the samples are the same
all.equal(rownames(lists_platform_batch_corrected_deseq_condition$a),rownames(lists_platform_batch_corrected_deseq_condition$a))

#Load ensembl
ensembl=useEnsembl(biomart = "ENSEMBL_MART_ENSEMBL", dataset = "drerio_gene_ensembl",mirror = "www")

#Add the attributes to the list of DEG values DESeq2 and edgeR


enriched_list_rownames <- biomaRt::getBM(c("ensembl_gene_id","entrezgene_id","external_gene_name","zfin_id_symbol","description"),mart = ensembl)

#enriched_list_rownames <- getBM(attributes = c("entrezgene_id","ensembl_gene_id","external_gene_name","zfin_id_symbol","description"),                                filters = "ensembl_gene_id",
                                #values=                                  rownames(lists_platform_batch_corrected_deseq_condition$a),
                                #uniqueRows = TRUE,                                mart = ensembl)#,'external_gene_name','hgnc_symbol','description'),)#"hsapiens_homolog_associated_gene_name","hsapiens_homolog_ensembl_gene","mmusculus_homolog_associated_gene_name","mmusculus_homolog_ensembl_gene"),##filters = "mirbase_id",#values = mylist_res_cleaned_005_3p_5p[[x]],
       
#enriched_list_rownames[which(duplicated(enriched_list_rownames$ensembl_gene_id)),]
enriched_list_rownames
```

```{r}
#Platform
lists_platform_batch_corrected_deseq_condition_ann <- lapply(names(lists_platform_batch_corrected_deseq_condition), function(fn)
  merge(x = as.data.frame(lists_platform_batch_corrected_deseq_condition[[fn]]),
        y = enriched_list_rownames,
        by.x="row.names",
        by.y="ensembl_gene_id"))
#names(lists_platform_batch_corrected_deseq_condition_ann) <- letters[1:length(lists_platform_batch_corrected_deseq_condition_ann)]
names(lists_platform_batch_corrected_deseq_condition_ann) <-lapply(lists_platform_batch_corrected_deseq_condition,function(r) paste(strsplit(mcols(r)$description[3]," ")[[1]][4:6],collapse = "_"))

# lapply(names(lists_platform_batch_corrected_deseq_condition_ann),function(x)
#   write.table(lists_platform_batch_corrected_deseq_condition_ann[[x]],paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/Platform/Contrasts_Results/",x,"_ann_Platform.txt"),sep = "\t",row.names = TRUE,col.names = TRUE))

lapply(names(lists_platform_batch_corrected_deseq_condition_ann),function(x)
  write.table(lists_platform_batch_corrected_deseq_condition_ann[[x]],
              # paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/Platform/Contrasts_Results/",x,"_ann_Platform.txt"),
              paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/Platform/Contrasts_Results/",x,"_ann_Platform.txt"),
              sep = "\t",
              row.names = TRUE,
              col.names = TRUE))
```

### Annotate edgeR
```{r}
# lists_library_batch_corrected_edger_condition <- lapply(names(lists_library_batch_corrected_edger_condition),function(fn) lists_library_batch_corrected_edger_condition[[fn]][,-c(2:39)])
# 
# lists_length_batch_corrected_edger_condition <- lapply(names(lists_length_batch_corrected_edger_condition),function(fn) lists_length_batch_corrected_edger_condition[[fn]][,-c(2:39)])

lists_platform_batch_corrected_edger_condition$mylist_res <- lapply(names(lists_platform_batch_corrected_edger_condition$mylist_res),function(fn) lists_platform_batch_corrected_edger_condition$mylist_res[[fn]][,-c(2:39)])

#Platform
names(lists_platform_batch_corrected_edger_condition$mylist_res) <- letters[1:length(lists_platform_batch_corrected_edger_condition$mylist_res)]
lists_platform_batch_corrected_edger_condition_ann <- lapply(names(lists_platform_batch_corrected_edger_condition$mylist_res), function(fn)
  merge(x = as.data.frame(lists_platform_batch_corrected_edger_condition$mylist_res[[fn]]),
        y = enriched_list_rownames,
        by.x="row.names",
        by.y="ensembl_gene_id"))
names(lists_platform_batch_corrected_edger_condition_ann) <- lapply(lists_platform_batch_corrected_edger_condition$mylist_res, function(r) gsub(x = r[["contrastName"]][1],pattern = "s4c\\$Condition",replacement = ""))


lapply(names(lists_platform_batch_corrected_edger_condition_ann),function(x)
  write.table(x = lists_platform_batch_corrected_edger_condition_ann[[x]],
              # file = paste0("D:/PhD/Projects/prsa/BatchCorrected/edgeR/Platform/Contrasts_Results/",x,"_ann_Platform.txt"),
              # file = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/Platform/Contrasts_Results/",x,"_ann_Platform.txt"),
              file = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/edgeR/Platform/Contrasts_Results/",x,"_ann_Platform.txt"),
              sep = "\t",
              row.names = TRUE,
              col.names = TRUE))
```

```{r}
lists_platform_batch_corrected_edger_condition_ann$Sham_vs_Uninjured |> mutate(convLogFC = logFC*(-1)) |> mutate(convlogCPM = logCPM*(-1)) |> write.table("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/edgeR/Platform/Contrasts_Results/Uninjured_vs_Sham_ann_Platform.txt",row.names = FALSE,col.names = TRUE,sep = "\t")
```

### Filtered DEG, padj < 0.05 and logFC |1|


###### DESeq2 filtered data.
```{r}

#Filter the data
#names(lists_platform_batch_corrected_deseq_condition_ann_fil) <- names(lists_platform_batch_corrected_deseq_condition_ann)
name_deseqs <- names(lists_platform_batch_corrected_edger_condition_ann)
length(name_deseqs)
name_deseqs[10] <- "Uninjured_vs_Sham"

#lapply(name_deseqs,function(x) x)

lists_platform_batch_corrected_deseq_condition_ann_fil <- lapply(name_deseqs,function(x)
  subset(x = lists_platform_batch_corrected_deseq_condition_ann[[x]],padj < 0.05 & abs(log2FoldChange) > 1))


names(lists_platform_batch_corrected_deseq_condition_ann_fil) <- name_deseqs#names(lists_platform_batch_corrected_edger_condition_ann)

#Save the data
lapply(names(lists_platform_batch_corrected_deseq_condition_ann_fil),function(x){
  write.table(x = lists_platform_batch_corrected_deseq_condition_ann_fil[[x]],
              # file = paste0("D:/PhD/Projects/prsa/BatchCorrected/DESeq2/Platform/Contrasts_Results/",x,"_ann_filtered_Platform.txt"),
              # file = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/DESeq2/Platform/Contrasts_Results/",x,"_ann_filtered_Platform.txt"),
              file = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/DESeq2/Platform/Contrasts_Results/",x,"_ann_filtered_Platform.txt"),
              sep = "\t",
              row.names = FALSE,
              col.names = TRUE)})


#names(lists_platform_batch_corrected_deseq_condition_ann_fil) <- names(lists_platform_batch_corrected_deseq_condition_ann)
names(lists_platform_batch_corrected_deseq_condition_ann_fil)
```


###### edgeR filtered data.
```{r}
#Filter the data
lists_platform_batch_corrected_edger_condition_ann_fil <- lapply(names(lists_platform_batch_corrected_edger_condition_ann),function(x)
  subset(x = lists_platform_batch_corrected_edger_condition_ann[[x]],FDR < 0.05 & abs(logFC) > 1))

names(lists_platform_batch_corrected_edger_condition_ann_fil) <- names(lists_platform_batch_corrected_edger_condition_ann)
#Save the data
lapply(names(lists_platform_batch_corrected_edger_condition_ann),function(x)
  write.table(x = lists_platform_batch_corrected_edger_condition_ann_fil[[x]],
              # file = paste0("D:/PhD/Projects/prsa/BatchCorrected/edgeR/Platform/Contrasts_Results/",x,"_ann_filtered_Platform.txt"),
              file = paste0("/media/marius/Samsung_T5/PhD/Projects/prsa/BatchCorrected/edgeR/Platform/Contrasts_Results/",x,"_ann_filtered_Platform.txt"),
              # file = paste0("/home/marius/Documents/Projects/prsa/BatchCorrected/edgeR/Platform/Contrasts_Results/",x,"_ann_filtered_Platform.txt"),
              sep = "\t",
              row.names = FALSE,
              col.names = TRUE))

#names(lists_platform_batch_corrected_edger_condition_ann_fil) <- names(lists_platform_batch_corrected_edger_condition_ann)
names(lists_platform_batch_corrected_edger_condition_ann_fil)

lists_platform_batch_corrected_edger_condition_ann_fil$Sham_vs_Uninjured
lists_platform_batch_corrected_edger_condition_ann_fil$Uninjured_vs_Sham <- lists_platform_batch_corrected_edger_condition_ann_fil$Sham_vs_Uninjured
#lists_platform_batch_corrected_edger_condition_ann_fil$Sham_vs_Uninjured
lists_platform_batch_corrected_edger_condition_ann_fil$Uninjured_vs_Sham <- lists_platform_batch_corrected_edger_condition_ann_fil$Uninjured_vs_Sham  |> mutate(logFC = logFC * -1)
lists_platform_batch_corrected_edger_condition_ann_fil$Uninjured_vs_Sham$contrastName <- c("s4c$ConditionUninjured_vs_s4c$ConditionSham")

# lists_platform_batch_corrected_edger_condition_ann_fil$Uninjured_vs_Sham$contrastName <- gsub(lists_platform_batch_corrected_edger_condition_ann_fil$Uninjured_vs_Sham$contrastName,
#                                                                                               pattern = "s4c$ConditionSham_vs_s4c$ConditionUninjured",
#                                                                                               replacement = "s4c$ConditionUninjured_vs_s4c$ConditionSham")
      
```


### COMPARE edgeR and DESeq2

```{r}
#  lapply(names(lists_platform_batch_corrected_deseq_condition_ann_fil), function(n) head(lists_platform_batch_corrected_edger_condition_ann_fil[[n]][2]))
# lapply(names(lists_platform_batch_corrected_edger_condition_ann_fil), function(n) n)
#ven_plot <- list()
#lists_platform_batch_corrected_edger_condition_ann_fil

#lapply(seq_along(0:length(lists_platform_batch_corrected_deseq_condition_ann_fil)),function(n) n)
#lapply(seq_along(0:length(lists_platform_batch_corrected_edger_condition_ann_fil)),function(n) n)

#lists_platform_batch_corrected_edger_condition_ann_fil[["Ablated_vs_Sham"]][[1]]
#lists_platform_batch_corrected_edger_condition_ann_fil[[1]][[1]]
# check for the name change in uninjured vs sham as before was reversed.
#ven_plot <- lapply(seq_along(1:length(lists_platform_batch_corrected_deseq_condition_ann_fil)),
ven_plot <- lapply(names(lists_platform_batch_corrected_deseq_condition_ann_fil),
                   function(n){
                     myCol <- c("blue","yellow")#"Pastel2")#Dark2
                     #ven_plot[[length(ven_plot)+1]] <- 
                     VennDiagram::venn.diagram(
                       x =list(lists_platform_batch_corrected_edger_condition_ann_fil[[n]][[1]],
                               lists_platform_batch_corrected_deseq_condition_ann_fil[[n]][[1]]),
                       filename = NULL,
                       units = "cm",
                       category.names = c(paste0(n,"_edgeR \nvs\n ",n,"_DESeq2")),
                       fill = myCol)})


lapply(seq(1:length(ven_plot)),function(n){
  grid.newpage()
  grid_plot_ven = grid.draw(ven_plot[[n]])})


lapply(seq(1:length(ven_plot)),function(n){
  ggsave(device = "svg",dpi = 300,
         plot = ven_plot[[n]],
         filename =paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/EdgeR_vs_DESeq2_results_comparissons_",names(lists_platform_batch_corrected_edger_condition_ann_fil)[[n]],"edgeR_vs",names(lists_platform_batch_corrected_deseq_condition_ann_fil)[[n]],"_DESeq2.svg"),
  width = 12,
  height = 8)
  })

```



### Volcanoes Function  deseq2

```{r}
volcano_plot_function <- function(df_cond,lg2fc_p,log2fc_n,p_adj,title_,colordown,colorup,down_e,up_e,top_rows_to_select_genes){
  
  df_cond_1 <- df_cond |>
    mutate(Regulation = ifelse(log2FoldChange > lg2fc_p & padj < p_adj,"UP",
                               ifelse(log2FoldChange < log2fc_n & padj < p_adj,"DOWN","NO"))) |>
    mutate(Regulation = ifelse(is.na(Regulation),"NO",Regulation)) |>
    mutate(HS = ifelse(padj > (df_cond |> arrange(padj) |> slice_head(n = top_rows_to_select_genes)  |> slice_tail(n=1) |> pull(padj)),"NO","YES"))
  
  #mutate(Selected_Gene_Names = filter(log2FoldChange > 10 & padj < 1e-20)[,8]) |> 
  p1 <- ggplot(data = df_cond_1,aes(x = log2FoldChange,y = -log10(padj),col = Regulation)) +
    #geom_point() +
    #geom_jitter(width = 0.01,height = 0.01,size=3) +
    geom_jitter(position = position_jitter(height = 0.0000001, width = 0.0000001,seed = 123456),size=1) + # + xlim(0,0.5) + ylim(0,0.5)width = 0.01, 
    scale_color_manual(labels = c(down_e,"NO",up_e),
                       values = c(colordown,
                                  "black",
                                  colorup)) +
    geom_vline(xintercept=c(-1, 1), col="red") +
    geom_hline(yintercept=-log10(0.05), col="red") + 
    labs(title = title_) +
    xlim(-30,30) +
    ylim(0,160) +
    theme(panel.grid = element_blank(),
          panel.background = element_rect(fill = "transparent"),
          panel.border = element_rect(fill = "transparent"),
          legend.title = element_text(size=25),
          legend.text = element_text(size=25),
          axis.text = element_text(size=25),
          axis.title = element_text(size=25,face="bold")) +
    #theme_minimal() +
    #geom_text_repel(data = df_cond_1,label = filter(df_cond_1$HS == "YES") |> pull(zfin_id_symbol))
    geom_text_repel(nudge_y = 3.15, #nudge_x = 0.91,
                    max.overlaps = top_rows_to_select_genes+1,
                    direction = "x",segment.size = 0.8,
                    data = df_cond_1,label = ifelse(df_cond_1$HS == "YES",as.character(df_cond_1$external_gene_name),""),
                    color="black",size=12) +
    geom_point(position = position_jitter(height = 0.0000001, width = 0.0000001,seed = 123456),#width = 0.00001
               pch=21,size=1.1,color="black",
               data = df_cond_1 |> filter(HS == "YES"),
               aes(x = log2FoldChange,y = -log10(padj))) +
    geom_point(position = position_jitter(height = 0.0000001, width = 0.0000001,seed = 123456),#width = 0.00001
               pch=21,size=1.2,color="black",
               data = df_cond_1 |> filter(HS == "YES"),
               aes(x = log2FoldChange,y = -log10(padj)))
  return(list(plot=p1,data_volc = p1$data))}
```


### DESeq2 vs Sham
```{r,fig.height=52,fig.width=36}

ablated_volcano <- volcano_plot_function(df_cond = lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Sham,
                                         lg2fc_p = 1,log2fc_n = -1,
                                         p_adj = 0.05,
                                         top_rows_to_select_genes = 0,
                                         title_= "DESeq2 Ablated vs Sham",
                                         colordown = "#999999",down_e ="Sham", up_e ="Ablated",
                                         colorup = "#E69F00")

amputated_volcano <- volcano_plot_function(df_cond=lists_platform_batch_corrected_deseq_condition_ann$Amputation_vs_Sham,
                                           lg2fc_p = 1,log2fc_n = -1,
                                           p_adj = 0.05,
                                           top_rows_to_select_genes = 0,
                                           title_ = "DESeq2 Amputated vs Sham",
                                           colordown = "#999999",down_e ="Sham",
                                           up_e ="Amputated",colorup = "#CC79A7")


cryoinjured_volcano <- volcano_plot_function(df_cond=lists_platform_batch_corrected_deseq_condition_ann$Cryoinjury_vs_Sham,
                                             lg2fc_p = 1,log2fc_n = -1,
                                             p_adj = 0.05,
                                             top_rows_to_select_genes = 0,
                                             title_ = "DESeq2 Cryoinjured vs Sham",
                                             colordown = "#999999",down_e ="Sham",
                                             up_e ="Cryoinjured",colorup = "#0072B2")

uninjured_volcano <- volcano_plot_function(df_cond=lists_platform_batch_corrected_deseq_condition_ann$Uninjured_vs_Sham,
                                           lg2fc_p = 1,log2fc_n = -1,
                                           p_adj = 0.05,
                                           top_rows_to_select_genes = 0,
                                           title_ = "DESeq2  Uninjured vs Sham",
                                           colordown = "#999999",down_e ="Sham",
                                           up_e ="Uninjured",colorup = "#009E73")

#dir.create(recursive = TRUE,path = "/home/marius/Documents/Projects/prsa/03_Graphs/Volcanos/DESeq2")
#dir.create(path = "D:/PhD/Writing/Graphs_f/VOLCANOES/DESeq2",recursive = TRUE)
#ggsave(filename = "/home/marius/Documents/Projects/prsa/03_Graphs/Volcanos/DESeq2/DESeq2_Conditions_vs_Sham.svg",
#ggsave(filename = "D:/PhD/Projects/Graphs/VOLCANOES/DESeq2_Conditions_vs_Sham_2.svg",
#ggsave(filename = "/media/marius/Samsung_T5/PhD/Projects/Graphs/VOLCANOES/DESeq2_Conditions_vs_Sham_2.svg",
#ggsave(filename = "D:/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Conditions_vs_Sham_2.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Conditions_vs_Sham_2.svg",
       device = "svg",
       limitsize = FALSE,
       width = 36,
       height = 52,
       dpi = 320,
       plot = grid.arrange(ablated_volcano$plot,
                           amputated_volcano$plot,
                           cryoinjured_volcano$plot,
                           uninjured_volcano$plot,nrow=2))

grid.arrange(ablated_volcano$plot,amputated_volcano$plot,cryoinjured_volcano$plot,uninjured_volcano$plot,nrow=2)
#ggsave(filename = "D:/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Ablated_vs_Sham_volc.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Ablated_vs_Sham_volc.svg",
       device = "svg",
       limitsize = FALSE,
       width = 210,
       height = 290,
       dpi = 320,
       units = "mm",
       plot = ablated_volcano$plot)
#ggsave(filename = "D:/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Amputated_vs_Sham_volc.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Amputated_vs_Sham_volc.svg",
       device = "svg",
       limitsize = FALSE,
       width = 210,
       height = 290,
       dpi = 320,
       units = "mm",
       plot = amputated_volcano$plot)
#ggsave(filename = "D:/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Cryoinjured_vs_Sham_volc.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Cryoinjured_vs_Sham_volc.svg",
       device = "svg",
       limitsize = FALSE,
       width = 210,
       height = 290,
       dpi = 320,
       units = "mm",
       plot = cryoinjured_volcano$plot)
#ggsave(filename = "D:/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Uninjured_vs_Sham_volc.svg",
ggsave(filename = "/media/marius/Samsung_T5/PhD/Writing/Graphs_f/VOLCANOES/DESeq2/DESeq2_Uninjured_vs_Sham_volc.svg",
       device = "svg",
       limitsize = FALSE,
       width = 210,
       height = 290,
       units = "mm",
       dpi = 320,
       plot = uninjured_volcano$plot)

#grid.arrange(ablated_volcano$plot)
```




### Venn Danio rerio before convert
```{r,fig.width=11,fig.height=8}
v_Sham_deseq_dr_entrez <- VennDiagram::venn.diagram(x = list(lists_platform_batch_corrected_deseq_condition_ann_fil$Amputation_vs_Sham |> filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                           lists_platform_batch_corrected_deseq_condition_ann_fil$Ablated_vs_Sham %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                           lists_platform_batch_corrected_deseq_condition_ann_fil$Uninjured_vs_Sham %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id)),
                                                    filename = NULL,
                                                    units = "cm",
                                                    category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                    fill = myCol)
# 


grid.newpage()
grid.draw(v_Sham_deseq_dr_entrez)

ggsave(device = "svg",dpi = 300,plot = v_Sham_deseq_dr_entrez,
       filename=paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/Danio_rerio_entrez_DEG_padj005_l2fc1_DESeq2.svg"),
  width = 12,
  height = 8)

```
```{r,fig.width=11,fig.height=8}
v_AblUninj_deseq_dr_entrez <- VennDiagram::venn.diagram(x = list(lists_platform_batch_corrected_deseq_condition_ann_fil$Amputation_vs_Sham |> filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                           lists_platform_batch_corrected_deseq_condition_ann_fil$Ablated_vs_Uninjured %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                           lists_platform_batch_corrected_deseq_condition_ann_fil$Uninjured_vs_Sham %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id)),
                                                    filename = NULL,
                                                    units = "cm",
                                                    category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Uninjured","Uninjured_vs_Sham"),
                                                    fill = myCol_3)
# 


grid.newpage()
grid.draw(v_AblUninj_deseq_dr_entrez)

ggsave(device = "svg",dpi = 300,plot = v_AblUninj_deseq_dr_entrez,
       filename=paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/Danio_rerio_entrez_DEG_padj005_l2fc1_Abla_vs_Uninj_DESeq2.svg"),
  width = 12,
  height = 8)

```
```{r,fig.width=11,fig.height=8}
v_Sham_deseq_dr_ensembl <- VennDiagram::venn.diagram(x = list(lists_platform_batch_corrected_deseq_condition_ann_fil$Amputation_vs_Sham |> 
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham |> 
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Ablated_vs_Sham |>
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Uninjured_vs_Sham |> 
                                                               pull(Row.names)),
                                                    filename = NULL,
                                                    units = "cm",
                                                    category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                    fill = myCol)
# 


grid.newpage()
grid.draw(v_Sham_deseq_dr_ensembl)

ggsave(device = "svg",dpi = 300,plot = v_Sham_deseq_dr_ensembl,
       filename=paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/Danio_rerio_ensembl_DEG_padj005_l2fc1_DESeq2.svg"),
  width = 12,
  height = 8)
```
```{r,fig.width=11,fig.height=8}
v_AblaUninj_deseq_dr_ensembl <- VennDiagram::venn.diagram(x = list(lists_platform_batch_corrected_deseq_condition_ann_fil$Amputation_vs_Sham |> 
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham |> 
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Ablated_vs_Uninjured |>
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Uninjured_vs_Sham |> 
                                                               pull(Row.names)),
                                                    filename = NULL,
                                                    units = "cm",
                                                    category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Uninjured","Uninjured_vs_Sham"),
                                                    fill = myCol_3)
# 


grid.newpage()
grid.draw(v_AblaUninj_deseq_dr_ensembl)

ggsave(device = "svg",dpi = 300,plot = v_AblaUninj_deseq_dr_ensembl,
       filename=paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/Danio_rerio_ensembl_DEG_padj005_l2fc1_Ablated_vs_Uninjured_DESeq2.svg"),
  width = 12,
  height = 8)
```


```{r,fig.width=12,fig.height=18}
ablated_uninj_volcano <- volcano_plot_function(df_cond = lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Uninjured,
                                         lg2fc_p = 1,log2fc_n = -1,
                                         p_adj = 0.05,
                                         top_rows_to_select_genes = 0,
                                         title_= "DESeq2 Ablated vs Uninjured",
                                         colordown = "#009E73",down_e ="Uninjured",
                                         up_e ="Ablated",colorup = "#E69F00")
ablated_uninj_volcano

```


### Ablated vs Uninjured Danio rerio Venn
```{r}
v_Sham_Uninjured_deseq_dr_ensembl <- VennDiagram::venn.diagram(x = list(lists_platform_batch_corrected_deseq_condition_ann_fil$Amputation_vs_Sham |> 
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham |> 
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Ablated_vs_Uninjured |>
                                                               pull(Row.names),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Uninjured_vs_Sham |> 
                                                               pull(Row.names)),
                                                    filename = NULL,
                                                    units = "cm",
                                                    category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Uninjured","Uninjured_vs_Sham"),
                                                    fill = myCol_3)
# 


grid.newpage()
grid.draw(v_Sham_Uninjured_deseq_dr_ensembl)

ggsave(device = "svg",dpi = 300,plot = v_Sham_Uninjured_deseq_dr_ensembl,
       filename=paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/Danio_rerio_Ablated_vs_Uninjured_ensembl_DEG_padj005_l2fc1_DESeq2_v2.svg"),
  width = 12,
  height = 8)
```

```{r,fig.width=11,fig.height=8}
v_Sham_Uninjured_deseq_dr_entrez <- VennDiagram::venn.diagram(x = list(lists_platform_batch_corrected_deseq_condition_ann_fil$Amputation_vs_Sham |> filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                             lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                           lists_platform_batch_corrected_deseq_condition_ann_fil$Ablated_vs_Uninjured %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id),
                                                           lists_platform_batch_corrected_deseq_condition_ann_fil$Uninjured_vs_Sham %>% filter(!is.na(entrezgene_id)) %>% pull(entrezgene_id)),
                                                    filename = NULL,
                                                    units = "cm",
                                                    category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Uninjured","Uninjured_vs_Sham"),
                                                    fill = myCol_3)
# 


grid.newpage()
grid.draw(v_Sham_Uninjured_deseq_dr_entrez)

ggsave(device = "svg",dpi = 300,plot = v_Sham_Uninjured_deseq_dr_entrez,
       filename=paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/Danio_rerio_Ablated_Uninjured_entrez_DEG_padj005_l2fc1_DESeq2_v2.svg"),
  width = 12,
  height = 8)

```

### Convert and clean d2m homolog perc id

```{r}
# lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham %>% distinct(Row.names,.keep_all = TRUE)
# lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham %>% group_by(Row.names) %>% filter(n() == 1)# %>%  count()
# lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham %>% group_by(Row.names) %>%
#   filter(n() > 1) %>%
#   distinct(Row.names,.keep_all = TRUE)
#m <- lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham %>% distinct(Row.names)

convertDanio2Mouse <- function(m){
  library("biomaRt")
  library("dplyr")
  library("curl")

  #biomaRt::listEnsembl()
  #Use ensembl
  #ensembl <- useEnsembl("ENSEMBL_MART_ENSEMBL")
  #biomaRt::listAttributes(ensembl)
  #biomaRt::listDatasets(ensembl)

  dre = useEnsembl(biomart = "ensembl", dataset = "drerio_gene_ensembl", host = "https://dec2021.archive.ensembl.org/")#mirror = "useast") # query organism (from this organism)
  mmus = useEnsembl(biomart = "ensembl", dataset = "mmusculus_gene_ensembl", host = "https://dec2021.archive.ensembl.org/")#mirror = "useast") # target organism (convert genes to this organism)
  
  danio_ensembl_id_attributes <-c("ensembl_gene_id", "mmusculus_homolog_perc_id")#,# attributes to query from danio database (query organism) 
                                  #"mmusculus_homolog_goc_score") #orthology score to mouse from danio side
                                                            
  danio_zfin_id_attributes <-c("ensembl_gene_id", "entrezgene_id", "zfin_id_symbol","description") # attributes to convert ensembl id to zfin id 
  
  
  mmus_attributes <- c("mgi_symbol", "ensembl_gene_id", "entrezgene_id") # attributes that you want from mouse database
  
  #make a converted dataframe with ensemble ids and mouse orthology scores 
  genes_conv_ensembl = getLDS(values = m ,
                              filters = "ensembl_gene_id",  # query using ensemble ids
                              attributes = danio_ensembl_id_attributes, # query these attributes from danio mart
                              mart = dre, # use danio mart
                              attributesL = mmus_attributes, # convert to attributes of mouse mart 
                              martL = mmus,# use mouse mart
                              uniqueRows=T) 
  
  
  #make dataframe for dre ensembl id and zfin
  genes_conv_zfin = getBM(attributes = danio_zfin_id_attributes,
                          filters = "ensembl_gene_id",
                          values = m,
                          mart = dre)
  
  
  #merge conversion df with zfin id df
  gene_merge = merge(genes_conv_ensembl, genes_conv_zfin, by.x= "Gene.stable.ID",by.y="ensembl_gene_id", all.y=TRUE)
  
  # rename column names to readable format
  colnames(gene_merge)[colnames(gene_merge)== "Gene.stable.ID"] <- "Ensembl_Danio"
  colnames(gene_merge)[colnames(gene_merge)== "X.id..target.Mouse.gene.identical.to.query.gene"] <- "Mouse_homology_percentage"
  colnames(gene_merge)[colnames(gene_merge)== "NCBI.gene..formerly.Entrezgene..ID"] <- "EntrezGeneID_Mouse"
  colnames(gene_merge)[colnames(gene_merge)== "entrezgene_id"] <- "EntrezGeneID_Danio"
  colnames(gene_merge)[colnames(gene_merge)== "Gene.stable.ID.1"] <- "Ensembl_Mouse"
  colnames(gene_merge)[colnames(gene_merge)== "MGI.symbol"] <- "MGI_Symbol"
  colnames(gene_merge)[colnames(gene_merge)== "zfin_id_symbol"] <- "Zfin_Symbol"
  colnames(gene_merge)[colnames(gene_merge)== "description"] <- "Description"
  
  
  #select the top ortholog with max homology percentage 
  
  gene_merge <- gene_merge %>% group_by(Ensembl_Danio) %>% # group by ensembl id (since ensemble ids are the one used for query)
    arrange(desc(Mouse_homology_percentage),.by_group = TRUE) %>% #arrange each of them by homology score then by GOC score
    mutate(num_orthologues=n()) %>% #write in a column how many orthologues are possible
    # dplyr::slice(n= 1,.preserve= TRUE ) %>% #doesn't work alphabetically(don't know how it works)
    distinct(Ensembl_Danio,.keep_all = TRUE) %>%   # if two orthologues have same homology percentage and goc then choose the top one
    mutate(unique_ortho_check=n()) %>%
    arrange(desc(num_orthologues)) # arrange by the genes which have maximum orthogues
  cat('If worked try again switching old host=\"https://www.ensembl.org\" and mirror = \"useast\" ')
  return(gene_merge)
}
```

```{r}
#All the universe detected in this fish.
#Apply function
Ablated_vs_Sham_mouse_enriched_all_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Sham$Row.names)
Amputation_vs_Sham_mouse_enriched_all_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann$Amputation_vs_Sham$Row.names)
Cryonjury_vs_Sham_mouse_enriched_all_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann$Cryoinjury_vs_Sham$Row.names)
Uninjured_vs_Sham_mouse_enriched_all_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann$Uninjured_vs_Sham$Row.names)
Ablated_vs_Uninjured_mouse_enriched_all_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Uninjured$Row.names)


#Filtered DEG genes only.
#Apply function
Ablated_vs_Sham_mouse_enriched_fil_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann_fil$Ablated_vs_Sham$Row.names)
Amputation_vs_Sham_mouse_enriched_fil_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann_fil$Amputation_vs_Sham$Row.names)
Cryonjury_vs_Sham_mouse_enriched_fil_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann_fil$Cryoinjury_vs_Sham$Row.names)
Uninjured_vs_Sham_mouse_enriched_fil_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann_fil$Uninjured_vs_Sham$Row.names)
Ablated_vs_Uninjured_mouse_enriched_fil_clean <- convertDanio2Mouse(m = lists_platform_batch_corrected_deseq_condition_ann_fil$Ablated_vs_Uninjured$Row.names)
```


### Numbers of converted ensembles gene ids to entrezgeneid

#### ALL
```{r}
Ablated_vs_Uninjured_mouse_enriched_all_clean[is.na(Ablated_vs_Uninjured_mouse_enriched_all_clean$Ensembl_Danio),]
dim(Ablated_vs_Uninjured_mouse_enriched_all_clean[is.na(Ablated_vs_Uninjured_mouse_enriched_all_clean$EntrezGeneID_Danio),])
length(Ablated_vs_Uninjured_mouse_enriched_all_clean$Zfin_Symbol[Ablated_vs_Uninjured_mouse_enriched_all_clean$Zfin_Symbol == ""])

Ablated_vs_Uninjured_mouse_enriched_all_clean |> filter()


```




```{r}
#Get normalized RLD transformed data.
#DESeq2_platform_normalized_counts_rld <- DESeq2::rlog(dds_Platform,blind = FALSE)
DESeq2_platform_normalized_counts <- DESeq2::counts(dds1_Platform,normalized=TRUE)
dim(DESeq2::counts(dds1_Platform,normalized=TRUE))
head(DESeq2::counts(dds1_Platform,normalized=TRUE))
head(lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Sham)
```

### UMAP platform corrected data, filtered by expression values of the counts.

```{r}
# if (!("umap" %in% installed.packages())) {
#   # Install umap package
#   BiocManager::install("umap", update = FALSE)
# }
```

```{r}
#Raw log distribution.
dim(DESeq2_platform_normalized_counts)
hist(log(DESeq2_platform_normalized_counts),xlim=c(0,20),breaks = 100,col='tomato3',border=F)
```


 
```{r,fig.width=6,fig.height=7}
umap_norm_batch_platform_filtered_0 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 0, subtitle = "filtered 0")
umap_norm_batch_platform_filtered_3 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 3, subtitle = "filtered 3")
umap_norm_batch_platform_filtered_7 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 7, subtitle = "filtered 7")
umap_norm_batch_platform_filtered_10 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 10, subtitle = "filtered 10")
umap_norm_batch_platform_filtered_20 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 20, subtitle = "filtered 20")
umap_norm_batch_platform_filtered_40 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 40, subtitle = "filtered 40")
umap_norm_batch_platform_filtered_80 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 80, subtitle = "filtered 80")
umap_norm_batch_platform_filtered_100 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 100, subtitle = "filtered 100")
umap_norm_batch_platform_filtered_300 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 300, subtitle = "filtered 300")
umap_norm_batch_platform_filtered_500 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 500, subtitle = "filtered 500")
umap_norm_batch_platform_filtered_1000 <- umap_func_by_filtered_expression_normalized(df = DESeq2_platform_normalized_counts,n = 1000, subtitle = "filtered 1000")
```
```{r,fig.width=18,fig.height=36}
grid.newpage()
grid.arrange(umap_norm_batch_platform_filtered_0$umap_plot,
             umap_norm_batch_platform_filtered_3$umap_plot,
             umap_norm_batch_platform_filtered_7$umap_plot,
             umap_norm_batch_platform_filtered_10$umap_plot,
             umap_norm_batch_platform_filtered_20$umap_plot,
             umap_norm_batch_platform_filtered_40$umap_plot,
             umap_norm_batch_platform_filtered_80$umap_plot,
             umap_norm_batch_platform_filtered_100$umap_plot,
             umap_norm_batch_platform_filtered_300$umap_plot,
             umap_norm_batch_platform_filtered_500$umap_plot,
             umap_norm_batch_platform_filtered_1000$umap_plot
             )
```

```{r,fig.width=8,fig.height=8}
umap_norm_batch_platform_filtered_0$umap_plot
```
 

```{r,echo=TRUE,fig.height=32, fig.width=35}
# PCA function to add the labels from s4c
pca_var_plot_func <- function(df,var,subtitle,percentVar_obj){
  n <- ggplot(data = df,aes(x = PC1,
                            y = PC2,
                            color = s4c[["Condition"]],
                            shape = s4c[[var]]))
  
  #n <- n + geom_point(size=3)
  n <- n + theme(panel.grid = element_blank(),
                 panel.border = element_rect(fill = "transparent"),
                 panel.background = element_rect(fill = "transparent"),
                 legend.title = element_text(size = 15),
                 legend.text = element_text(size = 12))
  
  n <- n + labs(title="PCA of RNA-seq",
                color = "Condition",
                shape = var,
                subtitle = paste0(subtitle," DESeq2 Normalized")) +
    xlab(paste0("PC",substr("PC1",start = 3,stop = 3),": ",
                round(percentVar_obj[as.numeric(substr("PC1",start = 3,stop = 3))] * 100), "% variance")) +
    ylab(paste0("PC",substr("PC2",start = 3,stop = 3),": ",
                round(percentVar_obj[as.numeric(substr("PC2",start = 3,stop = 3))] * 100), "% variance"))
  #n <- n + scale_color_brewer(palette = "Paired") #Accent #Dark2
  n <- n + scale_color_manual(values=cbf_1)
  n <- n + scale_shape_manual(values=shape_values_lab)
  #n <- n + geom_jitter(width = 0.01,height = 0.01,size=3)
  n <- n + geom_jitter(position = position_jitter(width = 0.01,height = 0.01,seed = 123456),size=8)# + xlim(0,0.5) + ylim(0,0.5)
  n}
```


```{r}
dim(lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Sham)
lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Sham |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6)

```


### Merge to counts and stats
#### all
```{r}
Ablated_vs_Sham_mouse_enriched_all_clean_m <- merge(x = Ablated_vs_Sham_mouse_enriched_all_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Sham[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Ablated_vs_Sham_mouse_enriched_all_clean_mm <- merge(x = Ablated_vs_Sham_mouse_enriched_all_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)

Amputation_vs_Sham_mouse_enriched_all_clean_m <- merge(x = Amputation_vs_Sham_mouse_enriched_all_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Amputation_vs_Sham[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Amputation_vs_Sham_mouse_enriched_all_clean_mm <- merge(x = Amputation_vs_Sham_mouse_enriched_all_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)

Cryonjury_vs_Sham_mouse_enriched_all_clean_m <- merge(x = Cryonjury_vs_Sham_mouse_enriched_all_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Cryoinjury_vs_Sham[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio", 
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Cryonjury_vs_Sham_mouse_enriched_all_clean_mm <- merge(x = Cryonjury_vs_Sham_mouse_enriched_all_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)

Uninjured_vs_Sham_mouse_enriched_all_clean_m <- merge(x = Uninjured_vs_Sham_mouse_enriched_all_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Uninjured_vs_Sham[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Uninjured_vs_Sham_mouse_enriched_all_clean_mm <- merge(x = Uninjured_vs_Sham_mouse_enriched_all_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)


Ablated_vs_Uninjured_mouse_enriched_all_clean_m <- merge(x = Ablated_vs_Uninjured_mouse_enriched_all_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Uninjured[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Ablated_vs_Uninjured_mouse_enriched_all_clean_mm <- merge(x = Ablated_vs_Uninjured_mouse_enriched_all_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)


```

```{r}
paste(dim(Ablated_vs_Sham_mouse_enriched_fil_clean),
      dim(Amputation_vs_Sham_mouse_enriched_fil_clean),
      dim(Cryonjury_vs_Sham_mouse_enriched_fil_clean),
      dim(Uninjured_vs_Sham_mouse_enriched_fil_clean),
      sep = "  ")
#head(Uninjured_vs_Sham_mouse_enriched_fil_clean)
```

#### filtered DEG 
```{r}
Ablated_vs_Sham_mouse_enriched_fil_clean_m <- merge(x = Ablated_vs_Sham_mouse_enriched_fil_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Sham[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Ablated_vs_Sham_mouse_enriched_fil_clean_mm <- merge(x = Ablated_vs_Sham_mouse_enriched_fil_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)

Amputation_vs_Sham_mouse_enriched_fil_clean_m <- merge(x = Amputation_vs_Sham_mouse_enriched_fil_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Amputation_vs_Sham[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Amputation_vs_Sham_mouse_enriched_fil_clean_mm <- merge(x = Amputation_vs_Sham_mouse_enriched_fil_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)

Cryonjury_vs_Sham_mouse_enriched_fil_clean_m <- merge(x = Cryonjury_vs_Sham_mouse_enriched_fil_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Cryoinjury_vs_Sham[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)

Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm <- merge(x = Cryonjury_vs_Sham_mouse_enriched_fil_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)

Uninjured_vs_Sham_mouse_enriched_fil_clean_m <- merge(x = Uninjured_vs_Sham_mouse_enriched_fil_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Uninjured_vs_Sham[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Uninjured_vs_Sham_mouse_enriched_fil_clean_mm <- merge(x = Uninjured_vs_Sham_mouse_enriched_fil_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)

Ablated_vs_Uninjured_mouse_enriched_fil_clean_m <- merge(x = Ablated_vs_Uninjured_mouse_enriched_fil_clean,
                                                  y = lists_platform_batch_corrected_deseq_condition_ann$Ablated_vs_Uninjured[,c(1:6)] |> distinct(Row.names,.keep_all = TRUE) |> dplyr::select(1:6),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "Row.names",
                                                  all.x = TRUE)
Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm <- merge(x = Ablated_vs_Uninjured_mouse_enriched_fil_clean_m,
                                                  y = DESeq2::counts(dds1_Platform,normalized=TRUE),
                                                  by.x = "Ensembl_Danio",
                                                  by.y = "row.names",
                                                  all.x = TRUE)


```


```{r}
paste(dim(Ablated_vs_Sham_mouse_enriched_fil_clean_mm),
      dim(Amputation_vs_Sham_mouse_enriched_fil_clean_mm),
      dim(Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm),
      dim(Uninjured_vs_Sham_mouse_enriched_fil_clean_mm),
      dim(Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm),
      sep = "  ")
```


```{r}
#Save data.
#all
#dir.create("D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/")
dir.create("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/")

#write.table(x = Sham_vs_Uninjured_mouse_enriched_clean,file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Uninjured_vs_Sham_mouse_enriched_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)

write.table(x = Uninjured_vs_Sham_mouse_enriched_all_clean_mm,
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Uninjured_vs_Sham_mouse_enriched_all_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)



#write.table(x = Ablated_vs_Sham_mouse_enriched_clean,file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Ablated_vs_Sham_mouse_enriched_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
# write.table(x = Ablated_vs_Sham_mouse_enriched_all_clean,file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Ablated_vs_Sham_mouse_enriched_all_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
write.table(x = Ablated_vs_Sham_mouse_enriched_all_clean_mm,
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Ablated_vs_Sham_mouse_enriched_all_clean.txt",
            sep = "\t",
            row.names = FALSE,
            col.names = TRUE,
            quote = FALSE)
#write.table(x = Cryonjury_vs_Sham_mouse_enriched_clean,file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Cryonjury_vs_Sham_mouse_enriched_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
# write.table(x = Cryonjury_vs_Sham_mouse_enriched_all_clean,file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Cryonjury_vs_Sham_mouse_enriched_all_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)

write.table(x = Cryonjury_vs_Sham_mouse_enriched_all_clean_mm,
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Cryonjury_vs_Sham_mouse_enriched_all_clean.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)


#write.table(x = Amputation_vs_Sham_mouse_enriched_clean,file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Amputation_vs_Sham_mouse_enriched_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
# write.table(x = Amputation_vs_Sham_mouse_enriched_all_clean,file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Amputation_vs_Sham_mouse_enriched_all_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)

write.table(x = Amputation_vs_Sham_mouse_enriched_all_clean_mm,
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Amputation_vs_Sham_mouse_enriched_all_clean.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)

write.table(x = Ablated_vs_Uninjured_mouse_enriched_all_clean_mm,
            #file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Ablated__vs_Uninjured_mouse_enriched_all_clean.txt",
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Ablated__vs_Uninjured_mouse_enriched_all_clean.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)






#filtered
# write.table(x = Uninjured_vs_Sham_mouse_enriched_fil_clean,file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Uninjured_vs_Sham_mouse_enriched_fil_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
write.table(x = Uninjured_vs_Sham_mouse_enriched_fil_clean_mm,
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Uninjured_vs_Sham_mouse_enriched_fil_clean.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)



#write.table(x = Ablated_vs_Sham_mouse_enriched_clean,file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Ablated_vs_Sham_mouse_enriched_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
# write.table(x = Ablated_vs_Sham_mouse_enriched_fil_clean,file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Ablated_vs_Sham_mouse_enriched_fil_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)

write.table(x = Ablated_vs_Sham_mouse_enriched_fil_clean_mm,
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Ablated_vs_Sham_mouse_enriched_fil_clean.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)



#write.table(x = Cryonjury_vs_Sham_mouse_enriched_clean,file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Cryonjury_vs_Sham_mouse_enriched_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
# write.table(x = Cryonjury_vs_Sham_mouse_enriched_fil_clean,file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Cryonjury_vs_Sham_mouse_enriched_fil_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)

write.table(x = Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm,
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Cryonjury_vs_Sham_mouse_enriched_fil_clean.txt",sep = "\t",
            row.names = FALSE,col.names = TRUE,quote = FALSE)


#write.table(x = Amputation_vs_Sham_mouse_enriched_clean,file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Amputation_vs_Sham_mouse_enriched_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
# write.table(x = Amputation_vs_Sham_mouse_enriched_fil_clean,file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Amputation_vs_Sham_mouse_enriched_fil_clean.txt",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
write.table(x = Amputation_vs_Sham_mouse_enriched_fil_clean_mm,
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Amputation_vs_Sham_mouse_enriched_fil_clean.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)

write.table(x = Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm,
            #file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/dr2mmus/Ablated__vs_Uninjured_mouse_enriched_fil_clean.txt",
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/dr2mmus/Ablated__vs_Uninjured_mouse_enriched_fil_clean.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)



```

### Shams to check the numbers
#### Ensembl Mouse

```{r}
cbf_1 <- c("#E69F00","#CC79A7","#0072B2","#999999","#009E73")
cbf_1_1 <- c("#E69F00","#CC79A7","#0072B2","#999999","#009E73","#D55E00")
myCol <- c("#CC79A7","#0072B2","#E69F00","#009E73")
myCol_2 <- c("#CC79A7","#0072B2","#E69F00","#009E73","#D55E00")

myCol <- c("#CC79A7","#0072B2","#E69F00","#999999")
```


```{r,fig.width=12,fig.height=9}
v_Sham_mus_deseq_mmh_c <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_all_clean %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Cryonjury_vs_Sham_mouse_enriched_all_clean %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Ablated_vs_Sham_mouse_enriched_all_clean %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Uninjured_vs_Sham_mouse_enriched_all_clean %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse)),
                                                  filename = NULL,
                                                  units = "cm",
                                                  category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                  fill = myCol)
grid.newpage()
grid.draw(v_Sham_mus_deseq_mmh_c)
```


```{r,fig.width=12,fig.height=9}
#post merge
v_Sham_mus_deseq_mmh_c <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_all_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Cryonjury_vs_Sham_mouse_enriched_all_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Ablated_vs_Sham_mouse_enriched_all_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Uninjured_vs_Sham_mouse_enriched_all_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse)),
                                                  filename = NULL,
                                                  units = "cm",
                                                  category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                  fill = myCol)
grid.newpage()
grid.draw(v_Sham_mus_deseq_mmh_c)
```

```{r,fig.width=12,fig.height=9}
v_Sham_mus_deseq_mmh_c <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Cryonjury_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Ablated_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Uninjured_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse)),
                                                  filename = NULL,
                                                  units = "cm",
                                                  category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                  fill = myCol)
grid.newpage()
grid.draw(v_Sham_mus_deseq_mmh_c)
```

```{r,fig.width=12,fig.height=9}
#post merge
v_Sham_mus_deseq_mmh_c <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Ablated_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse)),
                                                  filename = NULL,
                                                  units = "cm",
                                                  category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                  fill = myCol)


#ggven_ensembl <- grid.draw(v_Sham_mus_deseq_mmh_c)
#ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS.png"),
       #device = "svg",       plot = ggven_ensembl,       width = 12,       height = 9)
grid.newpage()
grid.draw(v_Sham_mus_deseq_mmh_c)
```
```{r,fig.width=12,fig.height=9}
#post merge
myCol_3 <- c("#CC79A7","#0072B2","#009E73","#999999")

v_Sham_uninjured_mus_deseq_fil_mm_ensem <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse)),
                                                  filename = NULL,
                                                  units = "cm",
                                                  category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Uninjured","Uninjured_vs_Sham"),
                                                  fill = myCol_3)


#ggven_ensembl <- grid.draw(v_Sham_mus_deseq_mmh_c)
#ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS.svg"),
#ggsave(filename = paste0("D:/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS.png"),
       #device = "svg",       plot = ggven_ensembl,       width = 12,       height = 9)
grid.newpage()
grid.draw(v_Sham_uninjured_mus_deseq_fil_mm_ensem)
```

```{r,fig.width=19,fig.height=13}
myCol_2 <- c("#CC79A7","#0072B2","#E69F00","#999999","#009E73")
#post merge
v_Sham_mus_deseq_mmh_c_5 <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Ablated_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse),
                                                           Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm %>% filter(!is.na(Ensembl_Mouse)) %>% pull(Ensembl_Mouse)),
                                                  filename = NULL,
                                                  units = "cm",
                                                  category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham","Ablated_vs_Uninjured"),
                                                  fill = myCol_2)

# 
# ggven_ensembl_5 <- grid.draw(v_Sham_mus_deseq_mmh_c_5)
# #ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS.svg"),
# ggsave(filename = paste0("D:/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS_5.png"),
#        #device = "svg",
#        plot = ggven_ensembl_5,
#        width = 19,
#        height = 13)

grid.newpage()
grid.draw(v_Sham_mus_deseq_mmh_c_5)

```
#### Entrez for mouse
### ablated_vs_sham and ablated_vs_uninjured
```{r,fig.width=19,fig.height=13}
v_Sham_and_Uninj_mus_deseq_mm_entrez <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                           Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                           Ablated_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                           Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                           Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)),
                                                      filename = NULL,
                                                      units = "cm",
                                                      category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham","Ablated_vs_Uninjured"),
                                                      fill = myCol_2)
# 
# ggven_ensembl_2 <- grid.draw(v_Sham_mus_deseq_mmh_c_2)
# #ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS.svg"),
# ggsave(filename = paste0("D:/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS_4_abl_uninj.png"),
#        #device = "svg",
#        plot = ggven_ensembl_2,
#        width = 12,
#        height = 9)

grid.newpage()
grid.draw(v_Sham_and_Uninj_mus_deseq_mm_entrez)

```

```{r,fig.width=12,fig.height=9}
v_Sham_mus_deseq_enntrez_c <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_all_clean$EntrezGeneID_Mouse[!is.na(Amputation_vs_Sham_mouse_enriched_all_clean$EntrezGeneID_Mouse)],
  Cryonjury_vs_Sham_mouse_enriched_all_clean$EntrezGeneID_Mouse[!is.na(Cryonjury_vs_Sham_mouse_enriched_all_clean$EntrezGeneID_Mouse)],
  Ablated_vs_Sham_mouse_enriched_all_clean$EntrezGeneID_Mouse[!is.na(Ablated_vs_Sham_mouse_enriched_all_clean$EntrezGeneID_Mouse)],
  Uninjured_vs_Sham_mouse_enriched_all_clean$EntrezGeneID_Mouse[!is.na(Uninjured_vs_Sham_mouse_enriched_all_clean$EntrezGeneID_Mouse)]),
  filename = NULL,
  units = "cm",
  category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
  fill=myCol)
grid.newpage()

grid.draw(v_Sham_mus_deseq_enntrez_c)
```


```{r,fig.width=12,fig.height=9}
v_Sham_mus_deseq_enntrez_c <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_all_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                 Cryonjury_vs_Sham_mouse_enriched_all_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                 Ablated_vs_Sham_mouse_enriched_all_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                 Uninjured_vs_Sham_mouse_enriched_all_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)),
                                                        filename = NULL,
                                                        units = "cm",
                                                        category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                        fill=myCol)
grid.newpage()
grid.draw(v_Sham_mus_deseq_enntrez_c)
```




```{r,fig.width=12,fig.height=9}
v_Sham_mus_deseq_enntrez_c <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                               Cryonjury_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                               Ablated_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                               Uninjured_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)),
                                                        filename = NULL,
                                                        units = "cm",
                                                        category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                        fill=myCol)
# 
# ggven_entrez <- grid.draw(v_Sham_mus_deseq_enntrez_c)
# #ggsave(filename =paste0("/media/marius/Samsung_T5/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENSEMBL_ID_MERGED_COUNTS_STATS.svg"),
# ggsave(filename = paste0("D:/PhD/Projects/Graphs/VENNS/FILTERED_CLEAN_ENTREZ_ID_MERGED_COUNTS_STATS.svg"),
#        #device = "svg",
#        plot = ggven_entrez,
#        width = 12,
#        height = 9)
# 


grid.newpage()
grid.draw(v_Sham_mus_deseq_enntrez_c)
```


### Venns, ablated vs Sham and Uninjured
```{r,fig.width=12,fig.height=9}
#post merge
v_Sham_mus_deseq_enntrez_c <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                 Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                 Ablated_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                 Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)),
                                                        filename = NULL,
                                                        units = "cm",
                                                        category.names = c("Amputation_vs_Sham","Cryoinjury_vs_Sham","Ablated_vs_Sham","Uninjured_vs_Sham"),
                                                        fill=myCol)

grid.newpage()
grid.draw(v_Sham_mus_deseq_enntrez_c)
```

```{r,fig.width=12,fig.height=9}
myCol_3 <- c("#CC79A7","#0072B2","#009E73","#999999")

v_Sham_uninjured_mus_deseq_fil_mm_entrez <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                               Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                               Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                                                               Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)),
                                                                      filename = NULL,
                                                                      units = "cm",
                                                                      category.names = c("Amputation\nvs\nSham","Cryoinjury\nvs\nSham","Ablated\nvs\nUninjured","Uninjured\nvs\nSham"),
                                                                      fill = myCol_3)


grid.newpage()
grid.draw(v_Sham_uninjured_mus_deseq_fil_mm_entrez)
```

```{r,fig.width=10,fig.height=7}
myCol_3 <- c("#CC79A7","#0072B2","#009E73","#999999")

v_Sham_uninjured_mus_deseq_fil_mm_entrez_down <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean_mm  |>
                                                                                      filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |>
                                                                                      pull(EntrezGeneID_Mouse),
                                                                                    Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> 
                                                                                      filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |>
                                                                                      pull(EntrezGeneID_Mouse),
                                                                                    Ablated_vs_Sham_mouse_enriched_fil_clean_mm |> 
                                                                                      filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |>
                                                                                      pull(EntrezGeneID_Mouse),
                                                                                    Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |> 
                                                                                      filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |>
                                                                                      pull(EntrezGeneID_Mouse)),
                                                                           filename = NULL,
                                                                           units = "cm",
                                                                           category.names = c("Amputation\nvs\nSham_down",
                                                                                              "Cryoinjury\nvs\nSham_down",
                                                                                              "Ablated\nvs\nSham_down",
                                                                                              "Uninjured\nvs\nSham_down"),
                                                                           fill = myCol)
grid.newpage()
grid.draw(v_Sham_uninjured_mus_deseq_fil_mm_entrez_down)
```
```{r}
#setdiff(setdiff(setdiff(d3,d2),d4),d1)
```

### anno bp function setdiff
```{r,fig.width=15,fig.height=27,echo=TRUE}
#require(enrichplot)
set.seed(123456)
anno_bp <- function(pvalCutOff,d1,d2,d3,d4,padjCO,size_paths_gg=9,d1.1,d2.2,d3.3,d4.4,orgdb,software){ #comparedvs
  #Select datasets to enrich
  c1 <- enrichGO(setdiff(setdiff(setdiff(d1,d2),d3),d4),
                 #universe = as.character(Amputation_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
                 OrgDb = orgdb,
                 ont = "BP",
                 pAdjustMethod = "BH",
                 pvalueCutoff  = pvalCutOff,
                 readable = T)
  # c1.1 <- simplify(c1)
  # g1 <- emapplot(c1.1)
  c2 <- enrichGO(setdiff(setdiff(setdiff(d2,d1),d3),d4),
                 #universe = as.character(Ablated_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
                 OrgDb = orgdb,
                 ont = "BP",
                 pAdjustMethod = "BH",
                 pvalueCutoff  = pvalCutOff,
                 readable = T)
  #g2 <- goplot(c2)
  c3 <- enrichGO(setdiff(setdiff(setdiff(d3,d2),d4),d1),
                 #universe = as.character(Uninjured_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
                 OrgDb = orgdb,
                 ont = "BP",
                 pAdjustMethod = "BH",
                 pvalueCutoff  = pvalCutOff,
                 readable = T)
  #g3 <- goplot(c3)
  c4 <- enrichGO(setdiff(setdiff(setdiff(d4,d3),d2),d1),
                 #universe = as.character(Cryonjury_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
                 OrgDb = orgdb,
                 ont = "BP",
                 pAdjustMethod = "BH" ,
                 pvalueCutoff  = pvalCutOff,
                 readable = T)
  #g4 <- goplot(c4)

  c1@result$Group <- d1.1
  c2@result$Group <- d2.2
  c3@result$Group <- d3.3
  c4@result$Group <- d4.4
  
  c5 <- rbind(c1@result[c1@result$p.adjust < padjCO ,],
              c2@result[c2@result$p.adjust < padjCO ,],
              c3@result[c3@result$p.adjust < padjCO ,],
              c4@result[c4@result$p.adjust < padjCO ,])
  c5$BgRatio <- log(as.numeric(unlist(lapply(c5$BgRatio,function(n) gsub(x = n,
                                                                         pattern = paste0("/",strsplit(c5$BgRatio[1],split = "/")[[1]][2]),
                                                                         replacement = ""))))/as.numeric(strsplit(c5$BgRatio[1],split = "/")[[1]][2]))*-1
  
  cat("Plotting\n")
  cat("Plotting Heatmap!\n")
  g <- ggplot(data = c5,aes(y = Group,
                            x = Description,
                            fill = BgRatio)) +
    geom_tile()+#aes(colour=ifelse(duplicated(Description),"green","red"))) + #,stat = "identity"
    # geom_point(aes(label = p.adjust,
    #                size = -log10(p.adjust))) +
    scale_fill_gradient2(low = "blue",mid = "azure3",high = "red",midpoint = mean(c5$BgRatio)) +
    labs(title = paste0(software," set Diff Conditons unique IDs \ncomparred to all others Annotations"),
         subtitle = paste0("padj < ",padjCO),
         fill = "Gene ratio",
         #colour = "Duplication",
         x = "GO:BP",
         y = "Conditions") +
    #geom_point(data = pick(~Species == "setosa"), colour = "red") +
    geom_vline(xintercept = c5$Description[which(duplicated(c5$Description))],color="black", linetype="twodash", size=1,alpha=0.25) +
    coord_flip() +
    theme(axis.text.x = element_text(angle = 90, hjust= 0.01,size = size_paths_gg))
  
  return(g)
}
```

```{r}
sham_bp_Mmus_05_v2_down <- anno_bp(d1 = Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>  filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |> pull(EntrezGeneID_Mouse),
                                  d2 = Ablated_vs_Sham_mouse_enriched_fil_clean_mm |>  filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |> pull(EntrezGeneID_Mouse),
                                  #d2 = Ablated_vs_Uninjured_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                  d3 = Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |> filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |> pull(EntrezGeneID_Mouse),
                                  d4 = Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |> pull(EntrezGeneID_Mouse),
                                  pvalCutOff = 0.05,
                                  padjCO = 0.05,
                                  #comparedvs = "Sham",
                                  d1.1 = "Amputation",
                                  d2.2 = "Ablated",
                                  d3.3 = "Uninjured",
                                  d4.4 = "Cryoinjury",
                                  size_paths_gg = 8,orgdb = MmOrgDb,software = "DESeq2")
#sham_bp_Mmus_05_nv_c$data
unique(sham_bp_Mmus_05_v2_down$data$Group)


sham_bp_Mmus_05_v2_down$data |> filter(Group == "Cryoinjury")
sham_bp_Mmus_05_v2_down$data |> filter(Group == "Amputation")
sham_bp_Mmus_05_v2_down$data |> filter(Group == "Uninjured")
#sham_bp_Mmus_05_nv_c$data |> filter(Group == "Ablation")
sham_bp_Mmus_05_v2_down$data |> filter(Group == "Ablated")


dim(sham_bp_Mmus_05_v2_down$data)
#write.table(x =sham_bp_Mmus_05_nv_c$data, file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Mmus_table_exact_extended_enrichment_vs_Sham_cleaned.tsv",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
#dir.create(path = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/GOBP")
#dir.create(path = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/GOBP")
write.table(x = sham_bp_Mmus_05_v2_down$data,
            #file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment.txt",
            #file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment_v2_abl_sham_v4.txt",
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment_v2_abl_sham_downregulated_1.txt",
            sep = "\t",
            row.names = FALSE,
            col.names = TRUE,
            quote = FALSE)
```

```{r}
length(Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> 
         filter(!is.na(EntrezGeneID_Mouse)) |>
         filter(log2FoldChange < -1) |>
         pull(EntrezGeneID_Mouse))


Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |>
  filter(!is.na(EntrezGeneID_Mouse)) |>
  filter(log2FoldChange < -1) |>
  pull(EntrezGeneID_Mouse)

Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |>
  filter(!is.na(EntrezGeneID_Mouse)) |>
  filter(log2FoldChange > 1) |>
  pull(EntrezGeneID_Mouse)

```


```{r,fig.width=10,fig.height=7}
v_Sham_uninjured_mus_deseq_fil_mm_entrez_up <- VennDiagram::venn.diagram(x = list(Amputation_vs_Sham_mouse_enriched_fil_clean_mm  |>
                                                                                    filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |>
                                                                                    pull(EntrezGeneID_Mouse),
                                                                                  Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> 
                                                                                    filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |>
                                                                                    pull(EntrezGeneID_Mouse),
                                                                                  Ablated_vs_Sham_mouse_enriched_fil_clean_mm |> 
                                                                                    filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |>
                                                                                    pull(EntrezGeneID_Mouse),
                                                                                  Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |> 
                                                                                    filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |>
                                                                                    pull(EntrezGeneID_Mouse)),
                                                                           filename = NULL,
                                                                           units = "cm",
                                                                           category.names = c("Amputation\nvs\nSham_up","Cryoinjury\nvs\nSham_up","Ablated\nvs\nSham_up","Uninjured\nvs\nSham_up"),
                                                                           fill = myCol)

grid.newpage()
grid.draw(v_Sham_uninjured_mus_deseq_fil_mm_entrez_up)
```


```{r}
sham_bp_Mmus_05_v2_up <- anno_bp(d1 = Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>  filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |> pull(EntrezGeneID_Mouse),
                                  d2 = Ablated_vs_Sham_mouse_enriched_fil_clean_mm |>  filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |> pull(EntrezGeneID_Mouse),
                                  #d2 = Ablated_vs_Uninjured_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                  d3 = Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |> filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |> pull(EntrezGeneID_Mouse),
                                  d4 = Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |> pull(EntrezGeneID_Mouse),
                                  pvalCutOff = 0.05,
                                  padjCO = 0.05,
                                  #comparedvs = "Sham",
                                  d1.1 = "Amputation",
                                  d2.2 = "Ablated",
                                  d3.3 = "Uninjured",
                                  d4.4 = "Cryoinjury",
                                  size_paths_gg = 8,orgdb = MmOrgDb,software = "DESeq2")
#sham_bp_Mmus_05_nv_c$data
unique(sham_bp_Mmus_05_v2_up$data$Group)


sham_bp_Mmus_05_v2_up$data |> filter(Group == "Cryoinjury")
sham_bp_Mmus_05_v2_up$data |> filter(Group == "Amputation")
sham_bp_Mmus_05_v2_up$data |> filter(Group == "Uninjured")
#sham_bp_Mmus_05_nv_c$data |> filter(Group == "Ablation")
sham_bp_Mmus_05_v2_up$data |> filter(Group == "Ablated")


dim(sham_bp_Mmus_05_v2_up$data)
#write.table(x =sham_bp_Mmus_05_nv_c$data, file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Mmus_table_exact_extended_enrichment_vs_Sham_cleaned.tsv",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
#dir.create(path = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/GOBP")
#dir.create(path = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/GOBP")
write.table(x = sham_bp_Mmus_05_v2_up$data,
            #file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment.txt",
            #file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment_v2_abl_sham_v4.txt",
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment_v2_abl_sham_upregulated_1.txt",
            sep = "\t",
            row.names = FALSE,
            col.names = TRUE,
            quote = FALSE)
```

### Heart regeneration core
```{r}
heart_regeneration_core_abl_sham <- setdiff(intersect(Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                             intersect(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                                       Ablated_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse))),
                                   intersect(Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                             intersect(Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>%filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                                       intersect(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                                                 Ablated_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)))))

length(heart_regeneration_core_abl_sham)
```



```{r}
heart_regeneration_core_abl_unj <- setdiff(intersect(Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                             intersect(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                                       Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse))),
                                   intersect(Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                             intersect(Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>%filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                                       intersect(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
                                                                 Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)))))

length(heart_regeneration_core_abl_unj)
```


```{r,fig.width=15,fig.height=27,echo=TRUE}
#require(enrichplot)
set.seed(123456)
anno_bp <- function(pvalCutOff,d1,d2,d3,d4,padjCO,size_paths_gg=9,d1.1,d2.2,d3.3,d4.4,orgdb,software){ #comparedvs
  #Select datasets to enrich
  c1 <- enrichGO(setdiff(setdiff(setdiff(d1,d2),d3),d4),
                 #universe = as.character(Amputation_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
                 OrgDb = orgdb,
                 ont = "BP",
                 pAdjustMethod = "BH",
                 pvalueCutoff  = pvalCutOff,
                 readable = T)
  # c1.1 <- simplify(c1)
  # g1 <- emapplot(c1.1)
  c2 <- enrichGO(setdiff(setdiff(setdiff(d2,d1),d3),d4),
                 #universe = as.character(Ablated_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
                 OrgDb = orgdb,
                 ont = "BP",
                 pAdjustMethod = "BH",
                 pvalueCutoff  = pvalCutOff,
                 readable = T)
  #g2 <- goplot(c2)
  c3 <- enrichGO(setdiff(setdiff(setdiff(d3,d2),d4),d1),
                 #universe = as.character(Uninjured_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
                 OrgDb = orgdb,
                 ont = "BP",
                 pAdjustMethod = "BH",
                 pvalueCutoff  = pvalCutOff,
                 readable = T)
  #g3 <- goplot(c3)
  c4 <- enrichGO(setdiff(setdiff(setdiff(d4,d3),d2),d1),
                 #universe = as.character(Cryonjury_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
                 OrgDb = orgdb,
                 ont = "BP",
                 pAdjustMethod = "BH" ,
                 pvalueCutoff  = pvalCutOff,
                 readable = T)
  #g4 <- goplot(c4)

  c1@result$Group <- d1.1
  c2@result$Group <- d2.2
  c3@result$Group <- d3.3
  c4@result$Group <- d4.4
  
  c5 <- rbind(c1@result[c1@result$p.adjust < padjCO ,],
              c2@result[c2@result$p.adjust < padjCO ,],
              c3@result[c3@result$p.adjust < padjCO ,],
              c4@result[c4@result$p.adjust < padjCO ,])
  c5$BgRatio <- log(as.numeric(unlist(lapply(c5$BgRatio,function(n) gsub(x = n,
                                                                         pattern = paste0("/",strsplit(c5$BgRatio[1],split = "/")[[1]][2]),
                                                                         replacement = ""))))/as.numeric(strsplit(c5$BgRatio[1],split = "/")[[1]][2]))*-1
  
  cat("Plotting\n")
  cat("Plotting Heatmap!\n")
  g <- ggplot(data = c5,aes(y = Group,
                            x = Description,
                            fill = BgRatio)) +
    geom_tile()+#aes(colour=ifelse(duplicated(Description),"green","red"))) + #,stat = "identity"
    # geom_point(aes(label = p.adjust,
    #                size = -log10(p.adjust))) +
    scale_fill_gradient2(low = "blue",mid = "azure3",high = "red",midpoint = mean(c5$BgRatio)) +
    labs(title = paste0(software," set Diff Conditons unique IDs \ncomparred to all others Annotations"),
         subtitle = paste0("padj < ",padjCO),
         fill = "Gene ratio",
         #colour = "Duplication",
         x = "GO:BP",
         y = "Conditions") +
    #geom_point(data = pick(~Species == "setosa"), colour = "red") +
    geom_vline(xintercept = c5$Description[which(duplicated(c5$Description))],color="black", linetype="twodash", size=1,alpha=0.25) +
    coord_flip() +
    theme(axis.text.x = element_text(angle = 90, hjust= 0.01,size = size_paths_gg))
  
  return(g)
}
```

```{r}
# c4 <- enrichGO(setdiff(setdiff(setdiff(d4,d3),d2),d1),
#                  #universe = as.character(Cryonjury_vs_Sham_mouse_enriched_all_clean |> filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)),
#                  OrgDb = orgdb,
#                  ont = "BP",
#                  pAdjustMethod = "BH" ,
#                  pvalueCutoff  = pvalCutOff,
#                  readable = T)
# 
# c4@result[c4@result$p.adjust < 0.05,]
```

### Start GEM for Enrichment Map
### GOBP Ablated vs Uninjured

```{r}
#################
#   Annotation  #
#################
sham_bp_Mmus_05_nv_c_3 <- anno_bp(d1 = Amputation_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                #d2 = Ablated_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                d2 = Ablated_vs_Uninjured_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                d3 = Uninjured_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                d4 = Cryonjury_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                pvalCutOff = 0.05,
                                padjCO = 0.05,
                                #comparedvs = "Sham",
                                d1.1 = "Amputation",
                                d2.2 = "Ablated",
                                d3.3 = "Uninjured",
                                d4.4 = "Cryoinjury",
                                size_paths_gg = 8,orgdb = MmOrgDb,software = "DESeq2")
#sham_bp_Mmus_05_nv_c$data
unique(sham_bp_Mmus_05_nv_c_3$data$Group)

sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Cryoinjury")
sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Amputation")
sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Uninjured")
#sham_bp_Mmus_05_nv_c$data |> filter(Group == "Ablation")
sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Ablated")


dim(sham_bp_Mmus_05_nv_c_3$data)
#write.table(x =sham_bp_Mmus_05_nv_c$data, file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Mmus_table_exact_extended_enrichment_vs_Sham_cleaned.tsv",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
#dir.create(path = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/GOBP")
#dir.create(path = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/GOBP")
write.table(x = sham_bp_Mmus_05_nv_c_3$data,
            #file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment.txt",
            #file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment_v3_abl_unj_v4.txt",
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment_v3_abl_unj.txt",
            sep = "\t",
            row.names = FALSE,
            col.names = TRUE,
            quote = FALSE)
```

### GOBP Ablated vs Sham
```{r}
#################
#   Annotation  #
#################
sham_bp_Mmus_05_nv_c_2 <- anno_bp(d1 = Amputation_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                  d2 = Ablated_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                  #d2 = Ablated_vs_Uninjured_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                  d3 = Uninjured_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                  d4 = Cryonjury_vs_Sham_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
                                  pvalCutOff = 0.05,
                                  padjCO = 0.05,
                                  #comparedvs = "Sham",
                                  d1.1 = "Amputation",
                                  d2.2 = "Ablated",
                                  d3.3 = "Uninjured",
                                  d4.4 = "Cryoinjury",
                                  size_paths_gg = 8,orgdb = MmOrgDb,software = "DESeq2")
#sham_bp_Mmus_05_nv_c$data
unique(sham_bp_Mmus_05_nv_c_2$data$Group)

sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Cryoinjury")
sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Amputation")
sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Uninjured")
#sham_bp_Mmus_05_nv_c$data |> filter(Group == "Ablation")
sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Ablated")


dim(sham_bp_Mmus_05_nv_c_2$data)
#write.table(x =sham_bp_Mmus_05_nv_c$data, file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/Mmus_table_exact_extended_enrichment_vs_Sham_cleaned.tsv",sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)
#dir.create(path = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/GOBP")
#dir.create(path = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/GOBP")
write.table(x = sham_bp_Mmus_05_nv_c_2$data,
            #file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment.txt",
            #file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment_v2_abl_sham_v4.txt",
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GOBP/Mmus_clean_gobp_enrichment_v2_abl_sham.txt",
            sep = "\t",
            row.names = FALSE,
            col.names = TRUE,
            quote = FALSE)
```

### GOBP Venn's
### Ablated vs Shams GOBP
```{r,fig.height=9,fig.width=12}

# sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Cryoinjury")
# sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Amputation")
# sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Uninjured")
# sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Ablated")


deseq_fil_entrez_goBP_v2 <- VennDiagram::venn.diagram(x = list(sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Amputation") %>% pull(Description),
                                                                               sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Cryoinjury") %>% pull(Description),
                                                                               sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Ablated") %>% pull(Description),
                                                                               sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Uninjured") %>% pull(Description)),
                                                                      filename = NULL,
                                                      units = "cm",
                                                      category.names = c("Amputation\nvs\nSham_GO:BP","Cryoinjury\nvs\nSham_GO:BP","Ablated\nvs\nSham_GO:BP","Uninjured\nvs\nSham_GO:BP"),
                                                      fill = myCol)
grid.newpage()
grid.draw(deseq_fil_entrez_goBP_v2)

```


### Ablated vs Uninjured GOBP
```{r,fig.height=9,fig.width=12}

myCol_3 <- c("#CC79A7","#0072B2","#009E73","#999999")

# sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Cryoinjury")
# sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Amputation")
# sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Uninjured")
# sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Ablated")

deseq_fil_entrez_goBP_v3 <- VennDiagram::venn.diagram(x = list(sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Amputation") %>% pull(Description),
                                                                               sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Cryoinjury") %>% pull(Description),
                                                                               sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Ablated") %>% pull(Description),
                                                                               sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Uninjured") %>% pull(Description)),
                                                                      filename = NULL,
                                                      units = "cm",
                                                      category.names = c("Amputation\nvs\nSham_GO:BP","Cryoinjury\nvs\nSham_GO:BP","Ablated\nvs\nUninjured_GO:BP","Uninjured\nvs\nSham_GO:BP"),
                                                      fill = myCol_3)
grid.newpage()
grid.draw(deseq_fil_entrez_goBP_v3)
```
### Select unique GOBP from each enrichment.

```{r}
#dir.create("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/GEM_files")
create_gem <- function(condition,stats_data_condition,enrichment_data,text_Extra,control){
  #,correct_merge
  library("dplyr")
  library("DESeq2")
  library("SummarizedExperiment")
  library("tibble")
  library("biomaRt")
  library("curl")
  
  #Select the file we want to have as a network
  cat("Starting to wrangle the tables\n")
  
  #For using with GOBP unique it is removed enrichment_data$data
  a <- enrichment_data |>
    dplyr::select(ID,Description,pvalue,p.adjust,Group,geneID) |>
    dplyr::rename(p.Val = pvalue,
                  FDR = p.adjust) |> 
    #mutate(Phenotype = ifelse(log2FoldChange > 0, 1,-1)) |>
    dplyr::mutate(Phenotype = 1) |>
    dplyr::filter(Group == condition) |>
    #filter(Group == condition) |>
    dplyr::mutate(Genes = toupper(noquote(gsub(x = geneID,pattern = "/",",")))) |>
    dplyr::select(-c(Group,geneID))
    

  #write.table(paste0("D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/setDiffConditions/GEM_files/",text_Extra,"_gobp_",condition,"_vs_",control,".txt"),
  #write.table(paste0("D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/setdif_conditions/GEM_files/",text_Extra,"_gobp_",condition,"_vs_",control,".txt"),
  #write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,".txt"),
  a |> 
    # write.table(paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/GOBPUniques/",text_Extra,"_",condition,"_vs_",control,".txt"),
    write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/GOBPUniques/",text_Extra,"_",condition,"_vs_",control,".txt"),
                row.names = FALSE,
                col.names = TRUE,
                sep = "\t",
                quote = FALSE)
  
  #Select the ids from the table.
  #b <- rbind(s4c |> filter(Condition == "Uninjured"),
  b <- rbind(s4c |> filter(Condition == control),
             s4c |> filter(Condition == condition)) |>
    rownames_to_column("ID") |>
    dplyr::select(Condition,ID) |>
    arrange(desc(Condition))
  
  #cat("Getting the RLD DESeq2 normalized counts\n")
  c <- DESeq2_platform_normalized_counts[,colnames(DESeq2_platform_normalized_counts) %in% b$ID]
  
  
  # #Save the data frame for working with it and have it available in R too.
  # cat("Make sure the dataframe using the column to merge are the ENSDARG...\n")
  # d <- merge(x = as.data.frame(c),
  #            #y = as.data.frame(stats_data_condition2),
  #            y = as.data.frame(stats_data_condition),
  #            by.x = "row.names",
  #            by.y = correct_merge) |>
  #   dplyr::select(MGI_Symbol,Description,log2FoldChange,Ensembl_Danio,Ensembl_Mouse,EntrezGeneID_Mouse,Zfin_Symbol,starts_with("SRR")) |>
  #   mutate(Symbols = toupper(MGI_Symbol)) |> #,Phenotype = if_else(log2FoldChange > 0, 1, -1)) |>
  #   dplyr::select(MGI_Symbol,Description_Mm,starts_with("SRR"),Ensembl_Danio,Ensembl_Mouse,EntrezGeneID_Mouse,MGI_Symbol,Zfin_Symbol)#,-c(external_gene_name.y))
  # 
  # 
  cat(paste0("Saving files for ",condition,"\n"))
  # 
  # 
  # #write the data
  # 
  # write.table(x = d,
  #             # file = paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/GEM_files/",text_Extra,"_",condition,"_sham_normalized_counts_expression_symbol_and_sample_name.txt"),
  #             # file = paste0("D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/GEM_files/",text_Extra,"_",condition,"_sham_normalized_counts_expression_symbol_and_sample_name.txt"),
  #             file = paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/GEM_files/",text_Extra,"_",condition,"_sham_normalized_counts_expression_symbol_and_sample_name.txt"),
  #             sep = "\t",
  #             row.names = FALSE,col.names = TRUE,
  #             quote = FALSE)
  # 
  #Change sample nanme to condition name
  e <- stats_data_condition[,c("MGI_Symbol","Description",b$ID)]#,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")]
  #colnames(e) <- c("Symbols","Description",b$ID)#,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")
  colnames(e) <- c("Symbols","Description",b$ID)#,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")
  # e$Ensembl_Danio <- noquote(e$Ensembl_Danio)
  # e$Ensembl_Mouse <- noquote(e$Ensembl_Mouse)
  # e$Zfin_Symbol <- noquote(e$Zfin_Symbol)
  #e$Symbols <- toupper(e$Symbols)
  e$Symbols <- noquote(e$Symbols)
  
  #all_conditions_ID_vs_sham_names$ID
  
  #head(e)
      #t() |>
    # write.table(paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/GEM_files/",text_Extra,"_",condition,"_sham_normalized_counts_expression_symbol_and_condition.txt"),
    #write.table(paste0("D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/setDiffConditions/GEM_files/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
    #write.table(paste0("D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/setdif_conditions/GEM_files/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
    #write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
  e |> 
    # write.table(paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/GOBPUniques/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
  write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/GOBPUniques/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),

                sep = "\t",
                row.names = FALSE,
                col.names = TRUE,
                quote = FALSE)
  
  #Save the classes.
  write.table(x = b$Condition,
              file = paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/GOBPUniques/",text_Extra,"_",condition,"_vs_",control,"_names.cls"),
              # file = paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/GOBPUniques/",text_Extra,"_",condition,"_vs_",control,"_names.cls"),
              sep = "\t",
              quote = FALSE,
              col.names = FALSE,
              row.names = FALSE)
  return(list(go_table_selected_condition = a,
              conditon_table=b,
              data_frame_ready_counts = e))#,data_frame_ready_counts_sample_id = d))
  Sys.sleep(time = 15)
}
```


### Ablated vs Uninjured Unique GOBP
```{r}
require("stringr")
###############################
#get the uniques GOBP elements#
###############################
m1 = sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Amputation")
m3 = sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Cryoinjury")
m2 = sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Ablated")
m4 = sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Uninjured")

amp_setdiff_gobp_v3 <- sham_bp_Mmus_05_nv_c_3$data |>
  filter(Group == "Amputation") |>
  filter(Description %in% setdiff(setdiff(setdiff(m1$Description,m2$Description),m3$Description),m4$Description))

abl_setdiff_gobp_v3 <- sham_bp_Mmus_05_nv_c_3$data |>
  filter(Group == "Ablated") |>
  filter(Description %in% setdiff(setdiff(setdiff(m2$Description,m3$Description),m4$Description),m1$Description))

uninj_setdiff_gobp_v3 <- sham_bp_Mmus_05_nv_c_3$data |>
  filter(Group == "Uninjured") |>
  filter(Description %in% setdiff(setdiff(m4$Description,m2$Description),m1$Description))

######################################################################
#Get the genes from the GOBP and clear them to get the counts values.#
######################################################################

uninj_setdiff_gobp_v3_genes <- data.frame("Genes" = uninj_setdiff_gobp_v3 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist())
#uninj_setdiff_gobp_v3 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist() %>% as.data.frame() %>% dplyr::distinct()
# symbols[[target]]<-unique(unlist(strsplit(as.character(FDR_pathways@result$geneID[i]),split = "\\/")))
uninj_setdiff_gobp_v3_genes <- uninj_setdiff_gobp_v3_genes %>% dplyr::distinct()  

uninj_setdiff_gobp_v3_genes_mm <- Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(MGI_Symbol,.keep_all = TRUE) |>
  filter(MGI_Symbol %in% uninj_setdiff_gobp_v3_genes$Genes)



abl_setdiff_gobp_v3_genes <- data.frame("Genes" = abl_setdiff_gobp_v3 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist())
abl_setdiff_gobp_v3_genes <- abl_setdiff_gobp_v3_genes %>% dplyr::distinct()  
abl_setdiff_gobp_v3_genes_mm <- Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm |>
  distinct(MGI_Symbol,.keep_all = TRUE) |>
  filter(MGI_Symbol %in% abl_setdiff_gobp_v3_genes$Genes)

amp_setdiff_gobp_v3_genes <- data.frame("Genes" = amp_setdiff_gobp_v3 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist())
amp_setdiff_gobp_v3_genes <- amp_setdiff_gobp_v3_genes %>% dplyr::distinct()  
amp_setdiff_gobp_v3_genes_mm <- Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(MGI_Symbol,.keep_all = TRUE) |>
  filter(MGI_Symbol %in% amp_setdiff_gobp_v3_genes$Genes)



amputation_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3 <- create_gem(enrichment_data = amp_setdiff_gobp_v3,
                                                                                 condition = "Amputation",
                                                                                 stats_data_condition = amp_setdiff_gobp_v3_genes_mm,
                                                                                 #correct_merge = "Row.names",
                                                                                 text_Extra = "cleaned_unique_go_bp_v3_",
                                                                                 control = "Sham")

ablated_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3 <- create_gem(enrichment_data = abl_setdiff_gobp_v3,
                                                                              condition = "Ablated",
                                                                              stats_data_condition = abl_setdiff_gobp_v3_genes_mm,
                                                                              #correct_merge = "Row.names",
                                                                              text_Extra = "cleaned_unique_go_bp_v3_",
                                                                              control = "Uninjured")

uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3 <- create_gem(enrichment_data = uninj_setdiff_gobp_v3,
                                                                                condition = "Uninjured",
                                                                                stats_data_condition = uninj_setdiff_gobp_v3_genes_mm,
                                                                                #correct_merge = "Row.names",
                                                                                text_Extra = "cleaned_unique_go_bp_v3_",
                                                                                control = "Sham")

uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3$go_table_selected_condition |> head()
uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3$conditon_table |> head()
uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3$data_frame_ready_counts |> head()
dim(uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3$go_table_selected_condition)
dim(ablated_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3$go_table_selected_condition)
dim(amputation_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v3$go_table_selected_condition)

```



### Ablated vs Sham Unique GOBP
```{r}
require("stringr")
###############################
#get the uniques GOBP elements#
###############################
m1 = sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Amputation")
m3 = sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Cryoinjury")
m2 = sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Ablated")
m4 = sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Uninjured")

amp_setdiff_gobp_v2 <- sham_bp_Mmus_05_nv_c_2$data |>
  filter(Group == "Amputation") |>
  filter(Description %in% setdiff(setdiff(setdiff(m1$Description,m2$Description),m3$Description),m4$Description))

abl_setdiff_gobp_v2 <- sham_bp_Mmus_05_nv_c_2$data |>
  filter(Group == "Ablated") |>
  filter(Description %in% setdiff(setdiff(setdiff(m2$Description,m3$Description),m4$Description),m1$Description))

uninj_setdiff_gobp_v2 <- sham_bp_Mmus_05_nv_c_2$data |>
  filter(Group == "Uninjured") |>
  filter(Description %in% setdiff(setdiff(m4$Description,m2$Description),m1$Description))


######################################################################
#Get the genes from the GOBP and clear them to get the counts values.#
######################################################################

uninj_setdiff_gobp_v2_genes <- data.frame("Genes" = uninj_setdiff_gobp_v2 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist())
#uninj_setdiff_gobp_v3 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist() %>% as.data.frame() %>% dplyr::distinct()
# symbols[[target]]<-unique(unlist(strsplit(as.character(FDR_pathways@result$geneID[i]),split = "\\/")))
uninj_setdiff_gobp_v2_genes <- uninj_setdiff_gobp_v2_genes %>% dplyr::distinct()  

uninj_setdiff_gobp_v2_genes_mm <- Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(MGI_Symbol,.keep_all = TRUE) |>
  filter(MGI_Symbol %in% uninj_setdiff_gobp_v2_genes$Genes)



abl_setdiff_gobp_v2_genes <- data.frame("Genes" = abl_setdiff_gobp_v2 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist())
abl_setdiff_gobp_v2_genes <- abl_setdiff_gobp_v2_genes %>% dplyr::distinct()  
abl_setdiff_gobp_v2_genes_mm <- Ablated_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(MGI_Symbol,.keep_all = TRUE) |>
  filter(MGI_Symbol %in% abl_setdiff_gobp_v2_genes$Genes)

amp_setdiff_gobp_v2_genes <- data.frame("Genes" = amp_setdiff_gobp_v2 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist())
amp_setdiff_gobp_v2_genes <- amp_setdiff_gobp_v2_genes %>% dplyr::distinct()  
amp_setdiff_gobp_v2_genes_mm <- Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(MGI_Symbol,.keep_all = TRUE) |>
  filter(MGI_Symbol %in% amp_setdiff_gobp_v2_genes$Genes)



amputation_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2 <- create_gem(enrichment_data = amp_setdiff_gobp_v2,
                                                                                 condition = "Amputation",
                                                                                 stats_data_condition = amp_setdiff_gobp_v2_genes_mm,
                                                                                 #correct_merge = "Row.names",
                                                                                 text_Extra = "cleaned_unique_go_bp_v2_",
                                                                                 control = "Sham")

ablated_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2 <- create_gem(enrichment_data = abl_setdiff_gobp_v2,
                                                                              condition = "Ablated",
                                                                              stats_data_condition = abl_setdiff_gobp_v2_genes_mm,
                                                                              #correct_merge = "Row.names",
                                                                              text_Extra = "cleaned_unique_go_bp_v2_",
                                                                              control = "Sham")

uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2 <- create_gem(enrichment_data = uninj_setdiff_gobp_v2,
                                                                                condition = "Uninjured",
                                                                                stats_data_condition = uninj_setdiff_gobp_v2_genes_mm,
                                                                                #correct_merge = "Row.names",
                                                                                text_Extra = "cleaned_unique_go_bp_v2_",
                                                                                control = "Sham")

uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2$go_table_selected_condition |> head()
uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2$conditon_table |> head()
uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2$data_frame_ready_counts |> head()
uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2$go_table_selected_condition |> dim()
ablated_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2$go_table_selected_condition |> dim()
amputation_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2$go_table_selected_condition |> dim()
```

### Create GEM files func
```{r}
######################
#   Clean GEM files  #
######################
#dir.create("D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/GEM_files")
#Transform the counts used.
#head(assay(dds_Platform))

#DESeq2_platform_normalized_counts <- DESeq2::counts(dds1_Platform,normalized=TRUE)
#DESeq2_platform_normalized_counts_rld <- DESeq2::rlog(dds_Platform,blind = FALSE)
#head(assay(DESeq2_platform_normalized_counts_rld))

# annotations_orgDb <- AnnotationDbi::select(org.Mm.eg.db, # database
#                                      #keys = Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm$Ensembl_Mouse,  # data to use for retrieval
#                                      columns = c("SYMBOL", "ENTREZID","GENENAME"), # information to retreive for given data
#                                      keytype = "ENSEMBL") 
# 
# length(which(is.na(MmOrgDb$SYMBOL) == FALSE))
```

```{r}
#dir.create("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/GEM_files")
create_gem <- function(condition,stats_data_condition,enrichment_data,text_Extra,control){
  #,correct_merge
  require("dplyr")
  library("DESeq2")
  library("SummarizedExperiment")
  library("tibble")
  library("biomaRt")
  library("curl")
  
  #Select the file we want to have as a network
  cat("Starting to wrangle the tables\n")
  
  #For using with GOBP unique it is removed enrichment_data$data
  a <- enrichment_data$data |>
    dplyr::select(ID,Description,pvalue,p.adjust,Group,geneID) |>
    rename(p.Val = pvalue,
           FDR = p.adjust) |> 
    #mutate(Phenotype = ifelse(log2FoldChange > 0, 1,-1)) |>
    mutate(Phenotype = 1) |>
    filter(Group == condition) |>
    #filter(Group == condition) |>
    mutate(Genes = toupper(noquote(gsub(x = geneID,pattern = "/",",")))) |>
    dplyr::select(-c(Group,geneID))
    

  
  a |>
    # write.table(paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,".txt"),
    write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,".txt"),
              row.names = FALSE,
              col.names = TRUE,
              sep = "\t",
              quote = FALSE)
  #Select the ids from the table.
  #b <- rbind(s4c |> filter(Condition == "Uninjured"),
  b <- rbind(s4c |> filter(Condition == control),
             s4c |> filter(Condition == condition)) |>
    rownames_to_column("ID") |>
    dplyr::select(Condition,ID) |>
    arrange(desc(Condition))
  
  #cat("Getting the RLD DESeq2 normalized counts\n")
  c <- DESeq2_platform_normalized_counts[,colnames(DESeq2_platform_normalized_counts) %in% b$ID]
  
  
  # #Save the data frame for working with it and have it available in R too.
  # cat("Make sure the dataframe using the column to merge are the ENSDARG...\n")
  # d <- merge(x = as.data.frame(c),
  #            #y = as.data.frame(stats_data_condition2),
  #            y = as.data.frame(stats_data_condition),
  #            by.x = "row.names",
  #            by.y = correct_merge) |>
  #   dplyr::select(MGI_Symbol,Description,log2FoldChange,Ensembl_Danio,Ensembl_Mouse,EntrezGeneID_Mouse,Zfin_Symbol,starts_with("SRR")) |>
  #   mutate(Symbols = toupper(MGI_Symbol)) |> #,Phenotype = if_else(log2FoldChange > 0, 1, -1)) |>
  #   dplyr::select(MGI_Symbol,Description_Mm,starts_with("SRR"),Ensembl_Danio,Ensembl_Mouse,EntrezGeneID_Mouse,MGI_Symbol,Zfin_Symbol)#,-c(external_gene_name.y))
  # 
  # 
  cat(paste0("Saving files for ",condition,"\n"))
 
  #Change sample nanme to condition name
  e <- stats_data_condition[,c("MGI_Symbol","Description",b$ID,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")]
  colnames(e) <- c("Symbols","Description",b$Condition,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")
  e$Ensembl_Danio <- noquote(e$Ensembl_Danio)
  e$Ensembl_Mouse <- noquote(e$Ensembl_Mouse)
  e$Zfin_Symbol <- noquote(e$Zfin_Symbol)
  #e$Symbols <- toupper(e$Symbols)
  e$Symbols <- noquote(e$Symbols)
  
  #all_conditions_ID_vs_sham_names$ID
  
  #head(e)
  e |> #t() |>
    # write.table(paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
    write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
                sep = "\t",
                row.names = FALSE,
                col.names = TRUE,
                quote = FALSE)
  
  #Save the classes.
  write.table(x = b$Condition,
              # file = paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_names.cls"),
              file = paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_names.cls"),
              sep = "\t",
              quote = FALSE,
              col.names = FALSE,row.names = FALSE)
  return(list(go_table_selected_condition = a,
              conditon_table=b,
              data_frame_ready_counts = e))#,data_frame_ready_counts_sample_id = d))
  Sys.sleep(time = 15)
}
```


#### Apply fnction GEM

```{r}

#check that the arguments are for Ablation vs Sham or Uninjured.
#sham_bp_Mmus_05_nv_c$data |> distinct(Group)

m1 = Amputation_vs_Sham_mouse_enriched_fil_clean_mm  %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)

m2_2 = Ablated_vs_Sham_mouse_enriched_fil_clean_mm  %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)

m2_3 = Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm  %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)

m3 = Uninjured_vs_Sham_mouse_enriched_fil_clean_mm  %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)

m4 = Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm  %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse)




m1_setdiff_Amputation_vs_Sham <- setdiff(setdiff(setdiff(m1,m2_3),m3),m4)
length(m1_setdiff_Amputation_vs_Sham)

m2_setdiff_Ablated_vs_Sham <- setdiff(setdiff(setdiff(m2_2,m3),m4),m1)
length(m2_setdiff_Ablated_vs_Sham)

m2_setdiff_Ablated_vs_Uninjured <- setdiff(setdiff(setdiff(m2_3,m3),m4),m1)
length(m2_setdiff_Ablated_vs_Uninjured)

m3_setdiff_Uninjured_vs_Sham <- setdiff(setdiff(setdiff(m3,m4),m1),m2_3)
length(m3_setdiff_Uninjured_vs_Sham)

m4_setdiff_Cryoinjury_vs_Sham <- setdiff(setdiff(setdiff(m4,m1),m2_3),m3)
length(m4_setdiff_Cryoinjury_vs_Sham)
# m1_setdiff_Amputation_vs_Sham |> unique() |>  length()
# m2_setdiff_Ablated_vs_Sham |> unique() |>  length()
# m3_setdiff_Uninjured_vs_Sham |> unique() |>  length()
# m4_setdiff_Cryoinjury_vs_Sham |> unique() |>  length()



m1_setdiff_Amp_df <- Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m1_setdiff_Amputation_vs_Sham)

m2_setdiff_Abl_Sham_df <- Ablated_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m2_setdiff_Ablated_vs_Sham) 

m2_setdiff_Abl_Un_df <- Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m2_setdiff_Ablated_vs_Uninjured) 

m3_setdiff_Uni_df <- Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m3_setdiff_Uninjured_vs_Sham)

m4_setdiff_Cry_df <- Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m4_setdiff_Cryoinjury_vs_Sham)






amputation_data_sham_mmus_exact_from_fil_gem_clean <- create_gem(enrichment_data = sham_bp_Mmus_05_nv_c_3,
                                                                 condition = "Amputation",
                                                                 stats_data_condition = m1_setdiff_Amp_df,
                                                                 #correct_merge = "Row.names",
                                                                 text_Extra = "cleaned_",
                                                                 control = "Sham")


ablation_data_sham_mmus_exact_from_fil_gem_clean <- create_gem(enrichment_data = sham_bp_Mmus_05_nv_c_2,
                                                               condition = "Ablated",
                                                               stats_data_condition = m2_setdiff_Abl_Sham_df,
                                                               #correct_merge = "Row.names",
                                                               text_Extra = "cleaned_",
                                                               control = "Sham")

ablation_data_uninjured_mmus_exact_from_fil_gem_clean <- create_gem(enrichment_data=sham_bp_Mmus_05_nv_c_3,
                                                                    condition = "Ablated",
                                                                    stats_data_condition=m2_setdiff_Abl_Un_df,
                                                                    #correct_merge = "Row.names",
                                                                    text_Extra = "cleaned_",
                                                                    control = "Uninjured")

cryoinjury_data_sham_mmus_exact_from_fil_gem_clean <- create_gem(enrichment_data = sham_bp_Mmus_05_nv_c_3,
                                                                 condition = "Cryoinjury",
                                                                 stats_data_condition = m4_setdiff_Cry_df,
                                                                 #correct_merge = "Row.names",
                                                                 text_Extra = "cleaned_",
                                                                 control = "Sham")

uninjured_data_sham_mmus_exact_from_fil_gem_clean <- create_gem(enrichment_data = sham_bp_Mmus_05_nv_c_3,
                                                                condition = "Uninjured",
                                                                stats_data_condition = m3_setdiff_Uni_df,
                                                                #correct_merge = "Row.names",
                                                                text_Extra = "cleaned_",control = "Sham")
```
### Compare tables to Amputated vs Sham now to before.
```{r}
# amp1 <- amputation_data_sham_mmus_exact_from_fil_gem_clean$go_table_selected_condition
# amp2 <- read.table(file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/setdif_conditions/GEM_files/cleaned__gobp_Amputation_vs_Sham.txt",
#                    header = TRUE,
#                    sep = "\t")
#amp2 <- read.table(file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/setdif_conditions/GEM_files/cleaned__gobp_Amputation_vs_Sham.txt",header = TRUE,sep = "\t")

#amp1 <- amp1 |> filter(FDR < 0.05) |> mutate(TABLE = "NEW")
#amp2 <- amp2 |> filter(FDR < 0.05) |> mutate(TABLE = "OLD")
#vs_Uninjured = NEW
#vs_Sham = OLD
amp1 <- sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Amputation") |> filter(p.adjust < 0.05) |> mutate(TABLE = "New")
amp2 <- sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Amputation") |> filter(p.adjust < 0.05) |> mutate(TABLE = "Old")
```

```{r,fig.width=11,fig.height=11}
amputated_go <- VennDiagram::venn.diagram(x = list(amp1$Description,
                                                   amp2$Description),
                                          filename = NULL,
                                          units = "cm",
                                          category.names = c("Amputated\nNew","Amputated\nOld"),
                                          fill = c("#009E73","#E69F00"))
grid.newpage()
grid.draw(amputated_go)
```

```{r,fig.width=27,fig.height=76}
amputated_new <- data.frame(Description = setdiff(amp1$Description,amp2$Description))
amputated_old <- data.frame(Description = setdiff(amp2$Description,amp1$Description))
amputated_intersect <- data.frame(Description = intersect(amp1$Description,amp2$Description))

amputated_intersect <- amputated_intersect |> mutate(COMMON = "COMMON")
amputated_old <- amputated_old |> mutate(COMMON = "SHAM")
amputated_new <- amputated_new |> mutate(COMMON = "UNINJURED")

write.table(x = rbind(amputated_intersect,
                      amputated_new,
                      amputated_old),
            file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/Amputation_comparissons_gobp_old_new_table.txt",
            #file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/Amputation_comparissons_gobp_old_new_table.txt",
            col.names = TRUE,
            row.names = FALSE,
            sep = "\t")


amputated_plot <- rbind(#amputated_intersect,
                      amputated_new,
                      amputated_old) |>
  arrange(desc(COMMON)) |>
  #count(COMMON) |> 
  #mutate(func = fct_reorder(COMMON,n)) |> 
  ggplot() +
  geom_tile(aes(y = Description,
                x = COMMON,
                fill = COMMON))
amputated_plot
```


### Compare tables to Ablated vs Uninjured now to before.
```{r}
# ab1 <- ablation_data_uninjured_mmus_exact_from_fil_gem_clean$go_table_selected_condition
# ab2 <- read.table(file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/setdif_conditions/GEM_files/cleaned__gobp_Ablated_vs_Sham.txt",header = TRUE,sep = "\t")
# #ab2 <- read.table(file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/setdif_conditions/GEM_files/cleaned__gobp_Ablated_vs_Sham.txt",header = TRUE,sep = "\t")
# 
# 
# ab1 <- ab1 |> filter(FDR < 0.05) |> mutate(TABLE = "NEW")
# ab2 <- ab2 |> filter(FDR < 0.05) |> mutate(TABLE = "OLD")
ab1 <- sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Ablated") |> filter(p.adjust < 0.05) |> mutate(TABLE = "New")
ab2 <- sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Ablated") |> filter(p.adjust < 0.05) |> mutate(TABLE = "Old")
```

```{r,fig.width=7,fig.height=7}
ablated_go <- VennDiagram::venn.diagram(x = list(ab1$Description,
                                                 ab2$Description),
                                        filename = NULL,
                                        units = "cm",
                                        category.names = c("Ablated\nNew","Ablated\nOld"),
                                        fill = c("#E69F00","#009E73"))

grid.newpage()
grid.draw(ablated_go)
```

```{r,fig.width=27,fig.height=82}
ablated_old <- data.frame(Description = setdiff(ab1$Description,ab2$Description))
ablated_new <- data.frame(Description = setdiff(ab2$Description,ab1$Description))
ablated_intersect <- data.frame(Description = intersect(ab1$Description,ab2$Description))



ablated_intersect <- ablated_intersect |> mutate(COMMON = "COMMON")
ablated_old <- ablated_old |> mutate(COMMON = "SHAM")
ablated_new <- ablated_new |> mutate(COMMON = "UNINJURED")

write.table(x = rbind(ablated_intersect,
                      ablated_new,
                      ablated_old),
            file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/Ablation_comparissons_gobp_old_new_table.txt",
            #file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/Ablation_comparissons_gobp_old_new_table.txt",
            col.names = TRUE,row.names = FALSE,sep = "\t")

ablated_plot <- rbind(#ablated_intersect,
                      ablated_new,
                      ablated_old) |>
  arrange(desc(COMMON)) |>
  #count(COMMON) |> 
  #mutate(func = fct_reorder(COMMON,n)) |> 
  ggplot() +
  geom_tile(aes(y = Description,
                x = COMMON,
                fill = COMMON))
ablated_plot
```


### Compare tables to Uninjured vs Sham now to before.
```{r,fig.width=7,fig.height=7}
# un1 <- uninjured_data_sham_mmus_exact_from_fil_gem_clean$go_table_selected_condition
# un2 <- read.table(file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/setdif_conditions/GEM_files/cleaned__gobp_Uninjured_vs_Sham.txt",header = TRUE,sep = "\t")
# #un2 <- read.table(file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/setdif_conditions/GEM_files/cleaned__gobp_Uninjured_vs_Sham.txt",header = TRUE,sep = "\t")
# 
# un1 <- un1 |> filter(FDR < 0.05) |> mutate(TABLE = "NEW")
# un2 <- un2 |> filter(FDR < 0.05) |> mutate(TABLE = "OLD")
un1 <- sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Uninjured") |> filter(p.adjust < 0.05) |> mutate(TABLE = "New")
un2 <- sham_bp_Mmus_05_nv_c_2$data |> filter(Group == "Uninjured") |> filter(p.adjust < 0.05) |> mutate(TABLE = "Old")
```


```{r,fig.width=11,fig.height=9}
uninj_go <- VennDiagram::venn.diagram(x = list(un1$Description,un2$Description),
                                      filename = NULL,
                                      units = "cm",
                                      category.names = c("Uninjured\nnew","Uninjured\nold"),
                                      fill = c("#009E73","#E69F00"))


grid.newpage()
grid.draw(uninj_go)
```


```{r,fig.width=27,fig.height=83}
uninjured_old <- data.frame(Description = setdiff(un2$Description,un1$Description))
uninjured_new <- data.frame(Description = setdiff(un1$Description,un2$Description))
uninjured_intersect <- data.frame(Description = intersect(un1$Description,un2$Description))



uninjured_intersect <- uninjured_intersect |> mutate(COMMON = "COMMON")
uninjured_old <- uninjured_old |> mutate(COMMON = "SHAM")
uninjured_new <- uninjured_new |> mutate(COMMON = "UNINJURED")

write.table(x = rbind(uninjured_intersect,
                      uninjured_new,
                      uninjured_old),
            file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/Uninjured_comparissons_gobp_old_new_table.txt",
            #file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/Uninjured_comparissons_gobp_old_new_table.txt",
            col.names = TRUE,row.names = FALSE,sep = "\t")


uninjured_plot <- rbind(#uninjured_intersect,
                        uninjured_new,
                        uninjured_old) |>
  arrange(desc(COMMON)) |>
  #count(COMMON) |> U
  #mutate(func = fct_reorder(COMMON,n)) |> 
  ggplot() +
  geom_tile(aes(y = Description,
                x = COMMON,
                fill = COMMON))
uninjured_plot


# g <- ggplot(data = unj,
#             aes(y = Description,
#                 x = TABLE,
#                 fill = FDR)) +
#   geom_tile()+#aes(colour=ifelse(duplicated(Description),"green","red"))) + #,stat = "identity"
#   #scale_fill_gradient2(low = "blue",mid = "azure3",high = "red",midpoint = mean(unj$FDR)) +
#   labs(title = paste0("AblSham to AblUninj Annotations table"),
#        subtitle = paste0("padj < ",padjCO),
#        fill = "FDR",
#        #colour = "Duplication",
#        x = "Data origin",
#        y = "Description") +
#   geom_vline(xintercept = unj$Description[which(duplicated(unj$Description))],
#              #geom_vline(aes(xintercept = vlines,
#              color="black",
#              linetype="twodash",
#              size=1,
#              alpha=0.25) +
#   #coord_flip() +
#   theme(axis.text.x = element_text(angle = 90, hjust= 0.01,size = 9))
# g
# cat("Done!")
```
### Regeneration core GEM
```{r}
# heart_regeneration_core <- setdiff(intersect(Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
#                                              intersect(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
#                                                        Ablated_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse))),
#                                    intersect(Uninjured_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
#                                              intersect(Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm %>%filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
#                                                        intersect(Amputation_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse),
#                                                                  Ablated_vs_Sham_mouse_enriched_fil_clean_mm %>% filter(!is.na(EntrezGeneID_Mouse)) |> pull(EntrezGeneID_Mouse)))))
# 
# length(heart_regeneration_core)
#heart_regeneration_core_abl_sham

heart_regeneration_core_abl_unj

#Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> group_by(Ensembl_Danio) |> filter(Ensembl_Danio == "ENSDARG00000004561")


# heart_regeneration_core_mm_1 <- Amputation_vs_Sham_mouse_enriched_fil_clean_mm |> distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |> filter(EntrezGeneID_Mouse %in% heart_regeneration_core_abl_unj) |> rename(log2FoldChange_Amp_vs_Sham = log2FoldChange)
# heart_regeneration_core_mm_2 <- Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |> filter(EntrezGeneID_Mouse %in% heart_regeneration_core_abl_unj) |> rename(log2FoldChange_Cryo_vs_Sham = log2FoldChange)
# heart_regeneration_core_mm_3 <- Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm |> distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |> filter(EntrezGeneID_Mouse %in% heart_regeneration_core_abl_unj) |> rename(log2FoldChange_Abl_vs_Uninjured = log2FoldChange)


# heart_regeneration_core_mm <- Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>
#   distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
#   filter(EntrezGeneID_Mouse %in% heart_regeneration_core_abl_unj) |>
#   rename(log2FoldChange_Amp_vs_Sham = log2FoldChange)
# 
# heart_regeneration_core_mm_2 <- Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |>
#   distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
#   filter(EntrezGeneID_Mouse %in% heart_regeneration_core_abl_unj) |>
#   rename(log2FoldChange_Cryo_vs_Sham = log2FoldChange)

# heart_regeneration_core_mm_3 <- Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm |>
#   distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
#   filter(EntrezGeneID_Mouse %in% heart_regeneration_core_abl_unj) |>
#   rename(log2FoldChange_Abl_vs_Uninjured = log2FoldChange)

heart_regeneration_core_mm_v3 <- Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% heart_regeneration_core_abl_unj) #|>rename(log2FoldChange_Abl_vs_Uninjured = log2FoldChange)


heart_regeneration_core_mm_v2 <- Ablated_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% heart_regeneration_core_abl_sham)
#|>rename(log2FoldChange_Abl_vs_Uninjured = log2FoldChange)
#|
# 
# write.table(x = heart_regeneration_core_mm_v2$MGI_Symbol,file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/core_reg_abl_vs_sham_genes.txt",sep = "\t",row.names = FALSE,col.names = TRUE)


# 
# heart_regeneration_core_mm <- merge(x = heart_regeneration_core_mm_1,y = heart_regeneration_core_mm_2,by.x = "Ensembl_Danio",by.y = "Ensembl_Danio", all = TRUE)
# heart_regeneration_core_mm <- merge(x = heart_regeneration_core_mm,y = heart_regeneration_core_mm_3,by.x = "Ensembl_Danio",by.y = "Ensembl_Danio", all = TRUE)
# colnames(heart_regeneration_core_mm)
colnames(heart_regeneration_core_mm_v2[,c(0:10,16:51)]) #because the DEG are dffernt expressed n each conditon

# dir.create( "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/coreRegeneration/GOBP/")
# dir.create( "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/coreRegeneration/GEM/")
#dir.create(recursive = TRUE,path = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/coreRegeneration/GOBP/")
#dir.create(path = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/coreRegeneration/GEM/")
write.table(x = heart_regeneration_core_mm_v2,#[,c(0:10,16:51)],
            #file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GEM/heart_regeneration_core_mm_fil_clean.txt",
            file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GEM/heart_regeneration_core_mm_fil_v2_clean.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE,quote = FALSE)



heart_regeneration_core_gobp_v3 <- enrichGO(heart_regeneration_core_abl_unj,
                                            OrgDb = MmOrgDb,
                                            ont = "BP",
                                            pAdjustMethod = "BH",
                                            pvalueCutoff  =  0.05,
                                            readable = T)
heart_regeneration_core_gobp_v2 <- enrichGO(heart_regeneration_core_abl_sham,
                                            OrgDb = MmOrgDb,
                                            ont = "BP",
                                            pAdjustMethod = "BH",
                                            pvalueCutoff  =  0.05,
                                            readable = T)

dim(heart_regeneration_core_gobp_v3@result)
dim(heart_regeneration_core_gobp_v2@result)

write.table(x = heart_regeneration_core_gobp_v3@result |> filter(p.adjust < 0.05),
            #file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GOBP/Mmus_clean_coreRegeneration_gobp_abl_uninjured_enrichment.txt",
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GOBP/Mmus_clean_coreRegeneration_gobp_abl_uninjured_enrichment.txt",
            sep = "\t",
            row.names = FALSE,
            col.names = TRUE,
            quote = FALSE)
write.table(x = heart_regeneration_core_gobp_v2@result |> filter(p.adjust < 0.05),
            #file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GOBP/Mmus_clean_coreRegeneration_gobp_abl_sham_enrichment.txt",
            file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GOBP/Mmus_clean_coreRegeneration_gobp_abl_sham_enrichment.txt",
            sep = "\t",
            row.names = FALSE,
            col.names = TRUE,
            quote = FALSE)
#run with same function and copy elsewhere
#before runnning check in the function, $data,  Group
```
```{r}
abl_core_counts <- Ablated_vs_Sham_mouse_enriched_fil_clean_mm |> filter(MGI_Symbol %in% heart_regeneration_core_mm_v2$MGI_Symbol) |>  distinct(MGI_Symbol,.keep_all = TRUE) |> select(MGI_Symbol,rownames(s4c)[s4c$Condition == "Ablated"])#|> pull(MGI_Symbol)
```


```{r}
amp_core_counts <- Amputation_vs_Sham_mouse_enriched_fil_clean_mm |> filter(MGI_Symbol %in% heart_regeneration_core_mm_v2$MGI_Symbol) |>  distinct(MGI_Symbol,.keep_all = TRUE) |>
  select(MGI_Symbol,rownames(s4c)[s4c$Condition == "Amputation"])#|>#|> pull(MGI_Symbol)
```


```{r}
cryo_core_counts <- Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> filter(MGI_Symbol %in% heart_regeneration_core_mm_v2$MGI_Symbol) |>  distinct(MGI_Symbol,.keep_all = TRUE) |> select(MGI_Symbol,rownames(s4c)[s4c$Condition == "Cryoinjury"])#|>
```
```{r}
sham_core_counts <- Ablated_vs_Sham_mouse_enriched_fil_clean_mm |> filter(MGI_Symbol %in% heart_regeneration_core_mm_v2$MGI_Symbol) |>  distinct(MGI_Symbol,.keep_all = TRUE) |> select(MGI_Symbol,rownames(s4c)[s4c$Condition == "Sham"])#|>
```
### core selected counts uniques
```{r}
core_selected_counts_from <- cbind(amp_core_counts,abl_core_counts,cryo_core_counts,sham_core_counts)
colnames(core_selected_counts_from)
core_selected_counts_from <- core_selected_counts_from[,-c(8,14,22)]
core_selected_counts_from
write.table(x = core_selected_counts_from,file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/core_selected_counts_148_Table.txt",
            sep = "\t",row.names = FALSE,col.names = TRUE)
```



```{r}
create_gem2 <- function(condition,stats_data_condition,enrichment_data,text_Extra){
  #,correct_merge
  library("dplyr")
  # library("DESeq2")
  # library("SummarizedExperiment")
  library("tibble")
  library("biomaRt")
  library("curl")
  
  #Select the file we want to have as a network
  cat("Starting to wrangle the tables\n")
  a <- enrichment_data |>
  #a <- enrichment_data$data |>
    # dplyr::select(ID,Description,pvalue,p.adjust,Group,geneID) |>
    dplyr::select(ID,Description,pvalue,p.adjust,geneID) |>
    rename(p.Val = pvalue,
           FDR = p.adjust) |> 
    #mutate(Phenotype = ifelse(log2FoldChange > 0, 1,-1)) |>
    mutate(Phenotype = 1) |>
    #filter(Group == condition) |>
    #filter(Group == condition) |>
    mutate(Genes = noquote(gsub(x = geneID,pattern = "/",","))) |>
    #dplyr::select(-c(Group,geneID))
    dplyr::select(-c(geneID))

  
  a |> 
    # write.table(paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl/GEM_files/",text_Extra,"_",condition,"_vs_Sham.txt"),
    #write.table(paste0("D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/coreRegeneration/GEM/heart_regeneration_core_mm_fil_",text_Extra,"_gobp.txt"),
    write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GEM/heart_regeneration_core_mm_fil_",text_Extra,"_gobp.txt"),
                row.names = FALSE,
                col.names = TRUE,
                sep = "\t",
                quote = FALSE)
  
  #Select the ids from the table.
  b <- rbind(s4c |> filter(Condition == "Sham"),
             s4c |> filter(Condition == "Uninjured"),
             s4c |> filter(Condition %in% condition)) |>
    rownames_to_column("ID") |>
    dplyr::select(Condition,ID) |>
    arrange(desc(Condition))
  
 
  cat(paste0("Saving files for ",condition,"\n"))
  #Change sample nanme to condition name
  e <- stats_data_condition[,c("MGI_Symbol","Description",b$ID,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")]
  colnames(e) <- c("Symbols","Description",b$Condition,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")
  e$Ensembl_Danio <- noquote(e$Ensembl_Danio)
  e$Ensembl_Mouse <- noquote(e$Ensembl_Mouse)
  e$Zfin_Symbol <- noquote(e$Zfin_Symbol)
  #e$Symbols <- toupper(e$Symbols)
  e$Symbols <- noquote(e$Symbols)
  e$MGI_Symbol <- toupper(e$MGI_Symbol)


  
  #all_conditions_ID_vs_sham_names$ID
    write.table(x = e,file = paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GEM/heart_regeneration_core_mm_fil_clean_",text_Extra,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
                sep = "\t",
                row.names = FALSE,col.names = TRUE,
                quote = FALSE)
  #Save the classes.
   write.table(x = b$Condition,file = paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/GEM/heart_regeneration_core_mm_fil_clean_",text_Extra,"_names.cls"),
              sep = "\t",
              quote = FALSE,
              col.names = FALSE,row.names = FALSE)
  
  return(list(go_table_selected_condition = a,
              conditon_table=b,
              data_frame_ready_counts = e))#,data_frame_ready_counts_sample_id = d))
  Sys.sleep(time = 15)
}
```


### GEM Ablated vs Uninjured heart core regeneration
```{r}

#####
##### add the hashtag to the condition interested not to appear in the table.
#####
#rm(heart_regeneration_core_from_fil_gem_clean)
heart_regeneration_core_from_fil_gem_clean_v3 <- create_gem2(
  enrichment_data=heart_regeneration_core_gobp_v3@result,
  #condition=c("Amputation","Ablated","Cryoinjury"),
  condition=c("Amputation","Ablated","Cryoinjury"),
  stats_data_condition=heart_regeneration_core_mm_v3,
  #correct_merge = "Row.names",
  text_Extra = "cleaned_Abl_Un_")
```
### GEM Ablated vs Sham heart core regeneration
```{r}
#rm(heart_regeneration_core_from_fil_gem_clean)
heart_regeneration_core_from_fil_gem_clean_v2<-create_gem2(
  enrichment_data=heart_regeneration_core_gobp_v2@result |> dplyr::filter(p.adjust < 0.05),
  condition=c("Amputation","Ablated","Cryoinjury"),
  #condition = "Ablated",
  stats_data_condition = heart_regeneration_core_mm_v2,
  #correct_merge = "Row.names",
  text_Extra = "cleaned_Abl_Sham_")
```




### Create GEM up and down regulated.
```{r}
unique(sham_bp_Mmus_05_v2_up$data$Group)
unique(sham_bp_Mmus_05_v2_down$data$Group)

sham_bp_Mmus_05_v2_up$data$Phenotype <- 1
sham_bp_Mmus_05_v2_down$data$Phenotype <- -1

sham_bp_Mmus_05_v2_up_and_down <- rbind(sham_bp_Mmus_05_v2_up$data,sham_bp_Mmus_05_v2_down$data)
sham_bp_Mmus_05_v2_up_and_down
```
```{r}
create_gem_updown <- function(condition,stats_data_condition,enrichment_data,text_Extra,control){
  #,correct_merge
  library("dplyr")
  library("DESeq2")
  library("SummarizedExperiment")
  library("tibble")
  library("biomaRt")
  library("curl")
  
  #Select the file we want to have as a network
  cat("Starting to wrangle the tables\n")
  
  #For using with GOBP unique it is removed enrichment_data$data
  a <- enrichment_data |>
    dplyr::select(ID,Description,pvalue,p.adjust,Group,geneID,Phenotype) |>
    rename(p.Val = pvalue,
           FDR = p.adjust) |> 
    #mutate(Phenotype = ifelse(log2FoldChange > 0, 1,-1)) |>
    #mutate(Phenotype = 1) |>
    filter(Group == condition) |>
    #filter(Group == condition) |>
    mutate(Genes = noquote(gsub(x = geneID,pattern = "/",","))) |>
    dplyr::select(-c(Group,geneID))
    

  
  a |>
    # write.table(paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,".txt"),
    write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,".txt"),
              row.names = FALSE,
              col.names = TRUE,
              sep = "\t",
              quote = FALSE)
  #Select the ids from the table.
  #b <- rbind(s4c |> filter(Condition == "Uninjured"),
  b <- rbind(s4c |> filter(Condition == control),
             s4c |> filter(Condition == condition)) |>
    rownames_to_column("ID") |>
    dplyr::select(Condition,ID) |>
    arrange(desc(Condition))
  
  #cat("Getting the RLD DESeq2 normalized counts\n")
  #cc <- DESeq2_platform_normalized_counts[,colnames(DESeq2_platform_normalized_counts) %in% b$ID]
  
  
  # #Save the data frame for working with it and have it available in R too.
  # cat("Make sure the dataframe using the column to merge are the ENSDARG...\n")
  # d <- merge(x = as.data.frame(c),
  #            #y = as.data.frame(stats_data_condition2),
  #            y = as.data.frame(stats_data_condition),
  #            by.x = "row.names",
  #            by.y = correct_merge) |>
  #   dplyr::select(MGI_Symbol,Description,log2FoldChange,Ensembl_Danio,Ensembl_Mouse,EntrezGeneID_Mouse,Zfin_Symbol,starts_with("SRR")) |>
  #   mutate(Symbols = toupper(MGI_Symbol)) |> #,Phenotype = if_else(log2FoldChange > 0, 1, -1)) |>
  #   dplyr::select(MGI_Symbol,Description_Mm,starts_with("SRR"),Ensembl_Danio,Ensembl_Mouse,EntrezGeneID_Mouse,MGI_Symbol,Zfin_Symbol)#,-c(external_gene_name.y))
  # 
  # 
  cat(paste0("Saving files for ",condition,"\n"))
 
  #Change sample nanme to condition name
  e <- stats_data_condition[,c("MGI_Symbol","Description",b$ID,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")]
  colnames(e) <- c("Symbols","Description",b$Condition,"Ensembl_Danio","Ensembl_Mouse","EntrezGeneID_Mouse","MGI_Symbol","Zfin_Symbol","log2FoldChange")
  e$Ensembl_Danio <- noquote(e$Ensembl_Danio)
  e$Ensembl_Mouse <- noquote(e$Ensembl_Mouse)
  e$Zfin_Symbol <- noquote(e$Zfin_Symbol)
  #e$Symbols <- toupper(e$Symbols)
  e$Symbols <- noquote(e$Symbols)
  
  #all_conditions_ID_vs_sham_names$ID
  
  #head(e)
  e |> #t() |>
    # write.table(paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
    write.table(paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_normalized_counts_expression_symbol_and_condition_enmap.txt"),
                sep = "\t",
                row.names = FALSE,
                col.names = TRUE,
                quote = FALSE)
  
  #Save the classes.
  write.table(x = b$Condition,
              # file = paste0("C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_names.cls"),
              file = paste0("/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/setDiffConditions/GEM/",text_Extra,"_",condition,"_vs_",control,"_names.cls"),
              sep = "\t",
              quote = FALSE,
              col.names = FALSE,row.names = FALSE)
  return(list(go_table_selected_condition = a,
              conditon_table=b,
              data_frame_ready_counts = e))#,data_frame_ready_counts_sample_id = d))
  Sys.sleep(time = 15)
}
```


#### apply function
```{r}
#get the downregulated  set
m1_down = Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>  filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |> pull(EntrezGeneID_Mouse)
m2_down = Ablated_vs_Sham_mouse_enriched_fil_clean_mm |>  filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |> pull(EntrezGeneID_Mouse)
#d2 = Ablated_vs_Uninjured_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
m3_down = Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |> filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |> pull(EntrezGeneID_Mouse)
m4_down = Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange < -1) |> pull(EntrezGeneID_Mouse)




m1_setdiff_Amputation_vs_Sham_down <- setdiff(setdiff(setdiff(m1_down,m2_down),m3_down),m4_down)
length(m1_setdiff_Amputation_vs_Sham_down)

m2_setdiff_Ablated_vs_Sham_down <- setdiff(setdiff(setdiff(m2_down,m3_down),m4_down),m1_down)
length(m2_setdiff_Ablated_vs_Sham_down)

#m2_setdiff_Ablated_vs_Uninjured_down <- setdiff(setdiff(setdiff(m2_down,m3_down),m4_down),m1_down)
#length(m2_setdiff_Ablated_vs_Uninjured_down)

m3_setdiff_Uninjured_vs_Sham_down <- setdiff(setdiff(setdiff(m3_down,m4_down),m1_down),m2_down)
length(m3_setdiff_Uninjured_vs_Sham_down)

m4_setdiff_Cryoinjury_vs_Sham_down <- setdiff(setdiff(setdiff(m4_down,m1_down),m2_down),m3_down)
length(m4_setdiff_Cryoinjury_vs_Sham_down)
##########################
#get the upregulated  set#
##########################

m1_up = Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>  filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |> pull(EntrezGeneID_Mouse)
m2_up = Ablated_vs_Sham_mouse_enriched_fil_clean_mm |>  filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |> pull(EntrezGeneID_Mouse)
#d2 = Ablated_vs_Uninjured_mouse_enriched_fil_clean %>% filter(!is.na(EntrezGeneID_Mouse)) %>% pull(EntrezGeneID_Mouse),
m3_up = Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |> filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |> pull(EntrezGeneID_Mouse)
m4_up = Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |> filter(!is.na(EntrezGeneID_Mouse)) |> filter(log2FoldChange > 1) |> pull(EntrezGeneID_Mouse)




m1_setdiff_Amputation_vs_Sham_up <- setdiff(setdiff(setdiff(m1_up,m2_up),m3_up),m4_up)
length(m1_setdiff_Amputation_vs_Sham_up)

m2_setdiff_Ablated_vs_Sham_up <- setdiff(setdiff(setdiff(m2_up,m3_up),m4_up),m1_up)
length(m2_setdiff_Ablated_vs_Sham_up)

#m2_setdiff_Ablated_vs_Uninjured_down <- setdiff(setdiff(setdiff(m2_down,m3_down),m4_down),m1_down)
#length(m2_setdiff_Ablated_vs_Uninjured_down)

m3_setdiff_Uninjured_vs_Sham_up <- setdiff(setdiff(setdiff(m3_up,m4_up),m1_up),m2_up)
length(m3_setdiff_Uninjured_vs_Sham_up)

m4_setdiff_Cryoinjury_vs_Sham_up <- setdiff(setdiff(setdiff(m4_up,m1_up),m2_up),m3_up)
length(m4_setdiff_Cryoinjury_vs_Sham_up)



##############################
#merge and get the GEM file  #
##############################


m1_setdiff_Amp_df_up_and_down <- c(m1_setdiff_Amputation_vs_Sham_down,m1_setdiff_Amputation_vs_Sham_up)
m2_setdiff_Abl_Sham_df_up_and_down <- c(m2_setdiff_Ablated_vs_Sham_down,m2_setdiff_Ablated_vs_Sham_up)
m3_setdiff_Uni_df_up_and_down <- c(m3_setdiff_Uninjured_vs_Sham_down,m3_setdiff_Uninjured_vs_Sham_up)
m4_setdiff_Cry_df_up_and_down <- c(m4_setdiff_Cryoinjury_vs_Sham_down, m4_setdiff_Cryoinjury_vs_Sham_up)

m1_setdiff_Amp_df_up_and_down <- Amputation_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m1_setdiff_Amputation_vs_Sham_down)

m2_setdiff_Abl_Sham_df_up_and_down <- Ablated_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m2_setdiff_Ablated_vs_Sham_down) 

# m2_setdiff_Abl_Un_df_up_and_down <- Ablated_vs_Uninjured_mouse_enriched_fil_clean_mm |>
#   distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
#   filter(EntrezGeneID_Mouse %in% m2_setdiff_Ablated_vs_Uninjured_down) 

m3_setdiff_Uni_df_up_and_down <- Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m3_setdiff_Uninjured_vs_Sham_down)

m4_setdiff_Cry_df_up_and_down <- Cryonjury_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(EntrezGeneID_Mouse,.keep_all = TRUE) |>
  filter(EntrezGeneID_Mouse %in% m4_setdiff_Cryoinjury_vs_Sham_down)




amputation_data_sham_mmus_exact_from_fil_gem_clean_up_down_regulated <- create_gem_updown(enrichment_data = sham_bp_Mmus_05_v2_up_and_down,
                                                                 condition = "Amputation",
                                                                 stats_data_condition = m1_setdiff_Amp_df_up_and_down,
                                                                 #correct_merge = "Row.names",
                                                                 text_Extra = "cleaned_up_and_down_",
                                                                 control = "Sham")


ablation_data_sham_mmus_exact_from_fil_gem_clean_up_down_regulated <- create_gem_updown(enrichment_data = sham_bp_Mmus_05_v2_up_and_down,
                                                               condition = "Ablated",
                                                               stats_data_condition = m2_setdiff_Abl_Sham_df_up_and_down,
                                                               #correct_merge = "Row.names",
                                                               text_Extra = "cleaned_up_and_down_",
                                                               control = "Sham")
# 
# ablation_data_uninjured_mmus_exact_from_fil_gem_clean <- create_gem(enrichment_data=sham_bp_Mmus_05_v2_up_and_down,
#                                                                     condition = "Ablated",
#                                                                     stats_data_condition=m2_setdiff_Abl_Un_df,
#                                                                     #correct_merge = "Row.names",
#                                                                     text_Extra = "cleaned_",
#                                                                     control = "Uninjured")

cryoinjury_data_sham_mmus_exact_from_fil_gem_clean_up_down_regulated <- create_gem_updown(enrichment_data = sham_bp_Mmus_05_v2_up_and_down,
                                                                 condition = "Cryoinjury",
                                                                 stats_data_condition = m4_setdiff_Cry_df_up_and_down,
                                                                 #correct_merge = "Row.names",
                                                                 text_Extra = "cleaned_up_and_down_",
                                                                 control = "Sham")

uninjured_data_sham_mmus_exact_from_fil_gem_clean_up_down_regulated <- create_gem_updown(enrichment_data = sham_bp_Mmus_05_v2_up_and_down,
                                                                condition = "Uninjured",
                                                                stats_data_condition = m3_setdiff_Uni_df_up_and_down,
                                                                #correct_merge = "Row.names",
                                                                text_Extra = "cleaned_up_and_down_",control = "Sham")

```


```{r}
require(dplyr)
uninj_setdiff_gobp_v3_genes <- data.frame("Genes" = uninj_setdiff_gobp_v3 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist())
#uninj_setdiff_gobp_v3 %>% group_by(geneID) %>% pull(geneID) %>% str_split("\\/") %>% unlist() %>% as.data.frame() %>% dplyr::distinct()
# symbols[[target]]<-unique(unlist(strsplit(as.character(FDR_pathways@result$geneID[i]),split = "\\/")))
uninj_setdiff_gobp_v3_genes <- uninj_setdiff_gobp_v3_genes %>% dplyr::distinct()  

uninj_setdiff_gobp_v3_genes_mm <- Uninjured_vs_Sham_mouse_enriched_fil_clean_mm |>
  distinct(MGI_Symbol,.keep_all = TRUE) |>
  filter(MGI_Symbol %in% uninj_setdiff_gobp_v3_genes$Genes)



v2_gobp_hc_genes <- data.frame("Genes" = heart_regeneration_core_gobp_v2@result |>  dplyr::group_by(geneID) |> dplyr::pull(geneID) |>  str_split("\\/") |> unlist())
v2_gobp_hc_genes <- v2_gobp_hc_genes |> dplyr::distinct()
v2_gobp_hc_genes |> filter(grepl("nek",Genes,ignore.case = TRUE))
v2_gobp_hc_genes |> filter(str_detect(Genes,regex("rbm",ignore_case = TRUE)))
v2_gobp_hc_genes |> filter(str_detect(Genes,regex("pdk",ignore_case = TRUE)))
v2_gobp_hc_genes |> filter(str_detect(Genes,regex("stmn",ignore_case = TRUE)))
v2_gobp_hc_genes |> filter(str_detect(Genes,regex("mrp",ignore_case = TRUE)))
v2_gobp_hc_genes |> filter(str_detect(Genes,regex("enpp",ignore_case = TRUE)))
v2_gobp_hc_genes |> filter(str_detect(Genes,regex("gpx",ignore_case = TRUE)))


v3_gobp_hc_genes <- data.frame("Genes" = heart_regeneration_core_gobp_v3@result |>  dplyr::group_by(geneID) |> dplyr::pull(geneID) |>  str_split("\\/") |> unlist())
v3_gobp_hc_genes <- v3_gobp_hc_genes |> dplyr::distinct()
v3_gobp_hc_genes |> filter(str_detect(Genes,regex("nek",ignore_case = TRUE)))
v3_gobp_hc_genes |> filter(str_detect(Genes,regex("rbm",ignore_case = TRUE)))
v3_gobp_hc_genes |> filter(str_detect(Genes,regex("pdk",ignore_case = TRUE)))
v3_gobp_hc_genes |> filter(str_detect(Genes,regex("stmn",ignore_case = TRUE)))
v3_gobp_hc_genes |> filter(str_detect(Genes,regex("mrp",ignore_case = TRUE)))
v3_gobp_hc_genes |> filter(str_detect(Genes,regex("gpx",ignore_case = TRUE)))



```


### Compare heart regeneration cores

```{r,fig.width=6,fig.height=6}
length(heart_regeneration_core_abl_unj)
length(heart_regeneration_core_abl_sham)
heartCoreReg_uninj_and_sham <- VennDiagram::venn.diagram(x = list(heart_regeneration_core_abl_unj,
                                   heart_regeneration_core_abl_sham),
                          filename = NULL,
                          units = "cm",
                          category.names = c("Ablated\nvs\nSham_core", "Ablated\nvs\nUninjured_core"),
                          fill = c("#009E73","#E69F00"))
grid.newpage()
grid.draw(heartCoreReg_uninj_and_sham)
```


```{r,fig.width=8,fig.height=12}
length(intersect(heart_regeneration_core_abl_unj,heart_regeneration_core_abl_sham))
length(setdiff(heart_regeneration_core_abl_unj,heart_regeneration_core_abl_sham))
length(setdiff(heart_regeneration_core_abl_sham,heart_regeneration_core_abl_unj))


intersect(heart_regeneration_core_abl_unj,heart_regeneration_core_abl_sham)
columns(org.Mm.eg.db)


symbols_setdiff_abl_uninjured_core <- AnnotationDbi::select(org.Mm.eg.db,
                                           keys = as.character(setdiff(heart_regeneration_core_abl_unj,heart_regeneration_core_abl_sham)),
                                           columns = c("ENTREZID","SYMBOL","GENENAME"),
                                           keytype = "ENTREZID")

symbols_setdiff_abl_sham_core <- AnnotationDbi::select(org.Mm.eg.db,
                                           keys = as.character(setdiff(heart_regeneration_core_abl_sham,heart_regeneration_core_abl_unj)),
                                           columns = c("ENTREZID","SYMBOL","GENENAME"),
                                           keytype = "ENTREZID")
  
symbols_intersect <- AnnotationDbi::select(org.Mm.eg.db,
                                           keys = as.character(intersect(heart_regeneration_core_abl_unj,heart_regeneration_core_abl_sham)),
                                           columns = c("ENTREZID","SYMBOL","GENENAME"),
                                           keytype = "ENTREZID")



symbols_intersect <- symbols_intersect |> mutate(COMMON = "COMMON")
symbols_setdiff_abl_sham_core <- symbols_setdiff_abl_sham_core |> mutate(COMMON = "SHAM")
symbols_setdiff_abl_uninj_core <- symbols_setdiff_abl_uninjured_core |> mutate(COMMON = "UNINJURED")


write.table(x = rbind(symbols_intersect,symbols_setdiff_abl_sham_core,symbols_setdiff_abl_uninj_core),
            #file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/coreRegeneration/heart_regeneration_core_old_vs_new_table.txt"  )
            file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/heart_regeneration_core_old_vs_new_table_genes.txt",
            sep = "\t",col.names = TRUE,row.names = FALSE)


symbols_plot <- rbind(#symbols_intersect,
  symbols_setdiff_abl_sham_core,
  symbols_setdiff_abl_uninj_core) |>
  arrange(desc(COMMON)) |>
  #count(COMMON) |> 
  #mutate(func = fct_reorder(COMMON,n)) |> 
  ggplot() +
  geom_tile(aes(y = toupper(SYMBOL),
                x = COMMON,
                fill = COMMON))+
  xlab("Comparison") +
  ylab("Genes") +
  labs(fill = "Comparison")
symbols_plot
```

### compare hr cores descriptions from GOBP table
```{r}
# h1 <- heart_regeneration_core_from_fil_gem_clean$go_table_selected_condition
# #h2 <- read.table("coreRegeneration/GEM/heart_regeneration_core_mm_fil_cleaned__gobp.txt",header = TRUE,sep = "\t")
# #h2 <- read.table("D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/coreRegeneration/GEM/heart_regeneration_core_mm_fil_cleaned__gobp.txt",header = TRUE,sep = "\t")
# h2 <- read.table("/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v2/coreRegeneration/GEM/heart_regeneration_core_mm_fil_cleaned__gobp.txt",header = TRUE,sep = "\t")
# 
# h1 <- h1 |> filter(FDR < 0.05) |> mutate(TABLE = "NEW")
# h2 <- h2 |> filter(FDR < 0.05) |> mutate(TABLE = "OLD")
# hcore <- rbind(h1,h2)
h1 <- heart_regeneration_core_gobp_v3@result |> filter(p.adjust < 0.05) |> mutate(TABLE = "New")
h2 <- heart_regeneration_core_gobp_v2@result |> filter(p.adjust < 0.05) |> mutate(TABLE = "Old")
```


```{r,fig.width=7,fig.height=7}
hcore_go <- VennDiagram::venn.diagram(x = list(h1$Description,h2$Description),
                                      filename = NULL,
                                      units = "cm",
                                      category.names = c("Heart_Core\nNew","Heart_Core\nOld"),
                                      fill = c("#009E73","#E69F00"))


grid.newpage()
grid.draw(hcore_go)
```
```{r,fig.width=13,fig.height=19}
heart_core_new <- data.frame(Description = setdiff(h1$Description,h2$Description))
heart_core_old <- data.frame(Description = setdiff(h2$Description,h1$Description))
heart_core_intersect <- data.frame(Description = intersect(h1$Description,h2$Description))


heart_core_intersect <- heart_core_intersect |> mutate(COMMON = "COMMON")
heart_core_old <- heart_core_old |> mutate(COMMON = "SHAM")
heart_core_new <- heart_core_new |> mutate(COMMON = "UNINJURED")

write.table(x = rbind(heart_core_intersect,
                      heart_core_new,
                      heart_core_old),
            file = "/media/marius/Samsung_T5/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/heart_regeneration_core_old_vs_new_table_gobp.txt",
            #file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/coreRegeneration/heart_regeneration_core_old_vs_new_table_gobp.txt",
            col.names = TRUE,row.names = FALSE,sep = "\t")


heart_core_plot <- rbind(#uninjured_intersect,
                        heart_core_new,
                        heart_core_old) |>
  arrange(desc(COMMON)) |>
  #count(COMMON) |> U
  #mutate(func = fct_reorder(COMMON,n)) |> 
  ggplot() +
  geom_tile(aes(y = Description,
                x = COMMON,
                fill = COMMON)) +
  xlab("Comparison") +
  ylab("Genes") +
  labs(fill = "Comparison")
heart_core_plot
```


### Get the GOBP unique enrichment map of them.

### Ablated vs Uninjured GO:BP Unique
```{r}
m1 = sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Amputation")
m3 = sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Cryoinjury")
m2 = sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Ablated")
m4 = sham_bp_Mmus_05_nv_c_3$data |> filter(Group == "Uninjured")

amp_setdiff_gobp_v3 <- sham_bp_Mmus_05_nv_c_3$data |>
  filter(Group == "Amputation") |>
  filter(Description %in% setdiff(setdiff(setdiff(m1$Description,m2$Description),m3$Description),m4$Description))

abl_setdiff_gobp_v3 <- sham_bp_Mmus_05_nv_c_3$data |>
  filter(Group == "Ablated") |>
  filter(Description %in% setdiff(setdiff(setdiff(m2$Description,m3$Description),m4$Description),m1$Description))

uninj_setdiff_gobp_v3 <- sham_bp_Mmus_05_nv_c_3$data |>
  filter(Group == "Uninjured") |>
  filter(Description %in% setdiff(setdiff(m4$Description,m2$Description),m1$Description))
```


### fgsea
#### Mouse Hallmarks
```{r}
library("fgsea")
library("data.table")
library("ggplot2")
library("dplyr")

#Create the pathway fo the Hallmarks data.
dir.create(recursive = T, "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks")

#Load the Hallmark data
#load("/mnt/haus/marius/Carol_data/gsea_bundle/mouse_H_v5p2.rdata")
#Got the data from ---> https://bioinf.wehi.edu.au/MSigDB/index.html
Mm.H <- readRDS("/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Collections/Mm.h.all.v7.1.entrez.rds")
#length(Mm.H)
#Save the 50 pathways composing the hallmark GSEAinto an object
#mm_hallmark_fgsea<-Mm.H
#Conduct analysis the GSEA enrichment for the Hallmarks!

#Get the RANKED list ready
Cryonjury_vs_Sham_mouse_enriched_all_clean_mm_ranked <- Cryonjury_vs_Sham_mouse_enriched_all_clean_mm |> dplyr::group_by(EntrezGeneID_Mouse) |>
  dplyr::select(EntrezGeneID_Mouse,log2FoldChange) |> dplyr::distinct(.keep_all = TRUE) |> dplyr::pull(log2FoldChange)

names(Cryonjury_vs_Sham_mouse_enriched_all_clean_mm_ranked) <- Cryonjury_vs_Sham_mouse_enriched_all_clean_mm |> dplyr::group_by(EntrezGeneID_Mouse) |>
  dplyr::select(EntrezGeneID_Mouse,log2FoldChange) |> dplyr::distinct(.keep_all = TRUE) |> dplyr::pull(EntrezGeneID_Mouse)

Amputation_vs_Sham_mouse_enriched_all_clean_mm_ranked <- Amputation_vs_Sham_mouse_enriched_all_clean_mm |> dplyr::group_by(EntrezGeneID_Mouse) |>
  dplyr::select(EntrezGeneID_Mouse,log2FoldChange) |> dplyr::distinct(.keep_all = TRUE) |> dplyr::pull(log2FoldChange)

names(Amputation_vs_Sham_mouse_enriched_all_clean_mm_ranked) <- Amputation_vs_Sham_mouse_enriched_all_clean_mm |> dplyr::group_by(EntrezGeneID_Mouse) |>
  dplyr::select(EntrezGeneID_Mouse,log2FoldChange) |> dplyr::distinct(.keep_all = TRUE) |> dplyr::pull(EntrezGeneID_Mouse)


Ablated_vs_Sham_mouse_enriched_all_clean_mm_ranked <- Ablated_vs_Sham_mouse_enriched_all_clean_mm |> dplyr::group_by(EntrezGeneID_Mouse) |>
  dplyr::select(EntrezGeneID_Mouse,log2FoldChange) |> dplyr::distinct(.keep_all = TRUE) |> dplyr::pull(log2FoldChange)


names(Ablated_vs_Sham_mouse_enriched_all_clean_mm_ranked) <- Ablated_vs_Sham_mouse_enriched_all_clean_mm |> dplyr::group_by(EntrezGeneID_Mouse) |>
  dplyr::select(EntrezGeneID_Mouse,log2FoldChange) |> dplyr::distinct(.keep_all = TRUE) |> dplyr::pull(EntrezGeneID_Mouse)

Uninjured_vs_Sham_mouse_enriched_all_clean_mm_ranked <- Uninjured_vs_Sham_mouse_enriched_all_clean_mm |> dplyr::group_by(EntrezGeneID_Mouse) |>
  dplyr::select(EntrezGeneID_Mouse,log2FoldChange) |>dplyr:: distinct(.keep_all = TRUE) |> dplyr::pull(log2FoldChange)

names(Uninjured_vs_Sham_mouse_enriched_all_clean_mm_ranked) <- Uninjured_vs_Sham_mouse_enriched_all_clean_mm |> dplyr::group_by(EntrezGeneID_Mouse) |>
  dplyr::select(EntrezGeneID_Mouse,log2FoldChange) |> dplyr::distinct(.keep_all = TRUE) |> dplyr::pull(EntrezGeneID_Mouse)



mm_cryoinjury_sham_fgsea_hallmarks <- fgsea(pathways = Mm.H,
                                            stats = Cryonjury_vs_Sham_mouse_enriched_all_clean_mm_ranked,
                                            minSize = 15, maxSize =500, nperm = 1000)
mm_amputation_sham_fgsea_hallmarks <- fgsea(pathways = Mm.H,
                                            stats = Amputation_vs_Sham_mouse_enriched_all_clean_mm_ranked,
                                            minSize = 15, maxSize =500, nperm = 1000)
mm_ablation_sham_fgsea_hallmarks <- fgsea(pathways = Mm.H,
                                          stats = Ablated_vs_Sham_mouse_enriched_all_clean_mm_ranked,
                                          minSize = 15, maxSize =500, nperm = 1000)
mm_uninjury_sham_fgsea_hallmarks <- fgsea(pathways = Mm.H,
                                          stats = Uninjured_vs_Sham_mouse_enriched_all_clean_mm_ranked,
                                          minSize = 15, maxSize =500, nperm = 1000)
```


```{r}
#Now lets see the 30 top adjusted.
head(mm_cryoinjury_sham_fgsea_hallmarks, n=30)
head(mm_amputation_sham_fgsea_hallmarks, n=30)
head(mm_ablation_sham_fgsea_hallmarks, n=30)
head(mm_uninjury_sham_fgsea_hallmarks, n=30)


#Map the leadingEdges from ENTREZIDs to SYMBOLS for being more human readable.
mm_cryoinjury_sham_fgsea_hallmarks[,leadingEdge:=lapply(leadingEdge,mapIds, x=org.Mm.eg.db, keytype="ENTREZID",column="SYMBOL")]
mm_amputation_sham_fgsea_hallmarks[,leadingEdge:=lapply(leadingEdge,mapIds, x=org.Mm.eg.db, keytype="ENTREZID",column="SYMBOL")]
mm_ablation_sham_fgsea_hallmarks[,leadingEdge:=lapply(leadingEdge,mapIds, x=org.Mm.eg.db, keytype="ENTREZID",column="SYMBOL")]
mm_uninjury_sham_fgsea_hallmarks[,leadingEdge:=lapply(leadingEdge,mapIds, x=org.Mm.eg.db, keytype="ENTREZID",column="SYMBOL")]

head(mm_cryoinjury_sham_fgsea_hallmarks$leadingEdge)
head(mm_cryoinjury_sham_fgsea_hallmarks)

#Order the data according to padj value
mm_cryoinjury_sham_fgsea_hallmarks <- mm_cryoinjury_sham_fgsea_hallmarks[order(mm_cryoinjury_sham_fgsea_hallmarks[,3])]
mm_amputation_sham_fgsea_hallmarks <- mm_amputation_sham_fgsea_hallmarks[order(mm_amputation_sham_fgsea_hallmarks[,3])]
mm_ablation_sham_fgsea_hallmarks <- mm_ablation_sham_fgsea_hallmarks[order(mm_ablation_sham_fgsea_hallmarks[,3])]
mm_uninjury_sham_fgsea_hallmarks <- mm_uninjury_sham_fgsea_hallmarks[order(mm_uninjury_sham_fgsea_hallmarks[,3])]


#Remove the data which padj=1
mm_cryoinjury_sham_fgsea_hallmarks <- mm_cryoinjury_sham_fgsea_hallmarks[mm_cryoinjury_sham_fgsea_hallmarks$padj != 1]
mm_amputation_sham_fgsea_hallmarks <- mm_amputation_sham_fgsea_hallmarks[mm_amputation_sham_fgsea_hallmarks$padj != 1]
mm_ablation_sham_fgsea_hallmarks <- mm_ablation_sham_fgsea_hallmarks[mm_ablation_sham_fgsea_hallmarks$padj != 1]
mm_uninjury_sham_fgsea_hallmarks <- mm_uninjury_sham_fgsea_hallmarks[mm_uninjury_sham_fgsea_hallmarks$padj != 1]


#Save the data of the GSEA HALLMARK PATHWAYS, using a special function for writting the data table data frame with data.table::fwrite..

data.table::fwrite(mm_uninjury_sham_fgsea_hallmarks, "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks/mm_uninjury_sham_fgsea_hallmarks.txt",sep = "\t",
                   col.names = T, row.names = F, quote = F)
data.table::fwrite(mm_amputation_sham_fgsea_hallmarks, "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks/mm_amputation_sham_fgsea_hallmarks.txt",sep = "\t",
                   col.names = T, row.names = F, quote = F)
data.table::fwrite(mm_ablation_sham_fgsea_hallmarks, "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks/mm_ablation_sham_fgsea_hallmarks.txt",sep = "\t",
                   col.names = T, row.names = F, quote = F)
data.table::fwrite(mm_cryoinjury_sham_fgsea_hallmarks, "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks/mm_cryoinjury_sham_fgsea_hallmarks.txt",sep = "\t",
                   col.names = T, row.names = F, quote = F)



mm_cryoinjury_sham_fgsea_hallmarks[[1]][1:10]
mm_ablation_sham_fgsea_hallmarks[[1]][1:10]
mm_amputation_sham_fgsea_hallmarks[[1]][1:10]
mm_uninjury_sham_fgsea_hallmarks[[1]][1:10]

# #Choose from the list any of the interested gsea plot enrichment interested.
# pdf("./Enrichment/GSEA/Hallmarks/Mmusculus/mm_fgsea_hallmarks_myogenesis.pdf",width = 16,height = 10)
# plotEnrichment(mm_hallmark_fgsea[["HALLMARK_MYOGENESIS"]], mm_star_compare_injured_ranked_genes) + labs(title="HALLMARK_MYOGENESIS")
# dev.off()

head(mm_cryoinjury_sham_fgsea_hallmarks$leadingEdge)
mm_cryoinjury_sham_fgsea_hallmarks_selected <- dplyr::select(mm_cryoinjury_sham_fgsea_hallmarks,
                                                             -leadingEdge,
                                                             -ES,
                                                             -nMoreExtreme) %>%
  arrange(padj)

head(mm_cryoinjury_sham_fgsea_hallmarks_selected)

#mm_cryoinjury_sham_fgsea_hallmarks |> group_by(padj) |> sort()
#as.data.frame(mm_cryoinjury_sham_fgsea_hallmarks) |> group_by(padj) |> sort()

colors_val<-c("#377EB8","#E41A1C")
```


```{r}
ggsave(filename = "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks/mm_cryoinjury_sham_fgsea_hallmarks_hallmark.pdf",width = 8,height =12,dpi = 300,
       plot = mm_cryoinjury_sham_fgsea_hallmarks |> 
         filter(padj < 0.05) |> 
         #top_n(30,wt =- sort(padj))%>%
         as.data.frame(select(pathway,padj,NES)) |> 
         #group_by(padj) |>
         #sort() |> 
         ggplot(aes(y = pathway,x = NES))+
         #shape=ifelse(p.adjust < 0.25,"False Discovery\nRate < 25%","False Discovery\nRate > 25%")))+
         geom_col(aes(fill=ifelse(NES > 0,"Up","Down")))+
         #Not necessary with col.
         #expand_limits(x=c(-5,5))+
         theme_classic() +
         #coord_flip() +
         scale_fill_manual(name="NES", values = colors_val) +
         theme_update() +
         #scale_shape_manual(name="False Discoveries\nAllowance for the\nGene Set Enrichment",values = c(2,19))+
         labs(y="MOUSE HALLMARKS",x="Normalized Enrichment Score",
              subtitle = "Cryoinjury vs Sham",
              title = "\tHallmarks"))#,size="FDR\nSignificance")
```


```{r}
ggsave(filename = "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks/mm_ablation_sham_fgsea_hallmarks_hallmark.pdf",width = 8,height =12,dpi = 300,
       plot = mm_ablation_sham_fgsea_hallmarks |> 
         filter(padj < 0.05) |> 
         #top_n(30,wt =- sort(padj))%>%
         as.data.frame(select(pathway,padj,NES)) |> 
         #group_by(padj) |>
         #sort() |> 
         ggplot(aes(y = pathway,x = NES))+
         #shape=ifelse(p.adjust < 0.25,"False Discovery\nRate < 25%","False Discovery\nRate > 25%")))+
         geom_col(aes(fill=ifelse(NES > 0,"Up","Down")))+
         #Not necessary with col.
         #expand_limits(x=c(-5,5))+
         theme_classic() +
         #coord_flip() +
         scale_fill_manual(name="NES", values = colors_val) +
         theme_update() +
         #scale_shape_manual(name="False Discoveries\nAllowance for the\nGene Set Enrichment",values = c(2,19))+
         labs(x="MOUSE HALLMARKS",y="Normalized Enrichment Score",
              subtitle = "Ablation vs Sham",
              title = "\tHallmarks"))#,size="FDR\nSignificance")
```


```{r}
ggsave(filename = "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks/mm_amputation_sham_fgsea_hallmarks_hallmark.pdf",width = 8,height =12,dpi = 300,
       plot = mm_amputation_sham_fgsea_hallmarks |> 
         filter(padj < 0.05) |> 
         #top_n(30,wt =- sort(padj))%>%
         as.data.frame(select(pathway,padj,NES)) |> 
         #group_by(padj) |>
         #sort() |> 
         ggplot(aes(y = pathway,x = NES))+
         #shape=ifelse(p.adjust < 0.25,"False Discovery\nRate < 25%","False Discovery\nRate > 25%")))+
         geom_col(aes(fill=ifelse(NES > 0,"Up","Down")))+
         #Not necessary with col.
         #expand_limits(x=c(-5,5))+
         theme_classic() +
         #coord_flip() +
         scale_fill_manual(name="NES", values = colors_val) +
         theme_update() +
         #scale_shape_manual(name="False Discoveries\nAllowance for the\nGene Set Enrichment",values = c(2,19))+
         labs(x="MOUSE HALLMARKS",y="Normalized Enrichment Score",
              subtitle = "Amputation vs Sham",
              title = "\tHallmarks"))#,size="FDR\nSignificance")
```
       

```{r}
ggsave(filename = "/media/marius/Samsung_T5/PhD/Projects/prsa/FGSEA/Enrichment/Mmusculus/Hallmarks/mm_uninjured_sham_fgsea_hallmarks_hallmark.pdf",width = 8,height =12,dpi = 300,
       plot = mm_uninjury_sham_fgsea_hallmarks |> 
         filter(padj < 0.05) |> 
         #top_n(30,wt =- sort(padj))%>%
         as.data.frame(select(pathway,padj,NES)) |> 
         #group_by(padj) |>
         #sort() |> 
         ggplot(aes(y = pathway,x = NES))+
         #shape=ifelse(p.adjust < 0.25,"False Discovery\nRate < 25%","False Discovery\nRate > 25%")))+
         geom_col(aes(fill=ifelse(NES > 0,"Up","Down")))+
         #Not necessary with col.
         #expand_limits(x=c(-5,5))+
         theme_classic() +
         #coord_flip() +
         scale_fill_manual(name="NES", values = colors_val) +
         theme_update() +
         #scale_shape_manual(name="False Discoveries\nAllowance for the\nGene Set Enrichment",values = c(2,19))+
         labs(x="MOUSE HALLMARKS",y="Normalized Enrichment Score",
              subtitle = "Uninjured vs Sham",
              title = "\tHallmarks"))#,size="FDR\nSignificance")
```

#### Mouse curated genesets

### Deconvolution

```{r}
# install devtools if necessary
install.packages('devtools')

# install the MuSiC package
devtools::install_github('xuranw/MuSiC')

# load
library(MuSiC)
```



### Plot in Volcanoes top 3 Clusters in Cytoscape
```{r}
require("ggplot2")
require("ggrepel")
volcano_plot_function <- function(df_cond,lg2fc_p,log2fc_n,p_adj,title_,colordown,colorup,down_e,up_e,top_rows_to_select_genes){
  
  gu1 <- c("abat", "abca12", "acvrl1", "adam19", "adam9", "adamts18", "adrb2", "agtr2", "akap6", "alox12", "amigo1", "angptl3", "ap3b2", "apbb1", "aph1c", "apln", "aqp11", "arf6", "arhgap21", "arhgap35", "arhgef10l", "arl3", "arntl", "arpc4", "asf1b", "ash1l", "asph", "atg2a", "atg9b", "atp6ap2", "aurkb", "avil", "azin1", "b2m", "bag3", "banf1", "bcl11b", "bcl2", "bicdl2", "bin1", "blk", "bmerb1", "bmper", "bora", "cacna1d", "calcr", "card11", "carmil1", "casp1", "cav1", "cckbr", "ccl11", "ccne2", "cd226", "cd74", "cd79a", "cd8a", "cdc42ep5", "cdc6", "cdh17", "cdh26", "cdk2", "celsr1", "cenpa", "cenpk", "cenpp", "cenpt", "cep72", "cflar", "cgas", "chek2", "chrna7", "chtf18", "chtf8", "cldn1", "cldn7", "clec4a2", "cnot6", "colgalt1", "cplx2", "crbn", "cst3", "ctcf", "cuedc2", "cyp2j6", "cyp4a31", "cyp51", "cyth3", "d130043k22rik", "dapk1", "dbn1", "def8", "dnaaf4", "dnai3", "dnajb9", "dnd1", "dnmt1", "drd1", "dscc1", "dsn1", "dtl", "dync2h1", "e2f7", "efcab7", "efhc2", "eif4ebp1", "enah", "enpep", "esam", "exosc6", "f10", "f2r", "f2rl1", "f2rl3", "faap24", "fabp7", "fam111a", "fancb", "fancf", "fanci", "fbln2", "fen1", "fermt1", "foxf1", "foxf2", "frem2", "fzd8", "gcnt3", "gem", "gfus", "ghrhr", "glra1", "gm5136", "gna11", "gna15", "gpr27", "grin2a", "gsk3b", "gspt2", "h1f0", "h3f3b", "hamp", "haus1", "haus3", "haus6", "hc", "hells", "hlx", "hmgcr", "hras", "hspa5", "htr2b", "hyal3", "id2", "ifih1", "ifng", "il10", "il12b", "il15", "il15ra", "inava", "ing5", "ino80", "ino80c", "insm1", "irs2", "irx3", "itga4", "itgb8", "itk", "jmy", "kank1", "kcne4", "kcng2", "kcng3", "kcnh3", "kcnip2", "kcnj13", "kcnj2", "kcnj6", "kcnj8", "kcnq3", "kifc5b", "kiss1r", "kit", "lgals8", "lima1", "llgl1", "lpcat1", "lrrk2", "lurap1", "ly9", "mag", "map6d1", "mapk1", "mapre1", "matn1", "mau2", "mcm2", "mcm4", "mcm5", "mcm6", "mcmbp", "mcph1", "meig1", "mgp", "midn", "mip", "mmp17", "mnd1", "mov10", "mpg", "mrrf", "mtfr2", "mul1", "mustn1", "myo15", "nadk", "nav2", "ncapd3", "nck1", "ncstn", "nedd4", "nedd9", "nell2", "nexmif", "ngf", "nid1", "nkd2", "npff", "npm2", "nppc", "nsd2", "nsf", "nsl1", "nuggc", "obi1", "orc5", "ormdl3", "osr1", "oxct1", "parp9", "pask", "pck1", "pcna", "pde12", "pdpk1", "perp", "pex11a", "pglyrp2", "phb2", "phf13", "phpt1", "pif1", "pip4p1", "pknox1", "pla2g3", "plag1", "plcb4", "pld2", "plekhh2", "poc1a", "pold1", "pole", "poli", "ppfia2", "ppp2ca", "prkacb", "prxl2c", "psmc3ip", "psme1", "ptpn5", "pttg1", "rab15", "rab27a", "rab27b", "rad23b", "rad51ap1", "radx", "rag2", "rangap1", "rapgef4", "rasal3", "rasip1", "rb1cc1", "rbbp4", "rcc1", "rcc2", "rec114", "recql4", "ret", "rev1", "rev3l", "rflnb", "rgs4", "rhoh", "rmi2", "rnf139", "rnf144a", "rnf180", "rpa2", "rpa3", "rrad", "rrm1", "rsad2", "scin", "scoc", "sec16b", "sema3d", "septin5", "serpina10", "serpinb1a", "serping1", "sesn2", "sgo2a", "sh2d1a", "sh3bp1", "sharpin", "shfl", "shh", "shpk", "sim1", "slc11a2", "slc12a2", "slc18a2", "slc2a5", "slc39a10", "slc4a3", "smarcad1", "smarcd3", "smc6", "smpdl3b", "socs4", "sox11", "sox8", "sp3", "spata22", "srgn", "sri", "stag1", "stag2", "stk25", "stx3", "stxbp1", "stxbp2", "stxbp4", "stxbp5l", "sumo3", "sun2", "syce1l", "tac1", "tacc2", "tacr2", "tafa5", "taok1", "tbc1d30", "tbx18", "tdg", "tdrd9", "terb2", "terf1", "themis", "thy1", "tjp2", "tlr7", "tmem230", "tnfrsf1b", "tnfsf13b", "tnks1bp1", "tnr", "tnrc6b", "top3b", "tpm1", "tprkb", "traf3ip1", "trib3", "tsku", "tubgcp4", "twf1", "tyrobp", "ubash3b", "ubb", "ube2i", "uhrf1", "vav3", "vegfc", "vgll4", "vps4b", "wdhd1", "wdpcp", "wdr45b", "wipf1", "wnt4", "xpo1", "xrcc5", "zbtb16", "zbtb32", "zfp384", "zfp703", "zfy2", "zswim7", "zwilch")
  #df_cond |> filter(zfin_id_symbol %in% gu1)
  gu11 <- do.call(rbind,lapply(gu1,function(g) 
    df_cond |> filter(str_detect(zfin_id_symbol,regex(g,ignore_case = TRUE)))))
  
  df_cond_1 <- df_cond |>
    mutate(Regulation = ifelse(log2FoldChange > lg2fc_p & padj < p_adj,"UP",
                               ifelse(log2FoldChange < log2fc_n & padj < p_adj,"DOWN","NO"))) |>
    mutate(Regulation = ifelse(is.na(Regulation),"NO",Regulation)) |>
    mutate(HS = ifelse(entrezgene_id %in% gu11$entrezgene_id,"Yes","No"))
    #mutate(HS = ifelse(padj > (df_cond |> arrange(padj) |> slice_head(n = top_rows_to_select_genes)  |> slice_tail(n=1) |> pull(padj)),"NO","YES"))
  
  #mutate(Selected_Gene_Names = filter(log2FoldChange > 10 & padj < 1e-20)[,8]) |> 
  p1 <- ggplot(data = df_cond_1,aes(x = log2FoldChange,y = -log10(padj),col = Regulation)) +
    #geom_point() +
    #geom_jitter(width = 0.01,height = 0.01,size=3) +
    geom_jitter(position = position_jitter(height = 0.0000001, width = 0.0000001,seed = 123456),size=1) + # + xlim(0,0.5) + ylim(0,0.5)width = 0.01, 
    scale_color_manual(labels = c(down_e,"NO",up_e),
                       values = c(colordown,
                                  "black",
                                  colorup)) +
    geom_vline(xintercept=c(-1, 1), col="red") +
    geom_hline(yintercept=-log10(0.05), col="red") + 
    labs(title = title_) +
    xlim(-30,30) +
    ylim(0,160) +
    theme(panel.grid = element_blank(),
          panel.background = element_rect(fill = "transparent"),
          panel.border = element_rect(fill = "transparent"),
          legend.title = element_text(size=25),
          legend.text = element_text(size=25),
          axis.text = element_text(size=25),
          axis.title=element_text(size=25,face="bold")) +
    #theme_minimal() +
    #geom_text_repel(data = df_cond_1,label = filter(df_cond_1$HS == "YES") |> pull(zfin_id_symbol))
    # geom_text_repel(data = df_cond_1,label = ifelse(df_cond_1$HS == "Yes",as.character(df_cond_1$zfin_id_symbol),""),
    #                 #data = df_cond_1,label = df_cond_1$zfin_id_symbol[df_cond_1$HS == "Yes"],
    #                 max.overlaps = 150,
    #                 nudge_y = 3.15,
    #                 color="black",size=5) +
    geom_text_repel(nudge_y = 3.15, nudge_x = 0.91,
                    max.overlaps = 20,
                    direction = "x",segment.size = 0.8,
                    data = df_cond_1,label = ifelse(df_cond_1$HS == "Yes",as.character(df_cond_1$zfin_id_symbol),""),
                    #data = df_cond_1,label = df_cond_1$zfin_id_symbol[df_cond_1$HS == "Yes"],
                    color="black",size=8) +
    geom_point(position = position_jitter(height = 0.0000001, width = 0.0000001,seed = 123456),#width = 0.00001
               pch=21,size=1.1,color="red",
               data = df_cond_1 |> filter(HS == "Yes"),
               aes(x = log2FoldChange,y = -log10(padj))) +
    geom_point(position = position_jitter(height = 0.0000001, width = 0.0000001,seed = 123456),#width = 0.00001
               pch=21,size=1.2,color="red",
               data = df_cond_1 |> filter(HS == "Yes"),
               aes(x = log2FoldChange,y = -log10(padj)))
  return(list(plot=p1,data_volc = p1$data))}
```

```{r,fig.width=13,fig.height=15}

#Load list from cytoscape clusters

#Extract the GOBP top from the clusters 
uninjured_data_sham_mmus_exact_from_fil_gem_clean_unique_go_bp_v2$go_table_selected_condition |> head()


#Extract the genes from the GOBP
gobps(c(GO:0043149
GO:0051492
GO:0030038
GO:0042059
GO:1901185
GO:0007175))
gu1 <- tolower(c("ARHGEF18","ARRB1","ERRFI1","INPP5K","PHLDB2","PPM1E","RNF126","SOCS5","SRC","SYNPO","TESK1","ZFYVE28"))

lists_platform_batch_corrected_deseq_condition_ann$Uninjured_vs_Sham |> filter(zfin_id_symbol %in% gu1)


gu11 <- do.call(rbind,lapply(gu1,function(g) 
  lists_platform_batch_corrected_deseq_condition_ann$Uninjured_vs_Sham |> filter(str_detect(zfin_id_symbol,regex(g,ignore_case = TRUE)))
))


tests <- lists_platform_batch_corrected_deseq_condition_ann$Uninjured_vs_Sham |> mutate(HS = ifelse(entrezgene_id %in% gu11$entrezgene_id,"Yes","No"))


#Plot the Genes in the Volcanoes.
```




```{r}
gu1 <- c("abat", "abca12", "acvrl1", "adam19", "adam9", "adamts18", "adrb2", "agtr2", "akap6", "alox12", "amigo1", "angptl3", "ap3b2", "apbb1", "aph1c", "apln", "aqp11", "arf6", "arhgap21", "arhgap35", "arhgef10l", "arl3", "arntl", "arpc4", "asf1b", "ash1l", "asph", "atg2a", "atg9b", "atp6ap2", "aurkb", "avil", "azin1", "b2m", "bag3", "banf1", "bcl11b", "bcl2", "bicdl2", "bin1", "blk", "bmerb1", "bmper", "bora", "cacna1d", "calcr", "card11", "carmil1", "casp1", "cav1", "cckbr", "ccl11", "ccne2", "cd226", "cd74", "cd79a", "cd8a", "cdc42ep5", "cdc6", "cdh17", "cdh26", "cdk2", "celsr1", "cenpa", "cenpk", "cenpp", "cenpt", "cep72", "cflar", "cgas", "chek2", "chrna7", "chtf18", "chtf8", "cldn1", "cldn7", "clec4a2", "cnot6", "colgalt1", "cplx2", "crbn", "cst3", "ctcf", "cuedc2", "cyp2j6", "cyp4a31", "cyp51", "cyth3", "d130043k22rik", "dapk1", "dbn1", "def8", "dnaaf4", "dnai3", "dnajb9", "dnd1", "dnmt1", "drd1", "dscc1", "dsn1", "dtl", "dync2h1", "e2f7", "efcab7", "efhc2", "eif4ebp1", "enah", "enpep", "esam", "exosc6", "f10", "f2r", "f2rl1", "f2rl3", "faap24", "fabp7", "fam111a", "fancb", "fancf", "fanci", "fbln2", "fen1", "fermt1", "foxf1", "foxf2", "frem2", "fzd8", "gcnt3", "gem", "gfus", "ghrhr", "glra1", "gm5136", "gna11", "gna15", "gpr27", "grin2a", "gsk3b", "gspt2", "h1f0", "h3f3b", "hamp", "haus1", "haus3", "haus6", "hc", "hells", "hlx", "hmgcr", "hras", "hspa5", "htr2b", "hyal3", "id2", "ifih1", "ifng", "il10", "il12b", "il15", "il15ra", "inava", "ing5", "ino80", "ino80c", "insm1", "irs2", "irx3", "itga4", "itgb8", "itk", "jmy", "kank1", "kcne4", "kcng2", "kcng3", "kcnh3", "kcnip2", "kcnj13", "kcnj2", "kcnj6", "kcnj8", "kcnq3", "kifc5b", "kiss1r", "kit", "lgals8", "lima1", "llgl1", "lpcat1", "lrrk2", "lurap1", "ly9", "mag", "map6d1", "mapk1", "mapre1", "matn1", "mau2", "mcm2", "mcm4", "mcm5", "mcm6", "mcmbp", "mcph1", "meig1", "mgp", "midn", "mip", "mmp17", "mnd1", "mov10", "mpg", "mrrf", "mtfr2", "mul1", "mustn1", "myo15", "nadk", "nav2", "ncapd3", "nck1", "ncstn", "nedd4", "nedd9", "nell2", "nexmif", "ngf", "nid1", "nkd2", "npff", "npm2", "nppc", "nsd2", "nsf", "nsl1", "nuggc", "obi1", "orc5", "ormdl3", "osr1", "oxct1", "parp9", "pask", "pck1", "pcna", "pde12", "pdpk1", "perp", "pex11a", "pglyrp2", "phb2", "phf13", "phpt1", "pif1", "pip4p1", "pknox1", "pla2g3", "plag1", "plcb4", "pld2", "plekhh2", "poc1a", "pold1", "pole", "poli", "ppfia2", "ppp2ca", "prkacb", "prxl2c", "psmc3ip", "psme1", "ptpn5", "pttg1", "rab15", "rab27a", "rab27b", "rad23b", "rad51ap1", "radx", "rag2", "rangap1", "rapgef4", "rasal3", "rasip1", "rb1cc1", "rbbp4", "rcc1", "rcc2", "rec114", "recql4", "ret", "rev1", "rev3l", "rflnb", "rgs4", "rhoh", "rmi2", "rnf139", "rnf144a", "rnf180", "rpa2", "rpa3", "rrad", "rrm1", "rsad2", "scin", "scoc", "sec16b", "sema3d", "septin5", "serpina10", "serpinb1a", "serping1", "sesn2", "sgo2a", "sh2d1a", "sh3bp1", "sharpin", "shfl", "shh", "shpk", "sim1", "slc11a2", "slc12a2", "slc18a2", "slc2a5", "slc39a10", "slc4a3", "smarcad1", "smarcd3", "smc6", "smpdl3b", "socs4", "sox11", "sox8", "sp3", "spata22", "srgn", "sri", "stag1", "stag2", "stk25", "stx3", "stxbp1", "stxbp2", "stxbp4", "stxbp5l", "sumo3", "sun2", "syce1l", "tac1", "tacc2", "tacr2", "tafa5", "taok1", "tbc1d30", "tbx18", "tdg", "tdrd9", "terb2", "terf1", "themis", "thy1", "tjp2", "tlr7", "tmem230", "tnfrsf1b", "tnfsf13b", "tnks1bp1", "tnr", "tnrc6b", "top3b", "tpm1", "tprkb", "traf3ip1", "trib3", "tsku", "tubgcp4", "twf1", "tyrobp", "ubash3b", "ubb", "ube2i", "uhrf1", "vav3", "vegfc", "vgll4", "vps4b", "wdhd1", "wdpcp", "wdr45b", "wipf1", "wnt4", "xpo1", "xrcc5", "zbtb16", "zbtb32", "zfp384", "zfp703", "zfy2", "zswim7", "zwilch")
```


```{r,fig.width=13,fig.height=15}

uninjured_volcano_t <- volcano_plot_function(df_cond=lists_platform_batch_corrected_deseq_condition_ann$Uninjured_vs_Sham,
                                           lg2fc_p = 1,log2fc_n = -1,
                                           p_adj = 0.05,
                                           #top_rows_to_select_genes = 0,
                                           title_ = "DESeq2  Uninjured vs Sham",
                                           colordown = "#999999",down_e ="Sham",
                                           up_e ="Uninjured",colorup = "#009E73")

uninjured_volcano_t
```

```{r,fig.width=13,fig.height=15}
amputated_volcano_t <- volcano_plot_function(df_cond=lists_platform_batch_corrected_deseq_condition_ann$Amputation_vs_Sham,
                                           lg2fc_p = 1,log2fc_n = -1,
                                           p_adj = 0.05,
                                           #top_rows_to_select_genes = 0,
                                           title_ = "DESeq2 Amputated vs Sham",
                                           colordown = "#999999",down_e ="Sham",
                                           up_e ="Amputated",colorup = "#CC79A7")
amputated_volcano_t
```


### Save all
```{r}
#save.image(file = "D:/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v3/Workflow_cytoscape_gem_pca_volcano_goprofiler.RData")
save.image(file = "/media/marius/Samsung_T5/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/R/Workflow_cytoscape_gem_pca_volcano_goprofiler_v4_check_2.RData")
#save.image(file = "C:/Users/marius/Documents/Phd/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/R/Workflow_cytoscape_gem_pca_volcano_goprofiler_v4.RData")
#save.image(file = "D:/PhD/Projects/prsa/Outputs/r/Cytoscape/Cleaned_Annotation_Ensembl_v4/R/Workflow_cytoscape_gem_pca_volcano_goprofiler_v4_check.RData")
```